# Protocol: Artificial Ecosystem Selection

### Andrew Morris

### Inoculation and Incubation Conditions

First, I will start 9 vials at 3 different inoculation rates with 3 vials per inoculation rate. Sterile soil collected from the Nature Conservancy seasonally wet prairie in Eugene will be mixed with 0.1%, 1%, and 5% by weight of soil inoculum from the South Slough estuary in Coos Bay. This site is known to be methanogenic and so offers the potential for a diverse methanotroph community. Sterile soil, inoculum, and sterile DI water will be mixed in sterile bins inside of a biosafety hood for 5 minutes. Then, 50 g of soil at 60% of field capacity will be added to a 75 mL vial and capped with a rubber septum. This process will be repeated for each transmission rate with separate lines being transferred one at a time to avoid cross-contamination between lines.

###### Alternative approach: 

Each vial will be filled with sterile soil by weight depending on transmission rate and then soil inoculum will be added to a total mass of 50 g. Sterile DI water will be added to bring the soil up to 60% of field capacity and then capped with a rubber septum. 

These will be incubated under ambient temperature and light sitting on the bench top. In order to maintain aerobic conditions, they will be uncapped and aerated then re-capped and spiked with methane every 2-3 days. This procedure will be carried out in a biosafety hood to minimize cross-contamination between lines, within lines, and from outside sources. I will test methane oxidation rate of these 9 vials every 2 or 3 days immediately following capping and re-spiking.

### Measuring Methane Oxidation

Methane oxidation will be determined by capping and spiking each vial to a headspace concentration of 1000 ppm methane (using approximately 5 mL of 1% methane). A 1 mL headspace sample will be collected at 0.33, 3, 6, and 9 hours and injected into a gas chromatograph with a flame ionization detector to determine methane concentration. I will then apply a first-order exponential decay function to the measurements to determine the maximum velocity rates (Vmax, units = Âµg CH4 cm-3 d-1) of oxidation. 

### Determining Generation Time

Over several oxidation measurements on each vial, I expect to see a gradual increase in methane oxidation rate as the methanotroph community is able to colonize its sterile substrate. Once this increase begins to plateau, I will set the time period from inoculation to plateau as the generation time. If this time period is not logistically feasible (e.g. less than one week or greater than three weeks) than I will choose an intermediate generation time to balance logistics and biology. If each inoculation rate requires a different generation time then I might have to choose a generation time that is intermediate between all of them or incubate them at different generation times for the second experiment. 

### Determining Transmission Rate

Once generation time has been determined, I will start the experiment again with three lines with different inoculation rates (0.1%, 1%, 5%) this time with 9 vials per line for a total of 27 vials. I will inoculate and incubate these vials under the same conditions as the previous experiment, except I will only measure methane oxidation rate once after the time period determined from the first experiment. Once oxidation rates have been determined for each vial, I will examine the distribution of oxidation rates within each line and choose the top 3 vials or the top 30th percentile. Soil from these vials will be homogenized and used to inoculate the next generation of vials. 

This procedure will continue for 10+ generations or until significant divergence from the first generation is achieved. This will establish whether or not there is a response to selection for low affinity methane oxidation. Each transmission rate offers a balance between fidelity of inheritance, or how much generation `t` resembles generation `t-1`, and increased variation due to bottlenecks. Whichever transmission rate offers the strongest response to selection will be used in the following experiment.

### The Actual Experiment

For this experiment, I will use the generation time determined in the first experiment and the transmission rate from the second experiment (probably with the generation time as determined for that transmission rate in experiment 1). I will set the total number of vials in this experiment based on how many vials I can measure the oxidation rate of within one day (approximately 40). Then I will split those vials into multiple lines as determined by the complementary goals of having enough variation within one line and enough variation among lines to break up spurious associations between taxa. For that I will choose either 4 lines of 10 vials each or 3 lines with 13 lines each (approximately). 

This experiment will be established similarly to the previous two. This time, a single inoculation rate will be used across all replicate lines. The first generation will probably be established by homogenizing sterile soil with natural soil and then aliquoting 50 g of that mixture into each of the ~40 vials. These vials will be incubated under aerobic conditions, 60% of field capacity, and 1000 ppm methane on the bench top, as before. At the end of generation one, methane oxidation rates will be determined and the top 30% or top 3-4 vials will be selected from each line to inoculate the next generation of vials for that line at the chosen transmission rate. Samples of 1-3 g of soil from each vial in each generation will be stored in eppendorf tubes at -80 for later dna extraction. Remaining sample not used for inoculation will be saved in the freezer for future chemical analyses.

This process will continue for 10+ generations. At the end, I will choose which vials to extract DNA from and sequence. This will certainly be the first and last generation and possibly a middle generation. I will sequence the 16S rRNA gene to establish taxonomy of the prokaryotic community.

### Analyses

- Differential abundance of taxa between the end and the beginning of the experiment (I expect this to be a comparison between high and low methane oxidation rates, but we'll see).
- Principle components regression for all samples to see which taxa are correlated with oxidation rates.
- One-way ANOVA to test the difference in methane oxidation rate between the beginning and end of the experiment.
- Alternatively, an autoregressive model to account for the covariance between consecutive generations within line to estimate the response to selection in methane oxidation.
- ...