---
title: Artificial ecosystem selection reveals relationships between microbiome composition and ecosystem function
author: "Andrew H. Morris^1^* and Brendan J. M. Bohannan^1^"
output:
  word_document:
    reference_docx: reference.docx
bibliography: bibliography.bib
csl: vancouver.csl
---

^1^ Institute of Ecology and Evolution, University of Oregon, Eugene, OR, USA  
\* correspondence: Andrew H. Morris, amorris3\@uoregon.edu  
Competing interests: We declare we have no competing interests.

# Introduction

Microbiomes mediate a variety of important ecosystem functions. This suggests
that understanding variation in microbiome composition might be important for
predicting the rates of ecosystem functions 
[@conrad1996; @schimel1998; @crowther2019]. 
Many studies have attempted to document such a relationship
[@cavigeli2000; @balser2005; @waldrop2006; @strickland2009; wagg2014; wagg2020].
However, it is difficult to isolate the effect of variation in microbiome
composition from other drivers of variation in ecosystem function. For example,
microbiome manipulations often confound variation in diversity, composition, and
abundance.  Likewise, comparative studies must control for abiotic variation
statistically to identify the direct effect of microbiome variation. This
approach relies on measuring the relevant abiotic variables and cannot account
for unmeasured environmental variation. Here, we overcome these limitations by
using a selection approach to estimate the degree to which an ecosystem function
varies with microbiome composition.

There have been two general categories of approaches that investigators have
used to estimate the degree to which an ecosystem function varies with
microbiome composition: comparative and manipulative. Comparative
studies sample natural variation in an ecosystem function across different
habitats and simultaneously measure variation in community composition.
Investigators can then correlate ecosystem function with community structure
while trying to control for environmental variation. These approaches have
documented important relationships between microbiomes and ecosystem
functions. For example, a meta-analysis of these studies observed a small but
significant contribution of the microbiome to variation in ecosystem
function after controlling for environmental variation [@graham2016]. In
addition, studies focusing on the correlation between the rate of an ecosystem
function and the abundance of an associated functional marker gene observe a
significant correlation, though this relationship is rare and contingent upon
both the function and the ecosystem sampled [@rocca2015]. However, comparative
studies come with some unique challenges and limitations. One issue is that
microbiome attributes tend to covary with the abiotic conditions within an
environment, and whether and how the investigator controls for these variables
can affect the conclusions of such a comparative study. In addition, it is
difficult to know a priori which environmental variables or community attributes
to measure. Finally, while these approaches can establish a potential magnitude
and direction for these relationships, it is often difficult to identify the taxa or
genes that explain the connection between composition and function.

The other broad category of approaches used to address this question are
manipulative approaches. Manipulative experiments try to alter microbial
community composition and observe the effect on function. For example,
reciprocal transplant and common garden experiments have shown that microbiomes
originating from different ecosystems inoculated into the same substrate or
introduced into a common environment display distinct functional rates 
[@cavigeli2000; @balser2005; @waldrop2006; @strickland2009]. In addition,
manipulating
diversity by filtering communities by cell size or through dilution has been
shown to alter the rate of ecosystem functions [@wagg2014; @wagg2020]. However,
manipulating the microbiome directly is challenging. Each
of the manipulative approaches applied confound community composition with other
factors. For example, reciprocal transplant and common garden experiments
confound community composition with the abiotic conditions carried on the
inoculum while manipulating composition through dilution
may confound composition with biomass [@reed2007]. 

In this study, we sought to build on the observations of comparative and
manipulative studies by applying a different approach to the question of whether
variation in the microbiome contributes to variation in the rate of an ecosystem
function. We used artificial ecosystem selection to select for microbiomes that
performed a greater rate of ecosystem function [@swenson2000; @blouin2015; @panke-buisse2015]. We then tested whether variation
in the microbiome contributed to variation in the rate of ecosystem function and
identified microbiome attributes that might explain this relationship. There are
several potential advantages to this approach for documenting
microbiome-function relationships and for investigating potential mechanisms
underlying those relationships. By passaging microbiomes over multiple rounds
using a common environment, we can weaken the covariance between microbes and
the environment by repeatedly diluting abiotic effects. In addition, by
inoculating our ecosystems with equally sized subsamples, we eliminate the need
to generate microbiome variation through methods that are confounded with
biomass or cell size. Lastly, by comparing our artificially selected community
to a control community resulting from random selection, we can both control for
changes in the environment over time and identify significant genes or taxa
associated with our selection treatment to establish a potential mechanism that
explains this variation. 

We applied artificial ecosystem selection to soil microbiomes by selecting on
soil methane oxidation rate. We chose this function because methane is a
globally important greenhouse gas and methane oxidation by soil bacteria is the
primary biological sink for atmospheric methane as well as possibly the main driver
of temporal methane dynamics [@kirschke2013]. In addition, there is evidence that
soil methane oxidation rate may vary with microbiome composition based on
comparative studies in a variety of arctic and tropical ecosystems 
[@meyer2020; @meyer2020a; @freitag2009; @freitag2010]. Finally, methanotrophy is one of
the most deeply conserved
microbial physiologies and is represented in a narrow range of taxa, which
suggests that the taxonomic composition of the microbiome is more
likely to be associated with the rate of methane oxidation than other broader or
more shallowly conserved functions [@schimel1998; @martiny2013]. 

In this study, we evaluated whether variation in the microbiome contributes to
variation in the rate of ecosystem functions. To address this problem, we
applied artificial ecosystem selection to methane oxidation rate of the soil
microbiome and answered the following questions: Does variation in the relative
abundance of microbial taxa contribute to variation in soil methane oxidation
rate in the lab? How much of the variation in soil methane oxidation rate can
be attributed to the microbiome? What attributes of the microbiome regulate
variation in methane oxidation rate? And do these attributes match our
assumptions about what regulates methane oxidation rate in nature?



The ultimate goal of this study could be summarized as "do microbes matter to
ecosystem function?" And there are perhaps three ways we could address that
question. One would be to ask whether microorganisms mediate the function. This
question is perhaps trivial. In the case of methane emissions from soil, we know
that microorganisms produce and consume methane. We could also ask what would
happen if we removed the microorganisms. Again, this question is trivial. We
don't know of any abiotic reactions in soil that consume methane (outside of the
troposphere, the stratosphere, and the marine boundary layer; kirschke).
Therefore, removing the microbiome would likely eliminate methane production and
consumption. The third question we could ask is whether *variation* in the
microbiome is associated with *variation* in the rate of ecosystem function. The
answer to this question is not so obvious.

![Figure 1. Conceptual diagram of the potential relationship between environmental variation, microbiome variation, and variation in the rate of an ecosystem function. Without experimentally or statistically controlling for the underlying environmental variation, it can be difficult to distinguish between the direct effect of the microbiome and the effect of the microbiome due to covariance with the environment.](Figure1.png)

If we look at variation in the rate of a microbially-mediated ecosystem function
across a set of habitats, that variation could be associated with variation in
the abiotic environmental conditions, variation in the composition of
microbiomes, or some interaction or covariance between the two (see Figure 1). To address the
question of whether variation in the microbiome matters for ecosystem function,
we need to identify the component of variation uniquely associated with the
microbiome. If we sample natural variation, as in the comparative approach, it
can be difficult to distinguish between the direct effect of the microbiome and
the effect of the microbiome due to covariance with the environment. Likewise,
with the manipulative approach even though we are placing ecosystem propagules
into a common environment or onto a common substrate, the propagule itself
carries with it environmental conditions that cannot be removed. Generating
variation in microbiome composition through filtering or other similar means
often confounds composition with other factors, for example, by extinguishing
all fungi while sparing the bacteria, in which case it is uncertain whether this
represents the effect of composition or simply the gross effects of mass
extinction.

Using artificial ecosystem selection, we can generate a set of ecosystems in
bottles using a common, sterile substrate and inoculate them with a propagule
containing a diverse microbiome that is a subsample of an existing microbiome.
This generates variation in microbiome composition among the ecosystems. We can
then measure the rate of an ecosystem function for each bottle and select the
bottles with the highest (or lowest) rate of ecosystem function. The selected
ecosystems are then homogenized and used as the propagule to inoculate a new set of
ecosystems containing the same sterile substrate as the initial set. By
passaging the microbiome in this way over multiple generations we can observe
whether varying the microbiome can generate variation in the rate of our
ecosystem function. If we observe no change in ecosystem function from
generation to generation, we would conclude that this ecosystem function in this
set of ecosystems is driven primarily by environmental variation or technical
error among the replicates. However, if we observe a response to selection, that
is, a shift in the mean ecosystem function between successive passages, we would
conclude variation in the microbiome can generate variation in the rate of
ecosystem function.

