---
title: Artificial ecosystem selection to deduce the mapping between microbial community structure and ecosystem function
author: "Andrew H. Morris and Brendan J. M. Bohannan"
output:
  html_document: default
  word_document:
    reference_docx: reference.docx
editor_options:
  chunk_output_type: inline
---

```{r options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
library(tidyverse)
library(captioner)
library(broom)
library(broman)
library(morris)
library(metafor)
figures <- captioner()
figures("deviance", "Deviance of CH~4~ oxidation rate for the Positive selection treatment relative to the Neutral selection treatment. Deviance is calculated as the mean of P within each passage minus the mean of N within each passage. Errorbars are the sums of the standard errors for P and N for each passage. The best fit line is an ordinary least squares fit taking into account the standard errors of the deviations. The gray ribbon is the 95% confidence interval for the regression.")
figures("herit", "Regression of offspring CH~4~ flux on mid-parent CH~4~ flux. Mid-parent is the mean of jars selected to inoculate the next passage. The mid-offsprings are the mean of all twelve jars produced in one passage. The best fit line is an ordinary least squares regression line and the gray ribbon is the 95% confidence interval for the regression coefficient.")


```

# Abstract

# Introduction 

# Materials & Methods

## Selection experiment
We performed an artificial ecosystem selection experiment similar to Swenson et al. (2000) by passaging replicate soil microbial communities. The trait we selected on was CH~4~ oxidation rate. Our experiment had two selection lines with twelve jars each for a total of twenty-four jars per passage. One line was a positive selection line where the two or three jars with the highest CH~4~ oxidation rate were chosen to inoculate the next set of jars. The other line was a neutral selection line where an equal number of jars as the positive line were chosen at random among the twelve. The number of jars chosen was based on the distribution of fluxes among the positive jars, i.e., we chose the top three jars unless only two jars had considerably greater oxidation rates based on visual inspection of histograms.

The initial soil microbial community was sampled from the top 10 cm of an upland mineral soil under a deciduous forest ecosystem near the University of Oregon campus. Incubations were performed in 500 mL mason jars (Ball Corporation, Broomfield, Colorado, USA) with rubber septa installed in the lids. Jars were incubated at ambient temperature on the benchtop. To create the first set of jars, twenty-four jars were created in an identical manner and then randomly assigned to either the positive or negative selection line. Each jar was sterilized with 70% ethanol to which was added 45 g of autoclaved potting mix, 5 g of live soil, and 3.5 mL of sterile deionized water to bring them to <mark> 60%??? of field capacity </mark>. The jars were then homogenized, capped, and injected with 4.3 mL of 99% CH~4~ to bring the headspace concentration to approximately 1000 ppm CH~4~. Jars were flushed and respiked twice per week to maintain elevated CH~4~ concentrations until travel, holidays, and COVID-19 got in the way and then we just did it as often as we could. Jars were incubated for approximately three weeks and then flux measurements were taken.

Once CH~4~ oxidation rates were determined, the top two or three jars were chosen for selection in the positive line and an equal number of jars were randomly chosen from the neutral line. The selected jars for each line were homogenized and used as the 5 g live soil for the subsequent passage. Twelve new jars were created for each line in the same manner as above using the inoculum for that line. This procedure continued until a divergence was observed in the mean flux of each line for two consecutive generations. Jars in passage number 2 and 5 were homogenized and a sample was collected and stored at -80$^\circ$C for later DNA extraction.

## Methane flux measurements
Methane oxidation rates were determined after flushing and spiking jars to approximately 1000 ppm CH~4~. Head-space samples of 1 mL were collected from each jar immediately after spiking and then at time points 3, 6, 24, and 48 hours for a 5-point curve. Samples were immediately injected into a <mark>???</mark> gas chromatograph fitted with an electron capture detector (company, city, state, USA) to determine the headspace CH~4~ concentration. Fluxes were calculated from a first-order exponential decay function as decay constant `k` with units<mark>???</mark>. Oxidation rates are presented as the additive inverse of `k` (i.e., `-k`) so that a more positive value represents a faster oxidation rate.

## Soil DNA extraction and sequencing

Soil DNA was extracted from a 0.25 g subsample of soil from the starting inoculum and from each jar in passage 2 and 5. Negative controls were extracted from autoclaved potting mix and DNase-free water. Extractions were performed using the DNeasy PowerSoil kit (QIAGEN, Hilden, Germany) and quantified using Qubit (company, city, state). In order to estimate the diversity and relative abundance of the bacterial and archaeal taxa in our soil ecosystems, we sequenced the V3-V4 region of the 16S rRNA gene using the Earth Microbiome Project protocol and primers<mark>???</mark>. In addition, to estimate the diversity and relative abundance of CH~4~ oxidizing taxa, we sequenced the particulate methane monooxygenase subunit A gene (*pmoA*) using primers<mark>???</mark>. Sequencing was performed on the Illumina MiSeq in v3 mode with 300 bp paired-end reads.

## Bioinformatics
Bioinformatic analyses were conducted using the `QIIME2` bioinformatics platform (Bolyen et al. 2019). Raw sequencing reads were demultiplexed using ??? (citation), denoised using `DADA2` (citation). Taxonomic assignment for the 16S reads was performed using ??? based on the Silva database. All subsequent tatistical analyses were performed in the `R` statistical programming environment (R Core Team 2020). Richness estimates were made using the `breakaway` package and diversity indices using the `DivNet` package (Willis et al. ???).

## Statistical Analysis

### Response to selection

To test whether there was a significant change in CH~4~ oxidation rate (k) as a response to selection, we tested the deviation between the positive and neutral selection lines over the five passages. We calculated deviation as the difference in the mean oxidation rate between the positive and neutral line within each passage with standard errors equal to the sum of the errors for each of the positive and neutral lines. We then fit an ordinary least squares regression model taking into account the standard errors of the deviations. The model was fit using the `rma` function from the `metafor` package in `r` (Viechtbauer 2010).

### Ecosystem heritability

To test the heritability of CH~4~ oxidation rate as an ecosystem property, we calculated narrow-sense heritability ($h^2$) as the regression of offspring on mid-parent (Goodnight 2000, Falconer 1997). The mid-parent was the mean for all three selected jars and the offspring was the mean for all jars produced by those parents.

In addition, we calculated the inbreeding coefficient, $F$, as:

$$
F = 1 - \left(1 - \frac{1}{2N} \right) ^{T}
$$

where $N$ is the population size and $T$ is the number of generations (Goodnight 2000 or Falconer 1997).

### Taxa differentially abundant in the positive selection treatment

To identify taxa that responded to selection on CH~4~ oxidation rate, we tested differential abundance using a beta-binomial model (Martin et al. 2020). We compared taxa in passage 5 between the positive and neutral treatment to identify taxa that 

# Results

## Response to selection


```{r dev_model}
fluxes <- read_tsv('../Output/fluxes.tsv')


deviance_se <- 
  fluxes %>% 
  select(treat, passage, estimate) %>% 
  group_by(passage, treat) %>% 
  summarize(se = se(estimate)) %>% 
  pivot_wider(names_from = treat, values_from = se) %>% 
  mutate(deviance_se = p + n) %>% 
  pull(deviance_se)
  

deviance <-
fluxes %>% 
  select(treat, passage, estimate) %>% 
  group_by(passage, treat) %>% 
  summarize(mean = mean(estimate)) %>% 
  pivot_wider(names_from = treat, values_from = mean) %>% 
  mutate(deviance = p - n)

deviance$se <- deviance_se

dev_fit <- rma(deviance~passage, se^2, method="FE", data = deviance)
dev_fit2 <- lm(deviance~passage, data = deviance)

flux_per_psg <-
  fluxes %>% 
  group_by(passage) %>% 
  summarize(mean = mean(estimate), se = se(estimate))

dev_pred <- predict(dev_fit)
```


Methane oxidation rate increased as a response to selection in the Positive selection treatment (`r figures("deviance", display="cite")`). 
This can be seen in the increase in deviation between the Positive and Neutral selection lines over the course of the experiment (z = `r myround(dev_fit$zval[2], 2)`, p = `r myround(dev_fit$pval[2], 3)`). 
Mean CH~4~ oxidation rate for the Positive treatment increased by `r myround(dev_fit$beta[2], 3)` $\pm$ `r myround(dev_fit$se[2], 3)` per passage relative to the Neutral treatment. The mean flux across all jars in all treatments in the first passage was `r myround(flux_per_psg[1, 2], 3)` $\pm$ `r myround(flux_per_psg[1, 3], 3)` so this represents a doubling of CH~4~ consumption in every passage due to selection. 

```{r dev_fig, fig.width=4, fig.height=3, fig.cap=figures("deviance")}

ggplot(deviance, aes(x = passage, y = deviance, ymin = deviance - se, ymax = deviance + se)) + 
  geom_ribbon(aes(ymin = dev_pred$ci.lb, ymax = dev_pred$ci.ub), alpha = 0.2) +
  geom_pointrange(color = 'darkorange2') + 
  #geom_abline(intercept = 0.1 + fit$beta[1], slope = fit$beta[2], color = 'blue') +
      geom_segment(aes(x = 1, xend = 5, y = dev_fit$beta[1] + dev_fit$beta[2], yend = dev_fit$beta[1] + dev_fit$beta[2]*5), color = 'darkorange2', size = 1) +
  labs(x = "Passage Number", y = "Deviation from Control (P - N)")+
  theme_bw()
```



## Ecosystem heritability

```{r herit_model}
heritability <- read_tsv('../Output/heritability.tsv')
her_fit <- lm(offspring ~ parental, data = heritability)
her_sum <- tidy(summary(her_fit))
her_aov <- tidy(anova(her_fit))


```

```{r inbreeding_coef}
N = 12 # population size
T = 5  # number of passages (generations)
F = 1 - (1 - 1 / (2 * N)) ^ T
```

Methane oxidation rate was heritable between consecutive passages (`r figures("herit", display="cite")`). Specifically, the narrow-sense heritability, calculated as the slope of the mid-parent and mid-offspring, was $h^2$ = `r myround(her_sum[2, 2], 2)` $\pm$ `r myround(her_sum[2, 3], 2)`. With an inbreeding coefficient of `r myround(F, 2)`, 


```{r herit_fig, fig.width=5, fig.height=4, fig.cap=figures("herit")}
ggplot(heritability, aes(x = parental, y = offspring)) +
  geom_point(aes(color = treat)) +
  stat_smooth(method = 'lm', formula = y ~ x, color = "black") + 
  labs(x = "Parental Mean Flux (-k)", y = "Offspring Mean Flux (-k)") +
    scale_color_manual(name = "Selection\nTreatment", labels = c('Neutral', 'Positive'), values = c("gray60", "darkorange2")) +
  theme_bw()

```




```{r}
R = dev_fit$beta[2]
h2 = pull(her_sum[2, 2])
S = R/h2
```


# Discussion

## Future Directions

This study focused on the practical implications of using AES as a tool to deduce how microbes map onto functions at the whole-ecosystem level. If one were interested in the evolutionary dynamics underlying this process, one could design a study to distinguish between individual selection on microbial strains and group selection or multilevel selection. That is, selection that cannot be recapitulated through selection at the individual level (Williams and Lenton 2007). This could be accomplished by culturing differentially abundant taxa in the positive selection treatment and then inoculating them in isolation (or in a non-selected background community) to see if the same level of ecosystem function could be recapitulated as in the final generation of the positive selection treatment. 

Another avenue for research could be to look at correlated responses of ecosystem-level traits to selection. This would address questions such as, "Is methane oxidation rate correlated with other ecosystem processes due to pleiotropy or linkage disequilibrium?" At the community level, pleiotropy might be defined as "the production by a single taxon of two or more unrelated effects." Further, linkage disequilbrium would represent correlated traits due to taxa who tend to co-associate due to shared environmental preferences or coupled metabolic pathways. 

These experiments would further help lus understand the nature of the "map" between community structure and ecosystem function. This would not only reveal how best to predict ecosystem function in earth system models, but also demonstrate the mechanism underlying that relationship. This is akin to deducing the genetic control of trait variation in molecular biology.

Community epistasis is analogous to functional redundancy. 

# Acknowledgments
This project was supported by the National Science Foundation Graduate Research Fellowship Program (grant no. DGE 1255832) and the ARCS Foundation Florence and Mike Nudelman Scholarship.

# Competing Interests
We declare we have no competing interests.

# References

