---
title: Artificial ecosystem selection to deduce the mapping between microbial community structure and ecosystem function
author: "Andrew H. Morris and Brendan J. M. Bohannan"
output:
  word_document:
    reference_docx: reference.docx
editor_options:
  chunk_output_type: inline
---

```{r options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
```

```{r initialize_packages}
library(tidyverse)
library(captioner)
library(broom)
library(broman)
library(morris)
#library(metafor)
```
```{r load_data}
heritability <- read_tsv('../Output/heritability.tsv')
fluxes <- read_tsv('../Output/fluxes.tsv')

load(file = '../Output/response_model.Rdata')
load(file = '../Output/herit_model.Rdata')

source('../R/functions.R')

```


```{r figure_captions, include=FALSE}

figures <- captioner()
figures("response", "Deviance of CH~4~ oxidation rate for the Positive selection treatment relative to the Neutral selection treatment. Deviance is calculated as the mean of P within each passage minus the mean of N within each passage. Errorbars are the standard error of the difference between P and N for each passage. The best fit line is an ordinary least squares fit taking into account the standard errors of the deviations. The gray ribbon is the 95% confidence interval for the regression.")
figures("herit", "Regression of offspring CH~4~ flux on mid-parent CH~4~ flux. Mid-parent is the mean of jars selected to inoculate the next passage. The mid-offsprings are the mean of all twelve jars produced in one passage. The best fit line is an ordinary least squares regression line and the gray ribbon is the 95% confidence interval for the regression coefficient.")

```

# Abstract

# Introduction 

One of the fundamental questions in Ecology is To what extent does biodiversity
drive variation in ecosystem function? Variation in abiotic conditions
regulates the rate of ecosystem processes, but it remains unclear to what
extent variation in community structure regulates the rate of ecosystem
processes. Many ecosystem functions are mediated by organisms. In particular,
microorganisms such as bacteria mediate the major biogeochemical cycles such
as the carbon and nitrogen cycles. Despite the central role of bacteria in
performing these processes, we have not found a consistent relationship between
microbial community structure and the rate of an ecosystem process. In order
to understand the importance of biodiversity to ecosystem functioning and to
better predict and manage specific ecosystem processes, we need to determine
the relationship between microbial community structure and ecosystem function.

In macroorganismal communities, there appears to be a relationship between the
diversity of a community and the resilience of ecosystem functions to
environmental perturbations (Tilman, ????... Hooper). Typically, this is the
diversity (or more specifically, the richness) of a particular functional
group, such as photosynthetic plants. Similar studies in microbial communities
rarely find a relationship between diversity and function (Graham, ). However,
these studies typically define diversity much more broadly than plant diversity
studies by looking at the phylogenetic diversity of the entire bacterial and
archaeal community. Studies that focus more narrowly on a particular ecosystem
process and a functional group involved in that process still rarely find a
significant correlation between the abundance of a functional group and the
rate of the corresponding process (Rocca, ????). For certain ecosystem
functions in certain ecosystems, these relationships can be informative. For
example, the abundance of a single methanogenic species (*G. species*) is
correlated with methane emissions in a permafrost ecosystem (cite). In
addition, bacterial and archaeal community composition is correlated with
methane emissions in forests and pastures of the Brazilian Amazon (Meyer et al.
2020). Therefore, there appears to be some indication that microbial community
structure drives variation in methane cycling in soils.

Why haven't we been more successful in identifying relationships between
microbial community structure and ecosystem functions? Microbial communities
are super diverse, particularly in soil environments, and
looking at the whole community using broad phylogenetic markers such as 16S
makes it difficult to sift through all of that diversity to find informative
markers. Looking at individual protein-coding genes manages to reduce that
diversity, but it also assumes we know what limits the rate of a process in a
particular ecosystem. Typically, we choose the gene that encodes the final step
in a pathway - for example, the mcrA gene, which encodes a subunit of the
enzyme the performs the final step in the methanogenesis pathway. However,
regulation may happen further up stream within the pathway and therefore the
abundance of the final gene wouldn't correlate with the rate of the process.
In fact, the limitation may be in substrate creation by other functional groups
unrelated to methanogens. To overcome these challenges, we chose a

To further investigate the relationship between microbial community structure
and the rate of methane cycling in soils, we want to determine which organisms
are important for regulating methane cycling. This ecological question is
analogous to the problem of genotype-phenotype mapping in quantitative genetics
(Morris 2020). Variation in an organismal trait is partially determined by
environmental conditions and partially determined by genetic variation. One goal
of quantitative genetics is to determine the proportion of the variation
attributable to genetics and to identify which alleles regulate the phenotypic
trait. These goals are commonly achieved through artificial selection
experiments and association mapping studies, such as genome-wide association
studies. To determine the mapping between microbial community structure and
ecosystem function, we applied these approaches of genotype-phenotype mapping
to whole microbial communities. In this case, the "alleles" are different
microbial taxa and the "phenotype" is the rate of ecosystem function at the
whole-ecosystem level.

To identify communities of bacteria and archaea that collectively perform a
high rate of methane oxidation, we performed an artificial ecosystem selection
experiment  (Swenson et al. 2000).  We generated twenty-four soil ecosystems in the lab using a sterilized
potting mix as the substrate. We then inoculated these ecosystems with a small
amount of living soil to generate variation in community structure across the
twenty-four ecosystems. These ecosystems were then assigned to one of two
selection treatments: neutral selection or positive selection. Each ecosystem
was maintained at 1000 ppm CH~4~ over several weeks to allow for colonization and
growth of the microbial community and to enrich for methane oxidizers. We then
determined the methane oxidation rate for each ecosystem. To impose selection,
we selected three jars from the positive treatment with the highest rate of
methane oxidation, homogenized them, and used that soil to inoculate the next
set of jars. For the neutral treatment, we selected three jars at random to
inoculate the next set of jars. This process continued for five passages. At
the end of the experiment, we extracted DNA from soils in the second and fifth
passage for both the neutral and positive treatments. By comparing the response
to selection within the community between the two treatments, we can identify
microbial species that contribute to variation in ecosystem methane emissions.

This approach is powerful because by using a common soil substrate and
maintaining a constant headspace concentration of methane, we can eliminate
much of the environmental variation that would be present in an observational
study. In addition, by applying selection on methane oxidation rate
in the positive treatment, we can enrich for taxa involved in methane cycling
making it easier to perform association mapping between microbial taxa and
ecosystem methane emissions. In this paper, we address whether there is a
response to selection on methane oxidation rate at the whole ecosystem level. We
then quantify the amount of variation in methane oxidation attributable to
variation in community composition within our laboratory environment. Finally,
we identify microbial markers of soil methane cycling using association mapping
in order to deduce the mapping between microbial community structure and
ecosystem function.

(Goodnight et al. 1997, Williams and Lenton
2007)

# Materials & Methods

## Selection experiment
We performed an artificial ecosystem selection experiment similar to Swenson et
al. (2000) by passaging replicate soil microbial communities. The trait we
selected on was CH~4~ oxidation rate. Our experiment had two selection lines
with twelve jars each for a total of twenty-four jars per passage. One line was
a positive selection line where the two or three jars with the highest CH~4~
oxidation rate were chosen to inoculate the next set of positive jars. The
other line was a neutral selection line where an equal number of jars as the
positive line were chosen at random to inoculate the next set of neutral jars.
The number of jars chosen was based on the distribution of fluxes among the
positive jars, i.e., we chose the top three jars unless only two jars had
considerably greater oxidation rates based on visual inspection of histograms.
The experiment was carried out over five passages.

The initial soil microbial community was sampled from the top 10 cm of an
upland mineral soil under a deciduous forest ecosystem near the University of
Oregon campus in Eugene, OR, USA. Incubations were performed in 500 mL mason
jars (Ball Corporation, Broomfield, Colorado, USA) with rubber septa installed
in the lids. Each jar was sterilized with 70% ethanol to which was added 45 g
of autoclaved potting mix, 5 g of living soil, and 3.5 mL of sterile deionized
water to bring them to <mark> 60%??? of field capacity</mark>. The jars were
then homogenized, capped, and injected with 4.3 mL of 99% CH~4~ to bring the
headspace concentration to approximately 1000 ppm CH~4~. To create the two
treatment groups, twenty-four jars were created in an identical manner and then
randomly assigned to either the positive or neutral selection line. Jars were
flushed and respiked twice per week to maintain elevated CH~4~ concentrations
and were incubated at ambient temperature for approximately three weeks.
Methane oxidation rates were determined at the end of the incubation period.
These values were then used to determine the jars selected to inoculate the
next set of jars.

## Methane oxidation rate measurements 

Methane oxidation rates were determined after flushing and spiking jars to
approximately 1000 ppm CH~4~. Headspace samples of 1 mL were collected from
each jar immediately after spiking and then at time points 3, 6, 24, and 48
hours for a 5-point curve. Samples were immediately injected into a
<mark>???</mark> gas chromatograph fitted with an electron capture detector
(company, city, state, USA) to determine the headspace CH~4~ concentration.
Fluxes were calculated from a first-order exponential decay function as decay
constant `k` with units day^-1^. Oxidation rates are presented as the additive
inverse of `k` (i.e., `-k`) so that a more positive value represents a faster
oxidation rate. The jars selected for the positive treatment in passage 2 had
the lowest methane oxidation rate of the twelve jars due to a calculation error
in the methane oxidation rate. All other passages correctly used the top three jars.

## Soil DNA extraction and sequencing

A subsample of soil from the starting inoculum and from every jar in passages 2
and 5 was collected and stored at -80$^\circ$C for later DNA extraction.
Soil DNA was extracted from a 0.25 g subsample of soil. Negative controls were extracted
from autoclaved potting mix and DNase-free water. Extractions were performed
using the DNeasy PowerSoil kit (QIAGEN, Hilden, Germany) and quantified using
Qubit (company, city, state). In order to estimate the diversity and relative
abundance of the bacterial and archaeal taxa in our soil ecosystems, we
sequenced the V3-V4 region of the 16S rRNA gene using the Earth Microbiome
Project protocol and primers<mark>???</mark>. Sequencing was performed on the
Illumina NovaSeq 6000 with paired-end 150 bp reads.

## Bioinformatics

Bioinformatic analyses were conducted using the `QIIME2`
bioinformatics platform (Bolyen et al. 2019). Raw sequencing reads were
demultiplexed using ??? (citation), denoised using `DADA2` (citation).
Taxonomic assignment for the 16S reads was performed using ??? based on the
Silva database. All subsequent tatistical analyses were performed in the `R`
statistical programming environment (R Core Team 2020). Richness estimates were
made using the `breakaway` package and diversity indices using the `DivNet`
package (Willis et al. ???).

## Statistical Analysis

### Response to selection

To test whether there was a significant change in CH~4~ oxidation rate (k) as a
response to selection, we tested a difference in slopes between the positive
and neutral selection lines. The CH~4~ oxidation rates were strongly right
skewed with most values close to zero and few large, positive values. This
resulted in residuals that did not meet the assumptions of constant variance
and normal distribution. Therefore, methane oxidation rates were log~10~
transformed to better meet the assumptions of a linear model and to make
figures easier to interpret. First, we tested the effect of treatment by
fitting two nested models with and without treatment using `lm` in `R`. The
model formulation for the full model was `flux ~ passage * treatment` and the
model formulation for the reduced model was `flux ~ passage`. We compared these
models using the likelihood ratio test with the `anova` function in `R`. We
then fit the full model to determine the slope of the positive line, which
represents the change in methane oxidation rate per passage as a response to
selection.

### Ecosystem heritability

To estimate the proportion of variance in ecosystem function due to additive
genetic variance of the microbial communtiy, we applied the Breeder's equation
to the results of our selection experiment. the heritability of
CH~4~ oxidation rate as an ecosystem property, we calculated narrow-sense
heritability ($h^2$) as the regression of offspring on mid-parent (Goodnight
2000, Falconer 1997). The mid-parent was the mean for all three selected jars
and the offspring was the mean for all jars produced by those parents.

In addition, we calculated the inbreeding coefficient, $F$, as:

$$
F = 1 - \left(1 - \frac{1}{2N} \right) ^{T}
$$

where $N$ is the population size and $T$ is the number of generations (Goodnight 2000 or Falconer 1997).

### Taxa differentially abundant in the positive selection treatment

To identify taxa that responded to selection on CH~4~ oxidation rate, we tested
differential abundance using a beta-binomial model (Martin et al. 2020). We
compared taxa in passage 5 between the positive and neutral treatment to
identify taxa that 

Possible tests: 

- Random forest or gradient boosting to predict Positive vs. Neutral. 

# Results

## Response to selection

We observed a response to selection on soil methane oxidation rate at the whole
ecosystem level (`r figures("response", display="cite")`).
At the start of the experiment, the Positive
treatment had a mean methane oxidation rate that was
`r percent_less(initial = tidy(response_model)[[1, 2]], final = tidy(response_model)[[3, 2]] + tidy(response_model)[[1, 2]])`%
lower than the
Neutral treatment (SE = 
`r myround(tidy(response_model)[[3, 3]], 2)`, t = 
`r myround(tidy(response_model)[[3, 4]], 2)`, p = 
`r myround(tidy(response_model)[[3, 5]], 2)`). There was no change in methane oxidation rate of the Neutral
treatment over the five passages (slope = 
`r myround(tidy(response_model)[[2, 2]], 2)`, SE = 
`r myround(tidy(response_model)[[2, 3]], 2)`, t = 
`r myround(tidy(response_model)[[2, 4]], 2)`, p = 
`r myround(tidy(response_model)[[2, 5]], 2)`). By contrast, the Positive treatment had a
`r round(per_slope)`% increase in methane
oxidation rate per passage (slope = 
`r myround(tidy(response_model)[[4, 2]], 2)`, SE = 
`r myround(tidy(response_model)[[4, 3]], 2)`, t =
`r myround(tidy(response_model)[[4, 4]], 2)`, p = 
`r myround(tidy(response_model)[[4, 5]], 2)`).

```{r response_figure, fig.width=4, fig.height=3, fig.cap=figures("response")}
ratio_log10 <- calc_plot_ratios(fluxes$passage, fluxes$log_ch4)

plot_flux(fluxes, passage, log_ch4, ratio_log10, log10 = TRUE)

```

## Ecosystem heritability

Here, I fit two models. The first is simply offspring flux on selected parent
flux, ignoring treatment. There is no correlation. Next, I added treatment and the interaction between treatment and parental
flux. Here, there is a significant correlation between parent and offspring for
both the Neutral treatment `r print_model(herit_model, 2)` and the Positive
treatment `r print_model(herit_model, 4)`. Notably, the sign of the effect is
reversed: negative for the Neutral treatment and positive for the Positive
treatment.

The negative heritability of methane oxidation rate in the Neutral selection
treatment may result from selection on alternative characters in those
populations. The strong, positive heritability in the Positive treatment ... .
It seems strange that the Methane oxidation rate was heritable between
consecutive passages (`r figures("herit", display="cite")`). Specifically, the
narrow-sense heritability, calculated as the slope of the mid-parent and
mid-offspring, was $h^2$ =  $\pm$  (t = , p = ). With an inbreeding coefficient of , 


```{r herit_fig, fig.width=5, fig.height=3, fig.cap=figures("herit")}
plot_herit(heritability)
```

# Discussion

## Future Directions

This study focused on the practical implications of using AES as a tool to deduce how microbes map onto functions at the whole-ecosystem level. If one were interested in the evolutionary dynamics underlying this process, one could design a study to distinguish between individual selection on microbial strains and group selection or multilevel selection. That is, selection that cannot be recapitulated through selection at the individual level (Williams and Lenton 2007). This could be accomplished by culturing differentially abundant taxa in the positive selection treatment and then inoculating them in isolation (or in a non-selected background community) to see if the same level of ecosystem function could be recapitulated as in the final generation of the positive selection treatment. 

Another avenue for research could be to look at correlated responses of ecosystem-level traits to selection. This would address questions such as, "Is methane oxidation rate correlated with other ecosystem processes due to pleiotropy or linkage disequilibrium?" At the community level, pleiotropy might be defined as "the production by a single taxon of two or more unrelated effects." Further, linkage disequilbrium would represent correlated traits due to taxa who tend to co-associate due to shared environmental preferences or coupled metabolic pathways. 

These experiments would further help lus understand the nature of the "map" between community structure and ecosystem function. This would not only reveal how best to predict ecosystem function in earth system models, but also demonstrate the mechanism underlying that relationship. This is akin to deducing the genetic control of trait variation in molecular biology.

Community epistasis is analogous to functional redundancy. 

# Acknowledgments
This project was supported by the National Science Foundation Graduate Research Fellowship Program (grant no. DGE 1255832) and the ARCS Foundation Florence and Mike Nudelman Scholarship.

# Competing Interests
We declare we have no competing interests.

# References

