---
title: Response of soil microbiome composition to selection on methane oxidation rate
author: "Andrew H. Morris^1^* and Brendan J. M. Bohannan^1^"
output:
  word_document:
    reference_docx: reference.docx
    number_sections: true
bibliography: bibliography.bib
csl: soil-biology-and-biochemistry.csl
---

^1^ Institute of Ecology and Evolution, University of Oregon, Eugene, OR, USA  
\* Correspondence: Andrew H. Morris, amorris3\@uoregon.edu  
Competing Interests: We declare we have no competing interests.

```{r options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
knitr::opts_knit$set(root.dir=normalizePath('../'))

```

```{r initialize-packages}
library(tidyverse)
library(broom)
library(broman)
library(morris)
library(ggpubr)
library(knitr)
library(corncob)
library(captioner)
library(knitr)
library(ggtext)
library(patchwork)
figures <- captioner()
tables <- captioner(prefix = "Table")

source('R/functions.R')
```

```{r load-data}
#heritability <- read_tsv(paste0(der_dir, 'heritability.tsv'))
#fluxes <- read_tsv(paste0(der_dir, 'fluxes.tsv'))

load(file = paste0(der_dir, 'decontam.RData'))

load(file = paste0(der_dir, 'models.RData'))

```

# Abstract

Microbiomes mediate important ecosystem functions, yet it has proven difficult
to determine the relationship between microbiome composition and the rate of
ecosystem functions. This challenge remains because it is difficult to
manipulate microbiome composition directly, we often cannot know _a priori_ which
microbiome members influence the rate of an ecosystem function, and
microbiomes can covary strongly with other drivers of ecosystem function, such
as the environment. To address these challenges, we imposed artificial
selection on whole soil ecosystems over multiple generations to select for microbial communities
with a high rate of CH~4~ oxidation. This approach
is potentially powerful because it is biologically "agnostic" in 
that it makes few assumptions about which taxa are important to function, and
repeated passaging with fresh substrate weakens the covariance between microbes
and the environment. As a response to selection, we observed a
 `r myround(per_slope, 1)`\% increase in
CH~4~ oxidation rate per passage relative to a control that experienced
random selection. We
estimated that `r myround(tidy(rh_model)[[2, 2]]*100, 1)`\% of the variation in
CH~4~ oxidation
rate in these soils can be attributed to microbiome variation (though this was not significant). We also found
that selection did not enrich for known CH~4~ oxidizers; instead, `r length(da_fams)` families not
known to oxidize CH~4~, including *Fimbriimonadaceae*, *Cytophagaceae*, and
*Diplorickettsiaceae*, were enriched by selection. This result is in contrast to the typical assumption that
the rate of an ecosystem function is limited by the final step in the associated
microbial pathway. Our study demonstrates that variation in microbiome
composition can contribute to variation in the rate of ecosystem function 
independent of the environment and that this may not always be limited by the final step
in a pathway. This suggests
that manipulating microbiome composition directly without altering the environment 
could be a viable strategy for managing
ecosystem functions.

# Introduction

Microbiomes mediate a variety of important ecosystem functions relevant to human
health, agriculture, and global change. As a result, there is great interest in
understanding how to manipulate the microbiome to achieve desirable outcomes
within these domains [@conrad1996; @schimel1998; @crowther2019]. However, for
microbiome manipulations to be successful, variation in the microbiome must
contribute directly to variation in the magnitude of the function of interest
independent of other factors. Many studies have attempted to document such a
relationship
[@cavigelli2000; @balser2005; @waldrop2006; @strickland2009; @wagg2014; @wagg2019].
However, it is difficult to isolate the direct effect of variation in microbiome
composition from other drivers of variation in ecosystem function, such as the
indirect effect of the environment on function through microbiome assembly.
Here, we overcome these limitations by using a selection approach to estimate
the degree to which an ecosystem function varies with microbiome composition.

Altering ecosystem functions via microbiome manipulations requires that the
microbiome contributes to variation in ecosystem function independent of other
drivers of ecosystem variation, such as variation in environmental conditions.
This is because the drivers of variation in ecosystem functions can interact in
complicated ways (`r figures("concept", display = "cite")`). Variation in the
microbiome can contribute directly to variation in ecosystem function, for
example, if a microbial population is replaced by one with a greater enzyme
efficiency. In addition, environmental conditions can contribute indirectly to
ecosystem function via covariance with the microbiome, for example, by providing
conditions that select for microbial groups that in turn alter the rates of
ecosystem functions. In this scenario, identifying the change in microbial
community composition without adequately controlling for the environmental
conditions would incorrectly attribute the change in ecosystem function to the
microbiome when it is ultimately an indirect effect of the environment.
Determining the independent contribution of microbiome variation to ecosystem
function is crucial because if microbiome composition is driven primarily by
environmental conditions, then introducing a desirable taxon through microbiome
manipulation without altering the environment will likely be unsuccessful at
shifting the targeted ecosystem function.

There have been two general categories of approaches that investigators have
used to estimate the degree to which an ecosystem function varies with
microbiome composition: comparative and manipulative. Comparative studies sample
natural variation in an ecosystem function across different habitats and
simultaneously measure variation in community composition.  Investigators can
then correlate ecosystem function with aspects of community composition while
attempting to control for environmental variation. These approaches have
documented important relationships between microbiomes and ecosystem functions.
For example, a meta-analysis of these studies observed a small but significant
contribution of the microbiome to variation in ecosystem function after
controlling for environmental variation [@graham2016]. In addition, studies
focusing on the correlation between the rate of an ecosystem function and the
abundance of an associated marker gene (i.e., a gene that codes for a protein
assumed to be involved in the ecosystem function) sometimes observe a
significant correlation, though this relationship is rare and contingent upon
both the function and the ecosystem sampled [@rocca2015]. However, comparative
studies come with unique challenges and limitations. One issue is that
microbiome attributes tend to covary with the abiotic conditions within an
environment, and it is difficult to control for these abiotic variables in order
to identify the unique contribution of the microbiome to ecosystem function. In
addition, it is difficult to know _a priori_ which environmental variables or
community attributes to measure. Finally, while these approaches can establish a
potential magnitude and direction for these relationships, it is often difficult
to identify the
taxa or genes that explain the connection between composition and function.

The other broad category of approaches used to address this question are
manipulative approaches. Manipulative experiments try to alter microbial
community composition and observe the effect on function. For example,
reciprocal transplant and common garden experiments have shown that microbiomes
originating from different ecosystems inoculated into the same substrate or
introduced into a common environment display distinct functional rates 
[@cavigelli2000; @balser2005; @waldrop2006; @strickland2009]. In addition,
manipulating diversity by filtering communities by cell size or through dilution
has been shown to alter the rate of ecosystem functions 
[@wagg2014; @wagg2019; @griffiths2001]. However, manipulating the microbiome
directly is challenging, and manipulative approaches often confound
community composition with other factors. For example, reciprocal transplant and
common garden experiments can confound community composition with the abiotic
conditions introduced with the inoculum, while manipulating composition through
dilution may confound composition with biomass [@reed2007]. 

In this study, we sought to build on the observations of comparative and
manipulative studies by applying a different approach to the question of whether
microbiome variation contributes to variation in the rate of an ecosystem
function. We used artificial ecosystem selection to select for microbiomes that
performed a greater rate of ecosystem function [@swenson2000; @blouin2015;
@panke-buisse2015]. We then tested whether variation in the microbiome
contributed to variation in the rate of ecosystem function and identified
microbiome attributes that might explain this relationship. There are several
potential advantages to this approach for documenting the direct contribution of
the microbiome to variation in ecosystem function and for investigating the
mechanisms underlying those relationships. Through repeated passaging of microbiomes
in a common environment, we can weaken the covariance between microbes and the
environment by repeatedly diluting the influence of variation in abiotic
conditions. In addition, our approach eliminates the need to generate microbiome
variation through methods that are confounded with biomass or cell size. Lastly,
by comparing our artificially selected community to a control community
resulting from random selection, we can both control for changes in the
environment over time and identify genes or taxa that are associated with the
ecosystem function under selection. 

We applied artificial ecosystem selection to soil microbiomes by selecting on
soil methane (CH~4~) oxidation rate. We chose this function because CH~4~ is a
globally important greenhouse gas and CH~4~ oxidation by soil bacteria is the
primary biological sink for atmospheric CH~4~ [@kirschke2013]. In addition,
there is evidence that soil CH~4~ oxidation rate may vary with microbiome
composition based on comparative studies in a variety of arctic and tropical
ecosystems [@meyer2020; @meyer2020a; @freitag2009; @freitag2010] as well as
studies that manipulate methanotroph richness [@schnyder2018]. Finally,
methanotrophy is one of the most deeply conserved microbial physiologies and is
represented in a narrow range of taxa, which suggests that the taxonomic
composition of the microbiome is more likely to be associated with the rate of
CH~4~ oxidation than other broader or more shallowly conserved functions
[@schimel1998; @martiny2013]. 

In this study, we used artificial ecosystem selection on CH~4~ oxidation rate
to address the following questions: Does variation in the relative abundance of
microbial taxa contribute to variation in soil CH~4~ oxidation rate
independent of the environment in our system? Which attributes of the microbiome
are associated with variation in CH~4~ oxidation rate, and do these attributes
match our assumptions about the factors that regulate CH~4~ oxidation rate in
nature?

# Materials and methods

## Experimental design

We performed an artificial ecosystem selection experiment (*sensu* @swenson2000)
by passaging replicate soil microbiomes. The trait we selected on was soil CH~4~
oxidation rate.  Soil microcosms were incubated at room temperature in sealed
500 mL glass jars with a rubber septum for gas sampling. Each jar was sterilized
with 70\% ethanol and was composed of 45 g of autoclaved artificial potting mix,
5 g of living soil inoculum, and 3.5 mL of sterile deionized water to bring the
soil to 60\% of field capacity. The potting mix consisted of bark fines, peat
moss, pumice, sand, composted manure, and biochar (Lane Potting Mix, Lane Forest
Products, Eugene, OR). The initial soil microbiome inoculum was sampled from the
top 10 cm of an upland mineral soil under a deciduous forest ecosystem near the
University of Oregon campus in Eugene, OR, USA. Each jar
was capped and injected with 0.43 mL of 99\% CH~4~, which produced a mean
headspace concentration of 763.9 ppm (SD = 183.1). Twice per week, jars were
flushed in a biosafety cabinet (to avoid contamination) and respiked with CH~4~ 
to maintain aerobic conditions and elevated CH~4~ concentrations.

For the selection experiment, we created two lines of soil microcosms with 12
jars each: a control line with random selection and an experimental line with
directional selection for greater soil CH~4~ oxidation rate. The selection line
underwent positive selection where the two or three jars
with the highest CH~4~ oxidation rate were homogenized to inoculate the next set
of jars. The control line underwent random selection where an equal number
of jars as the selection line were chosen at random to inoculate the next set of
jars. The number of jars chosen was based on the distribution of fluxes among
the positive jars: three jars were chosen in all generations except for passage 3
where two jars were selected. The experiment was carried out over five passages
with an average
incubation time per generation of four weeks. 
Methane oxidation rates were determined at the end of
the incubation period and selection was performed. For each treatment, the
selected jars were homogenized and 5 g of the homogenized soil was used as the
living soil inoculum for the next generation. The next set
of jars were created in an identical manner to the first generation with fresh
autoclaved potting mix and the same moisture and CH~4~ content.

## Methane oxidation rate

Methane oxidation rates were determined after flushing and spiking jars with 
0.43 mL of 99\% CH~4~. Headspace samples of 1 mL were collected from
each jar immediately after spiking and then at time points 3, 6, 24, and 48
hours for a 5-point curve. Samples were immediately injected into a SRI model
8610C gas chromatograph equipped with a flame ionization detector (SRI
Instruments, Torrance, CA, USA) to determine the headspace CH~4~ concentration.
We applied a first-order exponential decay function to determine the rate
constant (k, units = d^−1^; i.e., dCH~4~/dt = k[CH~4~]) of the exponential
decrease in CH~4~. Oxidation rates are presented as the additive inverse of
$k$ (i.e., $-k$) so that a more positive value represents a greater
oxidation rate. The jars selected to inoculate passage three for the positive 
selection treatment had
the lowest CH~4~ oxidation rate of the twelve jars due to a calculation error
in the rate constant.

## Soil DNA extraction and sequencing

A subsample of soil from the starting inoculum and from every jar in passages 2
and 5 was collected and stored at $-80^\circ$C. Soil DNA was extracted from
0.25 g soil. Negative controls were extracted from autoclaved potting mix and
DNase-free water. Extractions were performed using the DNeasy PowerSoil kit
(QIAGEN, Düsseldorf, Germany) and quantified using Qubit dsDNA HS Assay Kit
(Thermo Fisher Scientific, Inc., Waltham, MA, USA). To estimate the diversity
and relative abundance of the bacterial and archaeal taxa in our soil
ecosystems, we sequenced the V4 region of the 16S rRNA gene using the 515F -
806R primer combination [@caporaso2011]. PCR mixtures were: 10 $\mu$l NEBNext
Q5 Hot Start HiFi PCR master mix, 9.2 $\mu$l primer mixture (1.09 $\mu$M
concentration), and 0.8 $\mu$l of DNA template. Reaction conditions were:
$98^\circ$C for 30 s (initialization); 35 cycles of $98^\circ$C for 10 s
(denaturation), $61^\circ$C for 20 s (annealing), and $72^\circ$C for 20 s
(extension); and $72^\circ$C for 2 m (final extension). Reactions were
performed in triplicate and then combined.  Amplicons were purified twice using
0.8x ratio Mag-Bind RxnPure Plus isolation beads (Omega Bio-Tek, Norcross, GA,
USA). Sequencing libraries were prepared using a dual-indexing approach
[@fadrosh2014; @kozich2013]. Amplicon concentrations were quantified using
Qubit and multiplexed at equimolar concentration. Sequencing was performed at
the University of Oregon Genomics Core Facility on the Illumina NovaSeq 6000
with paired-end 150 bp reads (Illumina, Inc., San Diego, CA, USA).

## Bioinformatics

Bioinformatics processing was performed in 'R' [@rcoreteam2018].  Demultiplexed
sequencing reads were denoised using 'DADA2' to generate a table of amplicon
sequence variants (ASVs) [@callahan2016]. Taxonomic assignment was performed
using the Ribosomal Database Project naive Bayesian classifier [@wang2007]. The
presence of contaminants was evaluated using both the prevalence and frequency
methods from 'DECONTAM' by comparing samples to extraction controls of 
water [@davis2018]. Decontam identified 
`r length(which(contamdf$contaminant))` potential contaminants based on
prevalence and frequency. Visual inspection of abundance-concentration plots
indicated that `r length(contaminants)` of these were likely contaminants and
these ASVs were removed. Amplicon sequence variants that were assigned
chloroplast or mitochondria taxonomy were removed prior to analysis.

## Statistical analysis

### Response to selection

Statistical analyses were performed in 'R' [@rcoreteam2018]. To determine
whether there was a significant change in CH~4~ oxidation rate as a response to
selection, we tested a difference in slopes between the selection and control
lines. Residuals did not meet the assumptions of constant variance
and normal distribution. Therefore, CH~4~ oxidation rates were log~10~
transformed prior to analysis. Following transformation, these assumptions
were met. First, we tested if there was a difference of
slopes between the selection line and the control based on the interaction
between passage and treatment. To test the interaction, we fit two nested models
with and without the interaction term and compared them using an F-test with the
'anova' function. We then present the slopes for each treatment, which
represented the change in CH~4~ oxidation rate per passage as a response to
selection.

### Microbiability

We estimated the proportion of variation in CH~4~ oxidation rate due to
variation in the microbiome by comparing the response to selection to the 
imposed selection differential [@falconer1996].
This estimate is analogous
to estimates of "microbiability" from the animal breeding literature, which
quantifies the
variation in a host trait that is due to microbiome variation [@he2022].

For this purpose, the response to selection was calculated as the divergence between the selection treatment and the control. Divergence was the difference between the mean CH~4~ 
oxidation rate of the positive
treatment and the mean CH~4~ oxidation rate of the control in each
passage. In addition, the selection differential was calculated as the difference between
the mean of the three selected jars and the mean of all twelve jars in a passage. The slope of the divergence over the cumulative selection differential
provides an estimate of realized microbiability (h^2^ &pm; SE).
We regressed cumulative
divergence on cumulative selection differential using the 'lm' function. We report
the slope as percent change by back-calculating the percent change from the
log-transformed data into the original units using the formula $(10^\beta - 1) * 100$
where $\beta$ is the slope.

### Alpha and beta diversity

Richness was estimated using the method from @hurlbert1971 with a subsample
size of `r min_seqs` calculated via the 'rarefy' function in 'vegan' [@oksanen2019]. We
tested a difference in richness by both passage and treatment with a
Kruskal-Wallace test followed by a pairwise Wilcoxon test. Next, we estimated
beta-diversity as the Bray-Curtis dissimilarity by averaging 100 random subsets
with a subsample size of `r min_seqs` using the 'avgdist' function in 'vegan'
[@bray1957; @oksanen2019].  We tested a difference in centroid and dispersion
of beta diversity by passage and treatment using a permutational analysis of
variance (PERMANOVA) with 999 permutations using the 'adonis2' function from
'vegan' and tested a difference of group dispersions using 'betadisper' and
'anova' with 999 permutations [@mcardle2001; @oksanen2019].  Lastly, we tested
the correlation between CH~4~ oxidation rate and Bray-Curtis dissimilarity in
Passage 5 with a distance-based redundancy analysis (dbRDA) using the 'dbrda'
function in 'vegan' and estimated the p-value using a permutation F-test with
999 permutations [@mcardle2001; @oksanen2019]

### Differential abundance

To identify taxa that responded to selection on CH~4~ oxidation rate, we tested
differential abundance between the two treatments in passage 5. We first
grouped ASVs at the family level. We chose this level of agglomeration
because CH~4~ oxidation
is a relatively deeply conserved function [@martiny2013] and is restricted to
a handful of bacterial and archaeal families [@kang2019]. Therefore, we are most
likely to detect an enrichment of methanotrophs at this taxonomic scale.
Any ASVs that lacked a family-level
taxonomic assignment were grouped at a higher taxonomic level. We then subset the
samples in Passage 5 and removed all families with a prevalence of less than
10\% in either treatment. We used three methods for testing differential
abundance: ANCOM-II, ALDEx2, and CORNCOB 
[@kaul2017; @lin2020; @fernandes2014; @martin2020]. We then identified the
consensus taxa that were significant with all three tests and plotted their
relative abundances. For ANCOM-II, we used the 'ancom' function in the
'ANCOM-BC' package with a cutoff of W = 0.7 [@kaul2017; @lin2020]. For ALDEx2,
we used the 'aldex' function in the 'ALDEx2' package with Welch's t-test and we
used an effect size of 1 as our significance threshold [@fernandes2014].
Finally, we used CORNCOB with the 'differentialTest' function in the 'corncob'
package with the Wald test and without bootstrapping [@martin2020]. Lastly, to
test differentially abundant methanotrophs, we subset all ASVs within
methanotrophic families and tested their differential abundance aggregated at
the family and genus level using 'corncob'. For each test, p-values were adjusted
for multiple testing by controlling the false discovery rate using the 
Banjamini-Hochberg procedure [@benjamini1995].

# Results

## Response to selection on methane oxidation rate

We observed a response to artificial selection on whole-ecosystem soil CH~4~ oxidation
rate (`r figures("response", display = "cite")`; difference of slopes:
F~`r compare_response_mods[[2,4]]`,`r compare_response_mods[[2,2]]`~ = 
`r myround(compare_response_mods[[2,6]], 2)`,
p = `r myround(compare_response_mods[[2,7]], 2)`). At the start of the experiment,
the positive selection treatment had a mean CH~4~ oxidation rate that was
`r percent_less(initial = tidy(response_model)[[1, 2]], final = tidy(response_model)[[3, 2]] + tidy(response_model)[[1, 2]])`\%
lower than the control
`r print_lm(response_model, 3, 'difference of y-intercepts')`. 
There was no change in CH~4~ oxidation rate in the control over the
five passages `r print_lm(response_model, 2)`. By contrast, the selection
treatment had a `r myround(per_slope, 1)`\% increase in CH~4~ oxidation rate per
passage `r print_lm(response_model, 4)`.

To estimate the proportion of variation in CH~4~ oxidation rate due to
variation in microbiome composition--i.e., microbiability (as described in the Methods; [@he2022])--we regressed divergence between the
positive selection treatment and the control against the cumulative selection
differential. The microbiability was `r myround(tidy(rh_model)[[2, 2]], 2)` &pm;
`r myround(tidy(rh_model)[[2, 3]], 2)`, though this was not significant (F~`r glance(rh_model)$df`,`r glance(rh_model)$df.residual`~ = 
`r myround(glance(rh_model)$statistic, 2)`, 
p = `r myround(glance(rh_model)$p.value, 2)`).

## Taxonomic richness

Median ASV richness decreased from `r myround(rare_median[1, 2], 1)` (`r myround(rare_median[1, 3], 1)`) in passage 2 to `r myround(rare_median[2, 2], 1)` (`r myround(rare_median[2, 3], 1)`) in passage 5 (Kruskal-Wallace test: $\chi^2$ = 
`r myround(kt[1,1], 1)`, df = `r kt[1,3]`, p < 0.001; pairwise Wilcoxon test: p < 0.001). However, there was no
difference in richness between the selection treatment and the control in
passage 2 or 5 (pairwise Wilcoxon test; Passage 2: p = `r myround(wt[1, 3], 2)`,
Passage 5: p = `r myround(wt[6, 3], 2)`).  In addition, there was no
correlation between richness and CH~4~ oxidation rate across the two treatments in passage 5 (Spearman's rho = `r myround(s5[1,1], 1)`, p = `r myround(s5[1,3], 1)`).

## Community dissimilarity

Bray-Curtis dissimilarity of the soil microbiome varied strongly by passage and
weakly by treatment with an interaction between passage and treatment (`r figures("beta", display = "cite")`). 
Passage explained `r myround(100*tidy_permanova[1, 4])`\% of the
variation in Bray-Curtis dissimilarity (F~`r tidy_permanova[1, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[1, 5], 1)`, p = `r myround(tidy_permanova[1, 6], 3)`), treatment explained 
`r myround(100*tidy_permanova[2, 4])`\%
of the variation (F~`r tidy_permanova[2, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[2, 5], 1)`, p = `r myround(tidy_permanova[2, 6], 3)`), and the interaction
between treatment and passage explained 
`r myround(100*tidy_permanova[3, 4])`\% of the variation (F~`r tidy_permanova[3, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[3, 5], 1)`, p = `r myround(tidy_permanova[3, 6], 3)`). There was 
no difference in dispersion between treatments or passages (F~3,44~ = 0.91, p = 0.45). 
Finally, CH~4~ oxidation rate was correlated with Bray-Curtis dissimilarity across both treatments in passage 5 and explained `r myround(dbrda_percent_explained[1], 1)`\% of the variation in Bray-Curtis dissimilarity (dbRDA: F~`r dbrda_aov[1, 1]`,`r dbrda_aov[2, 1]`~ = `r myround(dbrda_aov[1, 3], 2)`, p = `r myround(dbrda_aov[1, 4], 2)`)

## Taxa that responded to selection

To identify taxa that responded
to selection on soil CH~4~ oxidation rate, we tested the differential relative
abundance of families in the selected jars relative to the control jars within
passage 5 using three methods and then plotted the taxa identified by all three methods. We identified `r length(da_fams)`
families that were enriched or depleted in the selection treatment relative to
the control (`r figures("da", display = "cite")`). 

Overall, none of the families enriched in the selection treatment
contain known methanotrophs. Several taxa identified had a 
higher taxonomic designation that contains methanotrophs, for example, the
Gammaproteobacteria class had a large effect size.
The Gammaproteobacteria include the type I and type X methanotrophs in the families *Methylococcaceae* and *Methylothermaceae* [@stein2012].
However, the Gammaproteobacteria is among the most diverse 
groups in the Prokaryotes, so this is not strong evidence for a selection response
by methanotrophs [@garrity2005].
In addition, the *Puniceicoccaceae* is a member of the phylum Verrucomicrobia. The Verrucomicrobia is a diverse group that contain
known methanotrophs as well as ammonia-oxidizing bacteria [@freitag2003]. Other
than these two groups, none of the other taxa enriched in the selection treatment
are known to be related to methanotrophs. Two groups in the Armatimonadales were
enriched in the selection treatment including the family *Fimbriimonadaceae* and
an unclassified ASV from the order Armatimonadales [@im2012].
*Cytophagaceae* was also enriched in
the selection treatment and contains a number of mainly aerobic heterotrophs
that can digest a variety of macromolecules [@mcbride2014]. 
The remaining families include the uncultured family 0319-6G20, *Diplorickettsiaceae*, *Rhodospirillaceae*, and an unclassified Kapabacteriales.

We did not identify any methanotrophic families in the overall differential abundance analysis. 
However, we wanted to look more closely at the known methanotrophs in our dataset to be sure that they did not have an effect.
To do this, we subset the ASVs in our dataset that were in families that contained methanotrophs.
Only two families were represented: *Methylacidiphilaceae* and *Beijerinckiaceae*. 
Aggregating reads at the family level, neither family was differentially abundant between the two treatments.
However, aggregated at the genus level, a group of unclassified genera in the *Beijerinckiaceae* were depleted in the selection treatment and the genus
*Rhodoblastus*, a member of the *Beijerinckiaceae*, was enriched in the
selection treatment.  While many *Beijerinckiaceae* are methanotrophs, several
taxa in this family have lost the ability to oxidize CH~4~ and it appears
that *Rhodoblastus* species are not able to grow on CH~4~, though they can grow
on methanol [@tamas2014; @dedysh2016]. Based on this analysis,
it appears that no known methanotrophs were enriched in the positive selection treatment.


# Discussion

We used artificial ecosystem selection to estimate the contribution of variation
in microbiome composition to variation in the rate of an ecosystem function,
CH~4~ oxidation in soil, independent of environmental variation. Understanding
how and to what degree microbiome variation contributes to variation in
ecosystem function is important for many reasons. For example, successful
microbiome manipulations require that the manipulated microbiome contribute to
variation in ecosystem function independent of other drivers of ecosystem
variation (such as variation in environmental conditions). This is because the
drivers of variation in ecosystem functions can interact in complicated ways
(`r figures("concept", display = "cite")`);
for example, environmental variation can indirectly contribute by providing
conditions that select for microbial groups that in turn alter the rates of
ecosystem functions. Determining the independent contribution of microbiome
variation to ecosystem function is crucial because if microbiome composition is
driven primarily by environmental conditions, then introducing a desirable taxon
through microbiome manipulation without altering the environment will likely be
unsuccessful at shifting the targeted ecosystem function. 
The artificial selection approach is different from the comparative and
manipulative approaches used in past attempts at answering this question,
because it can control for both the direct effect of environment on function as
well as the indirect effect of environment via its impact on microbiome
assembly. 

In our study, we observed an increase in CH~4~ oxidation rate in the
selection treatment relative to the control, which demonstrates that there was a
response to selection. In addition, we observed a significant correlation 
between CH~4~ oxidation rate and overall microbiome composition. Given these observations, we
conclude that variation in the microbiome contributes to variation in the
CH~4~ oxidation rate independent of the environment. This suggests that
microbiome manipulations could be an effective approach for altering the rate of
CH~4~ oxidation in this soil, and that the artificial selection approach may
be useful in determining the potential for microbiome manipulations for other
functions in other ecosystems. 

Given that variation in the microbiome is associated with variation in the rate
of an ecosystem function in our system, we next wanted to estimate how
much variation in ecosystem function is associated with microbiome variation.
The relationship between microbiome variation and ecosystem function variation
is analogous to the concept of "heritability" [@lynch1998] used by quantitative
geneticists, or more precisely the concept of "microbiability" [@he2022] proposed
by microbiome scientists who study host-associated microbiomes. Although rarely
used in the study of environmental microbiomes, this concept could be
useful for understanding and manipulating microbially-mediated functions in a
variety of ecosystems. 

In our experiment, the microbiability was not statistically
significant and, therefore, we were not able to accurately estimate
the percentage of variation in CH~4~ oxidation rate explained by the microbiome. 
However, we did observe a significant divergence between the 
positive selection and control lines as a result of passaging the microbiome.
This suggests that the imposed
selection was sufficient to generate
variation in soil CH~4~ oxidation rate via the microbiome. Future studies with
greater replication could more precisely estimate the microbiability.
Even though we attempted to estimate the microbiability in this study, 
it is likely that the
“environmental microbiability” will be different for other ecosystem functions
in this soil and for CH~4~ oxidation in other soils. However, our experiment
provides an example of how
this relationship can be measured

We next wanted to determine which aspects of the
microbiome might explain the divergence in CH~4~ oxidation rate between the two treatments.
There are three inter-related ways that microbiomes could have responded
to selection in this experiment: gain or loss of taxa, changes in the
relative abundances of taxa, or changes within the genomes of the constituent
taxa. We surveyed microbiome variation via 16s rRNA ribotyping in our
experiment, which allowed us to deeply sample taxonomic diversity but did not
allow us to directly address whether taxa in this experiment evolved genomic changes
as a result of selection. However, if such genomic changes resulted in increased
persistence or abundance of the population with these changes, this would be
detectable. Therefore, we will focus on the first
two possibilities. 

Richness at the ASV level did not vary between the two treatments and there
were relatively few taxa gained or lost in the selection treatment and
none of these were prevalent across the 12 jars in passage 5. Therefore, the
gain or loss of species is unlikely to explain the increase in CH~4~
oxidation rate. However, we found that Bray-Curtis dissimilarity was greater
between the two treatments in Passage 5 than within each treatment and was
correlated with CH~4~ oxidation rate, which suggests that changes in the
relative abundance of taxa could explain the response to selection.

Even though we observed an increase in CH~4~ oxidation rate in the selection
treatment and a difference in composition between the two treatments, we did
not observe an increase in the relative abundance of known methanotrophs. This was
surprising given that CH~4~ consumption is not a common trait among microbes
and that it is often assumed that the rate of an ecosystem function is limited
by the final enzymatic step in the underlying metabolic pathway [@rocca2015]. In certain
ecosystems, CH~4~ production and consumption are correlated with the
abundance of methanogens and methanotrophs as estimated from marker genes
[@meyer2020; @meyer2020a]. However, our results suggest that in this system
ecosystem-scale CH~4~ oxidation rates can be altered by non-methanotrophs,
perhaps through ecological interactions with methanotrophic species, or by
unknown methanotrophs. This
suggests that simple assumptions about how microbes contribute to rate variation
in ecosystem function may not apply universally, and it demonstrates the
importance of using biologically "agnostic" approaches (that make few starting
assumptions) to linking microbial taxa to ecosystem functions [@morris2020].
Artificial ecosystem selection is an important example of such an approach.

# Conclusions

There is increasing interest in using artificial selection for understanding and
manipulating the microbiomes associated with plants and animals (a.k.a., “microbiome
breeding”; @mueller2022). Our study demonstrates that artificial ecosystem
selection can also be an important tool for exploring the relationship between
microbiome composition and ecosystem function in non-host systems. This
approach can provide unique information about the independent contribution of
microbiomes to ecosystem functions. For example, we demonstrated that passaging
of soil microbiomes can produce a response to selection on soil CH~4~ oxidation
rate. This demonstrates that microbiome variation can contribute to variation in 
ecosystem function independent of the environment.
Such information is crucial if we are to
successfully manipulate environmental microbiomes to alter ecosystem functions,
whether to improve crop productivity [@chaparro2012]  or
ameliorate the impacts of environmental change [@jansson2020].  

# Acknowledgments {.unnumbered}

This project was supported by the National Science Foundation Graduate Research
Fellowship Program (grant no. DGE 1255832) and the ARCS Foundation Florence and
Mike Nudelman Scholarship. Figure 1 was created with BioRender.com.

# Competing interests {.unnumbered}

We declare we have no competing interests.

# Data availability statement {.unnumbered}

The 16S rRNA sequencing data generated during the current study are available
in the NCBI Sequence Read Archive (SRA) under BioProject accession number
PRJNA832314, https://www.ncbi.nlm.nih.gov/sra/PRJNA832314. The metadata
generated during the current study as well as the scripts to recreate the
analysis are available on Github,
https://github.com/amorris28/artificial_ecosystem_selection.

# References {.unnumbered}

::: {#refs}
:::

# Figure Captions {.unnumbered}

`r figures("concept", "Variation in the rate of an ecosystem function is the result of at least three components: variation in the abiotic environmental conditions, variation in microbiome composition, and the covariance between microbiomes and the environment. The arrows represent causal relationships between the components. It is important to isolate the direct effect of the microbiome from the effect of the environment via covariance with the microbiome. Here, we attempt to isolate the effect of the microbiome through artificial selection on microbiome composition. For simplicity, we omitted the reverse arrows as well as the interactions, though these relationships may also exist.")`

`r figures("response", paste0("Response to selection on soil CH~4~ oxidation rate. The y-axis is CH~4~ oxidation rate as the log~10~ of the additive inverse of the first-order exponential decay constant $k$ (i.e., $-k$) with units day^-1^ so that a more positive value represents a higher CH~4~ oxidation rate. Orange points and regression line are for the positive selection treatment and gray points and regression line are for the control. There was a significant difference of slopes between the positive selection treatment and the control (F~", compare_response_mods[[2,4]], ",", compare_response_mods[[2,2]], "~ = ", myround(compare_response_mods[[2,6]], 2), ", p = ", myround(compare_response_mods[[2,7]], 2), ")."))`  

`r figures("beta", "Non-metric multidimensional scaling plot of beta diversity for all jars. Dissimilarities are rarefied Bray-Curtis dissimilarity averaged over 100 subsamples. Orange points are the positive selection treatment and gray points are the control. Circles are passage 2 and triangles are passage 5.")`

`r figures("da", "Differentially abundant family-level taxa identified by ANCOM-II, ALDEx2, and CORNCOB. Values on the x-axis are relative abundances on a log~10~ scale. Taxa in the top panel are enriched in the positive selection treatment relative to the control and taxa in the bottom panel are depleted in the positive selection treatment relative to the control. Taxa are sorted by their effect size with taxa at the top having the largest positive effect size and taxa at the bottom with the largest negative effect size.")`

