---
title: Artificial ecosystem selection reveals relationships between microbiome composition and ecosystem function
author: "Andrew H. Morris^1^* and Brendan J. M. Bohannan^1^"
output:
  word_document:
    reference_docx: reference.docx
bibliography: bibliography.bib
csl: vancouver.csl
---

^1^ Institute of Ecology and Evolution, University of Oregon, Eugene, OR, USA  
\* Correspondence: Andrew H. Morris, amorris3\@uoregon.edu  
Competing Interests: We declare we have no competing interests.

```{r options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
knitr::opts_knit$set(root.dir=normalizePath('../'))

```

```{r initialize-packages}
library(tidyverse)
library(broom)
library(broman)
library(morris)
# library(phyloseq)
library(ggpubr)
library(knitr)
library(corncob)
# library(ggthemes)
library(captioner)
library(knitr)
library(ggtext)
figures <- captioner()
tables <- captioner(prefix = "Table")

source('R/functions.R')
```

```{r load-data}
heritability <- read_tsv(paste0(der_dir, 'heritability.tsv'))
fluxes <- read_tsv(paste0(der_dir, 'fluxes.tsv'))

load(file = paste0(der_dir, 'decontam.RData'))


# CH4 models
load(file = paste0(der_dir, 'response_model.RData'))
load(file = paste0(der_dir, 'response_plot.RData'))
load(file = paste0(der_dir, 'herit_model.RData'))
load(file = paste0(der_dir, 'herit_plot.RData'))

# 16S models
load(file = paste0(der_dir, 'richness_models.RData'))
load(file = paste0(der_dir, 'phyloseq.RData'))
load(file = paste0(der_dir, 'beta_model.RData'))
load(file = paste0(der_dir, 'beta_fig.RData'))
load(file = paste0(der_dir, 'da_corncob.RData'))
load(file = paste0(der_dir, 'da_fig.RData'))
load(file = paste0(der_dir, 'dbrda.RData'))



```

# Abstract

Microbiomes mediate important ecosystem functions, yet it has proven difficult
to determine the relationship between microbiome composition and the rate of
ecosystem functions. This challenge remains because it is difficult to
manipulate microbiome composition directly, we often cannot know a priori which
members of a microbiome influence the rate of an ecosystem function, and
microbiomes can covary strongly with other drivers of ecosystem function, such
as the environment. We propose that artificial ecosystem selection can overcome
these challenges. Artificial ecosystem selection involves creating replicate
ecosystems, selecting replicates with a desired ecosystem-level trait, and
using the microbiomes from the selected ecosystems to inoculate a new set of
ecosystems. We performed artificial ecosystem selection on replicate soil
microbiomes to select for increased soil methane oxidation rate. We observed a
strong response to selection with a `r myround(per_slope, 1)`\% increase in
methane oxidation rate per passage. In addition, we
estimated that `r myround((coefficients(herit_model)[2] +
coefficients(herit_model)[4])*100, 1)`\% of the variation in methane oxidation
rate in these soils can be attributed to microbiome variation. We also found
that selection did not enrich for known methane oxidizers; rather microbes that
responded to selection included members of `r length(da_fams)` families not
known to oxidize methane, such as
*Fimbriimonadaceae*, *Cytophagaceae*, and *Diplorickettsiaceae*. This result is
in contrast to the typical assumption that the rate of an ecosystem function is
limited by the final step in the associated microbial pathway. Our study
demonstrates that variation in microbiome composition can contribute to
variation in the rate of ecosystem function and that manipulating microbiome composition
could be a viable strategy for managing ecosystem functions.

# Introduction

Microbiomes mediate a variety of important functions in ecosystems, and there
is great interest in understanding how attributes of microbiomes may influence
variation in ecosystem functions [@conrad1996; @schimel1998; @crowther2019].
Biodiversity-ecosystem function relationships have been documented for a
variety of macroorganismal communities. For example, plant species richness
correlates with productivity and marine community diversity correlates with
production, consumption, and nutrient release 
[@cardinale2006; @hooper2012; @gamfeldt2015]. However, we have been unsuccessful
in documenting similar relationships for microbial communities. While there is
some evidence that microbiomes from different ecosystems display different
functional rates [@balser2005; @strickland2009], adding microbiome attributes to
ecosystem models adds very little to total variance explained [@graham2016].
Therefore, we still lack a detailed understanding of what
functions are altered by microbiome composition and what attributes of the
microbiome explain this relationship.

There are several reasons for this difficulty. One reason is that it is
challenging to manipulate microbiome composition and diversity directly.
Experimental manipulations of community composition and diversity have been
central to documenting the relationships between ecosystem functions and the
biodiversity of plants and animals [@hooper2012], but this is seldom possible
with microbiomes. Most microorganisms cannot yet be cultured and most available
methods for manipulation of microbiomes (e.g., antibiotics, dilution, etc.) are
not sufficiently specific to manipulate individual microbial lineages. In
addition, many studies have looked for relationships between an ecosystem
function and the abundance or diversity of a microbial gene that codes for an
enzyme or enzyme-subunit known to be involved in that function. However, it is
difficult to identify the “right” marker gene a priori and for most functions
in most ecosystems there is no correlation between the abundance of a marker
gene and the rate of the corresponding process [@rocca2015]. Lastly,
microbiome-ecosystem function relationships have been difficult to document
because microbiomes often covary with other drivers of variation in ecosystem
functions, such as environmental conditions or resources. Such covariation can
make it impossible to determine whether microbiomes directly contribute to
functional variation. Understanding the degree to which microbiomes directly
influence functional variation is crucial if scientists are to develop
microbiome-focused methods for altering or maintaining ecosystem functions. 

One way to overcome these challenges is to manipulate the relationship between
microbiome biodiversity and ecosystem function using artificial ecosystem selection 
[@swenson2000; @blouin2015; @panke-buisse2015; @sanchez2021]. 
Microbiomes have been shown to respond to selection at the ecosystem-trait
level [@swenson2000], and some ecosystem-scale microbiome traits have been
shown to be transferrable through microbiome inoculations [@panke-buisse2015].
Therefore, applying artificial selection to whole microbiomes could be a novel
approach for describing a more detailed microbiome-ecosystem function
relationship. This approach works by subsampling a microbiome to inoculate
multiple replicate ecosystems that vary in microbiome composition and then
measuring the rate of ecosystem function for each replicate. The investigator
then selects the replicates with the highest (or lowest) rates and subsamples
those ecosystems to passage them to a new set of ecosystems. By holding
environmental conditions relatively constant among the replicates and from
passage to passage (or by using isogenic hosts, in the case of host-associated
microbiomes), investigators can determine whether variation in the microbiome
contributes to variation in the rate of ecosystem function independent of the
environment.

A valuable feature of this approach is that it can be used to estimate how much
of the variation in ecosystem function is due to microbiome variation.  By
calculating the slope of the regression between the "donor" ecosystems selected
in one passage and the "recipient" ecosystems inoculated by those donors in the
next passage, we can quantify an estimate of "ecosystem heritability"
[@goodnight2000; @blouin2015]. The concept of heritability has been borrowed
from quantitative genetics where heritability is defined as *the ratio of
additive genetic variance to the total phenotypic variance* and is commonly
estimated as *the slope of the linear regression between offspring phenotype
and parental phenotype* [@bell1977; @falconer1996]. Here we apply this concept
to donor and recipient communities in an artificial ecosystem selection
experiment to estimate *the variance in ecosystem function due to microbiome
variance*. We do not assume that environmental variation is zero, instead we
quantify the proportion of variation due to microbiome effects. If the slope of
the regression is zero, we would conclude that environmental variation among
the replicates and technical error are the primary drivers of variation in
ecosystem function. On the other hand, a significant slope between donors and
recipients would demonstrate that microbiome variation generates variation in
the rate of an ecosystem function. We then use our artificially selected
microbiomes to investigate what aspects of microbiome variation explain this
response. By "microbiome variation" we mean aspects of microbial diversity
commonly measured by microbial ecologists, such as the relative abundance of
taxonomic groups inferred from amplicon sequence variants. With this approach,
we can not only determine whether an ecosystem function is influenced by
microbiome variation, but also estimate how much of the variation and which
taxa are associated with higher rates of ecosystem function.

We applied artificial ecosystem selection to soil microbial communities by
selecting on soil methane oxidation rate. We chose this function because
methane is a globally important greenhouse gas and methane oxidation by soil
bacteria is the primary biological sink for atmospheric methane
[@kirschke2013]. In addition, there is evidence that soil methane oxidation
rate may vary with microbiome composition. For example, soil microbiome
phylogenetic variation is a strong predictor of methane oxidation in forests
and pastures of the Brazilian Amazon as well as soils of the Congo Basin
[@meyer2020; @meyer2020a]. Finally, methanotrophy is one of the most deeply
conserved microbial physiologies and is represented in a narrow range of taxa
and so the taxonomic composition of the microbiome is more likely to be
associated with the rate of methane oxidation than other broader or more
shallowly conserved functions [@schimel1998; @martiny2013]. This suggests that
microbiome variation might drive variation in methane oxidation in soils. One
other study has estimated ecosystem heritability for an environmental
microbiome. Blouin et al. [@blouin2015] selected for water microbiomes with
low carbon dioxide respiration rates and estimated a heritability of 52%.
However, this appeared to be driven primarily by biomass, not microbiome
composition. Here we want to build on this work by selecting for higher rates
of a more narrow function that is less likely to be driven by total biomass.
That way we can begin to assess whether variation in microbiome composition is
important for understanding variation in ecosystem functions.

In this study, we evaluated whether variation in the microbiome contributes to
variation in the rate of ecosystem functions. To address this problem, we
applied artificial ecosystem selection to methane oxidation rate of the soil
microbiome and answered the following questions: Does variation in the relative
abundance of microbial taxa contribute to variation in soil methane oxidation
rate in the lab? How much of the variation in soil methane oxidation rate can
be attributed to the microbiome? What attributes of the microbiome regulate
variation in methane oxidation rate? And do these attributes match our
assumptions about what regulates methane oxidation rate in nature?

# Materials and Methods

## Experimental design

We performed an artificial ecosystem selection experiment 
[*sensu* @swenson2000] by passaging replicate soil microbiomes. The trait we
selected on was CH~4~ oxidation rate. This experiment had two selection lines
with twelve jars each for a total of twenty-four jars per passage. One line was
a positive selection line where the two or three jars with the highest CH~4~
oxidation rate were homogenized to inoculate the next set of positive jars. The
other line was a neutral selection line where an equal number of jars as the
positive line were chosen at random to inoculate the next set of neutral jars.
The number of jars chosen was based on the distribution of fluxes among the
positive jars, i.e., we chose the top three jars unless only two jars had
considerably greater CH~4~ oxidation rates based on visual inspection of
histograms. The experiment was carried out over five passages until a
significant divergence in functional rates was observed between the two
selection treatments based on a difference of slopes.

The initial soil microbiome was sampled from the top 10 cm of an upland mineral
soil under a deciduous forest ecosystem near the University of Oregon campus in
Eugene, OR, USA. Incubations were performed in 500 mL mason jars with rubber
septa installed in the lids. Each jar was sterilized with 70\% ethanol to which
was added 45 g of autoclaved potting mix, 5 g of living soil, and 3.5 mL of
sterile deionized water to bring them to 60\% of field capacity. The potting
mix was Lane Potting Mix from Lane Forest Products (Eugene, OR, USA). Each jar
was then capped and injected with 4.3 mL of 99\% CH~4~ to bring
the headspace concentration to approximately 1000 ppm CH~4~. To create the two
treatment groups, twenty-four jars were created in an identical manner and then
randomly assigned to either the positive or neutral selection line. Jars were
flushed and respiked with methane twice per week to maintain aerobic conditions and elevated
CH~4~ concentrations and were incubated at ambient temperature for
approximately four weeks. Methane oxidation rates were determined at the end
of the incubation period. For the positive treatment, the three jars with the
greatest CH~4~ oxidation rates were chosen to inoculate the next generation.
For the neutral treatment, three jars were randomly selected to inoculate the
next generation. For each treatment, the selected jars were homogenized and 5 g
of the homogenized soil was used to inoculate the next set of jars, which
represents a 10% subsampling. The next set of jars were created in an identical
manner to the first generation with fresh autoclaved potting mix and the same
moisture and CH~4~ content.

## Methane oxidation rate

Methane oxidation rates were determined after flushing and spiking jars to
approximately 1000 ppm CH~4~. Headspace samples of 1 mL were collected from
each jar immediately after spiking and then at time points 3, 6, 24, and 48
hours for a 5-point curve. Samples were immediately injected into a SRI model
8610C gas chromatograph equipped with a flame ionization detector (SRI
Instruments, Torrance, CA, USA) to determine the headspace CH~4~ concentration.
We applied a first-order exponential decay function to determine the rate
constant (k, units = d^−1^; i.e., dCH~4~/dt = k[CH~4~]) of the exponential
decrease in methane. Oxidation rates are presented as the additive inverse of
$k$ (i.e., $-k$) so that a more positive value represents a greater
oxidation rate. The jars selected in passage two for the positive treatment had
the lowest CH~4~ oxidation rate of the twelve jars due to a calculation error
in the rate constant. Despite this error, we still observed a strong
response to selection demonstrating the power of this technique. In addition,
this likely reintroduced diversity lost during the selection process,
potentially increasing the variation available to be selected upon, as
is recommended for microbiome selection experiments [@sanchez2021].

## Soil DNA extraction and sequencing

A subsample of soil from the starting inoculum and from every jar in passages 2
and 5 was collected and stored at $-80^\circ$C. Soil DNA was extracted from
0.25 g soil. Negative controls were extracted from autoclaved potting mix and
DNase-free water. Extractions were performed using the DNeasy PowerSoil kit
(QIAGEN, Düsseldorf, Germany) and quantified using Qubit dsDNA HS Assay Kit
(Thermo Fisher Scientific, Inc., Waltham, MA, USA). To estimate the diversity
and relative abundance of the bacterial and archaeal taxa in our soil
ecosystems, we sequenced the V4 region of the 16S rRNA gene using the 515F -
806R primer combination [@caporaso2011]. PCR mixtures were: 10 $\mu$l NEBNext
Q5 Hot Start HiFi PCR master mix, 9.2 $\mu$l primer mixture (1.09 $\mu$M
concentration), and 0.8 $\mu$l of DNA template. Reaction conditions were:
$98^\circ$C for 30 s (initialization); 35 cycles of $98^\circ$C for 10 s
(denaturation), $61^\circ$C for 20 s (annealing), and $72^\circ$C for 20 s
(extension); and $72^\circ$C for 2 m (final extension). Reactions were
performed in triplicate and then combined.  Amplicons were purified twice using
0.8x ratio Mag-Bind RxnPure Plus isolation beads (Omega Bio-Tek, Norcross, GA,
USA). Sequencing libraries were prepared using a dual-indexing approach
[@fadrosh2014; @kozich2013]. Amplicon concentrations were quantified using
Qubit and multiplexed at equimolar concentration. Sequencing was performed at
the University of Oregon Genomics Core Facility on the Illumina NovaSeq 6000
with paired-end 150 bp reads (Illumina, Inc., San Diego, CA, USA).

## Bioinformatics

Bioinformatics processing was performed in 'R' [@rcoreteam2018].  Demultiplexed
sequencing reads were denoised using 'DADA2' to generate a table of amplicon
sequence variants (ASVs) [@callahan2016]. Taxonomic assignment was performed
using the Ribosomal Database Project naive Bayesian classifier [@wang2007]. The
presence of contaminants was evaluated using both the prevalence and frequency
methods from 'DECONTAM' [@davis2018]. Decontam identified 
`r length(which(contamdf$contaminant))` potential contaminants based on
prevalence and frequency. Visual inspection of abundance-concentration plots
indicated that `r length(contaminants)` of these were likely contaminants and
these ASVs were removed. Amplicon sequence variants that were assigned
chloroplast or mitochondria taxonomy were removed prior to analysis.

## Statistical Analysis

Statistical analyses were performed in 'R' [@rcoreteam2018]. To determine
whether there was a significant change in CH~4~ oxidation rate as a response to
selection, we tested a difference in slopes between the positive and neutral
selection lines. Residuals did not meet the assumptions of constant variance
and normal distribution. Therefore, CH~4~ oxidation rates were log~10~
transformed prior to analysis. First, we tested if there was a difference of
slopes between the positive and neutral treatments based on the interaction
between passage and treatment. To test the interaction, we fit two nested models
with and without the interaction term and compared them using an F-test with the
'anova' function. We then present the slopes for each treatment, which
represented the change in CH~4~ oxidation rate per passage as a response to
selection.

We estimated the ecosystem heritability of methane oxidation rate, which is the
proportion of variance in CH~4~ oxidation rate due to variation in the
microbiome, as the regression of mid-recipient on mid-donor [@falconer1996].
The mid-donor was the mean for all three selected jars and the mid-recipient was
the mean methane oxidation rate for the twelve jars produced by those three
donors. First, we tested if there was a difference of slopes between the
positive and neutral treatments. To test this, we fit two nested models with and
without the treatment term and compared them using an F-test with the 'anova'
function. We then present the slopes for each treatment, which represented the
ecosystem heritability or the proprotion of total variance in methane oxidation
rate due to microbiome variance.


First, we tested if there was an effect of treatment by comparing
nested models with and without treatment using an F-test with the 'anova'
function. We then fit the full model to estimate the slope of the relationship
between recipient and donor for the Positive and Neutral treatments.

Richness was estimated using the method from @hurlbert1971 with a subsample
size of `r min_seqs` using the 'rarefy' function in 'vegan' [@oksanen2019]. We
tested a difference in richness by both passage and treatment with a
Kruskal-Wallace test followed by a pairwise Wilcoxon test. Next, we estimated
beta-diversity as the Bray-Curtis dissimilarity by averaging 100 random subsets
with a subsample size of `r min_seqs` using the 'avgdist' function in 'vegan'
[@bray1957; @oksanen2019].  We tested a difference in centroid and dispersion
of beta diversity by passage and treatment using a permutational analysis of
variance (PERMANOVA) with 999 permutations using the 'adonis2' function from
'vegan' and tested a difference of group dispersions using 'betadisper' and
'anova' with 999 permutations [@mcardle2001; @oksanen2019].  Lastly, we tested
the correlation between CH~4~ oxidation rate and Bray-Curtis dissimilarity in
Passage 5 with a distance-based redundancy analysis (dbRDA) using the 'dbrda'
function in 'vegan' and estimated the p-value using a permutation F-test with
999 permutations [@mcardle2001; @oksanen2019]

To identify taxa that responded to selection on CH~4~ oxidation rate, we tested
differential abundance between the two treatments in passage 5. We first
grouped ASVs at the family level. Any ASVs that lacked a family-level
taxonomic assignment were grouped at a higher taxonomic level. We then subset the
samples in Passage 5 and removed all families with a prevalence of less than
10\% in either treatment. We used three methods for testing differential
abundance: ANCOM-II, ALDEx2, and CORNCOB 
[@kaul2017; @lin2020; @fernandes2014; @martin2020]. We then identified the
consensus taxa that were significant with all three tests and plotted their
relative abundances. For ANCOM-II, we used the 'ancom' function in the
'ANCOM-BC' package with a cutoff of W = 0.7 [@kaul2017; @lin2020]. For ALDEx2,
we used the 'aldex' function in the 'ALDEx2' package with Welch's t-test and we
used an effect size of 1 as our significance threshold [@fernandes2014].
Finally, we used CORNCOB with the 'differentialTest' function in the 'corncob'
package with the Wald test and without bootstrapping [@martin2020]. Lastly, to
test differentially abundant methanotrophs, we subset all ASVs within
methanotrophic families and tested their differential abundance aggregated at
the family and genus level using 'corncob'.

# Results

## Response to selection on methane oxidation rate

We observed a response to artificial selection on whole-ecosystem soil CH~4~ oxidation
rate (`r figures("response", display = "cite")`; difference of slopes:
F~`r compare_response_mods[2,4]`,`r compare_response_mods[2,2]`~ = `r myround(compare_response_mods[2,5], 2)`, p = `r myround(compare_response_mods[2,6], 2)`). At the start of the experiment, the Positive
treatment had a mean CH~4~ oxidation rate that was
`r percent_less(initial = tidy(response_model)[[1, 2]], final = tidy(response_model)[[3, 2]] + tidy(response_model)[[1, 2]])`\% lower than the
Neutral treatment `r print_lm(response_model, 3, 'difference of y-intercepts')`. 
There was no change in CH~4~ oxidation rate of the Neutral
treatment over the five passages `r print_lm(response_model, 2)`. By contrast,
the Positive treatment had a `r myround(per_slope, 1)`\% increase in CH~4~
oxidation rate per passage `r print_lm(response_model, 4)`.

We estimated the ecosystem heritability defined as the 
proportion of variation in methane oxidation due rate to variation
in the microbiome as the regression of mid-recipient on mid-donor 
(`r figures("herit", display = "cite")`).  Recipient CH~4~ oxidation rates were
correlated with donor CH~4~ oxidation rates in both the Positive treatment
`r paste0('(', 'slope', ' = ', myround(tidy(herit_model)[2, 2] + tidy(herit_model)[4, 2], 2), ', SE = ', myround(tidy(herit_model)[4, 3], 2), ', t = ', myround(tidy(herit_model)[4, 4], 2), ', p = ', myround(tidy(herit_model)[4, 5], 2), ')')` and 
the Neutral treatment `r print_lm(herit_model, 2)`. 

## Taxonomic richness

Median ASV richness decreased from `r myround(rare_median[1, 2], 1)` (`r myround(rare_median[1, 3], 1)`) in Passage 2 to `r myround(rare_median[2, 2], 1)` (`r myround(rare_median[2, 3], 1)`) in Passage 5 (Kruskal-Wallace test: $\chi^2$ = 
`r myround(kt[1,1], 1)`, df = `r kt[1,3]`, p < 0.001; pairwise Wilcoxon test: p < 0.001). However, there was no
difference in richness between the Positive and the Neutral treatment in
Passage 2 or 5 (pairwise Wilcoxon test: p = `r myround(wt[1, 3], 1)`).  In addition, there was no
correlation between richness and CH~4~ oxidation rate across the two treatments in Passage 5 (Spearman's rho = `r myround(s5[1,1], 1)`, p = `r myround(s5[1,3], 1)`).

## Community dissimilarity

Bray-Curtis dissimilarity of the soil microbiome varied strongly by passage and
weakly by treatment with an interaction between passage and treatment (`r figures("beta", display = "cite")`). 
Passage explained `r myround(100*tidy_permanova[1, 4])`\% of the
variation in Bray-Curtis dissimilarity (F~`r tidy_permanova[1, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[1, 5], 1)`, p = `r myround(tidy_permanova[1, 6], 3)`), treatment explained 
`r myround(100*tidy_permanova[2, 4])`\%
of the variation (F~`r tidy_permanova[2, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[2, 5], 1)`, p = `r myround(tidy_permanova[2, 6], 3)`), and the interaction
between treatment and passage explained 
`r myround(100*tidy_permanova[3, 4])`\% of the variation (F~`r tidy_permanova[3, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[3, 5], 1)`, p = `r myround(tidy_permanova[3, 6], 3)`). There was 
no difference in dispersion between treatments or passages (F~3,44~ = 0.91, p = 0.45). 
Finally, CH~4~ oxidation rate was correlated with Bray-Curtis dissimilarity across both treatments in Passage 5 and explained `r myround(dbrda_percent_explained[1], 1)`\% of the variation in Bray-Curtis dissimilarity (dbRDA: F~`r dbrda_aov[1, 1]`,`r dbrda_aov[2, 1]`~ = `r myround(dbrda_aov[1, 3], 2)`, p = `r myround(dbrda_aov[1, 4], 3)`)

## Taxa that responded to selection

To identify taxa that responded
to selection on soil CH~4~ oxidation rate, we tested the differential relative
abundance of families in the Positive jars relative to the Neutral jars within
Passage 5 using three methods and then plotted the taxa identified by all three methods. We identified `r length(da_fams)`
families that were enriched or depleted in the Positive treatment relative to
the Neutral treatment (`r figures("da", display = "cite")`). 

Overall, none of the families enriched in the Positive selection treatment
contain known methanotrophs. Several taxa identified had a 
higher taxonomic designation that contains methanotrophs, for example, the
Gammaproteobacteria class had a large effect size.
The Gammaproteobacteria include the type I and type X methanotrophs in the families *Methylococcaceae* and *Methylothermaceae* [@stein2012].
However, the Gammaproteobacteria is among the most diverse 
groups in the Prokaryotes, so this is not strong evidence for a selection response
by methanotrophs [@garrity2005].
In addition, the *Puniceicoccaceae* is a member of the phylum Verrucomicrobia. The Verrucomicrobia is a diverse group that contain
known methanotrophs as well as ammonia-oxidizing bacteria [@freitag2003]. Other than these two groups, none of the other taxa enriched in the Positive treatment are known to be related to methanotrophs. Two groups in the Armatimonadales were enriched in the Positive treatment including the family *Fimbriimonadaceae* and an unclassified ASV from the order Armatimonadales [@im2012].
*Cytophagaceae* was also enriched in
the Positive treatment and contains a number of mainly aerobic heterotrophs
that can digest a variety of macromolecules [@mcbride2014]. 
The remaining families include the uncultured family 0319-6G20, *Diplorickettsiaceae*, *Rhodospirillaceae*, and an unclassified Kapabacteriales.

We did not identify any methanotrophic families in the overall differential abundance analysis. 
However, we wanted to look more closely at the known methanotrophs in our dataset to be sure they did not have an effect.
To do this, we subset all of the ASVs in our dataset that were in families that contained methanotrophs.
Only two families were represented: *Methylacidiphilaceae* and *Beijerinckiaceae*. 
Aggregating reads at the family level, neither family was differentially abundant between the two treatments.
However, aggregated at the genus level, a group of unclassified genera in the *Beijerinckiaceae* were depleted in the Positive treatment and the genus
*Rhodoblastus*, a member of the *Beijerinckiaceae*, was enriched in the
Positive treatment.  While many *Beijerinckiaceae* are methanotrophs, several
taxa in this family have lost the the ability to oxidize CH~4~ and it appears
that *Rhodoblastus* species are not able to grow on CH~4~, though they can grow
on methanol [@tamas2014, @dedysh2016]. Based on this analysis,
it appears that no methanotrophs were enriched in the Positive treatment.


# Discussion

While it is well understood that the microbiome mediates a variety of ecosystem
functions, a fundamental question in microbial ecology is whether we need to
understand variation in the composition of microbiomes to understand variation
in the rate of ecosystem functions [@conrad1996; @schimel1998]. To address this
question, we performed artificial ecosystem selection on soil microbiomes by
selecting for ecosystems with high methane oxidation rates. We found that there
was a significant increase in methane oxidation rate in the Positive selection
treatment relative to the Neutral selection treatment. This demonstrates that
passaging the microbiome through artificial ecosystem selection resulted in a
shift in the mean methane oxidation rate of the soil microbiome. 
This suggests that variation in the microbiome can generate variation in
ecosystem functions independent of the environment.

We quantified how much of the variation in methane oxidation rate was due to
variation in the microbiome by regressing the mean methane oxidation rate of
the 12 recipient jars against the mean methane oxidation rate of the three
donor jars selected to inoculate the recipients. We observed a heritability of
`r myround((coefficients(herit_model)[2] + coefficients(herit_model)[4])*100,1)`\% 
for the Positive selection treatment. This indicates that 
`r myround((coefficients(herit_model)[2] + coefficients(herit_model)[4])*100,1)`\% 
of the variation in methane oxidation rate can be attributed to variation in the
microbiome. This is comparable to the ecosystem heritability of carbon dioxide
emissions from water [@blouin2015]. Notably, we observed a negative heritability
in the Neutral treatment (`r myround(coefficients(herit_model)[2]*100, 1)`\%).
Negative heritability estimates are not uncommon in quantitative genetics
[@steinsaltz2020]. This can occur for random selection controls in artificial
selection experiments and may indicate that methane oxidation rate is negatively
correlated with other traits associated with persistence in jars. The
heritability we observed in the Positive treatment indicates that a large
portion of the variation in methane oxidation rate is due to variation in the
microbiome in our experimental system.

Given that variation in the microbiome is driving the observed variation in
methane oxidation rate, we next wanted to address what aspects of the
microbiome, in terms of alpha diversity, beta diversity, or the relative
abundance of taxa, explain this contribution of the microbiome to variation in
methane oxidation rate. There are three ways that the microbiome could respond
to selection in this experiment: gain or loss of species, changes in the
relative abundances of species, or changes within the genomes of the constituent
species. Using 16S rRNA gene sequences, we are unable to address the third
possibility of whether taxa in this experiment could have evolved changes in
their genomes as a result of selection.  Therefore, we will focus on the first
two possibilities. 

Richness at the ASV level did not vary between the two treatments and there
were relatively few taxa gained or lost in the Positive selection treatment and
none of these were prevalent across the 12 jars in passage 5. Therefore, the
gain or loss of species is unlikely to explain the increase in methane
oxidation rate. However, we found that Bray-Curtis dissimilarity was greater
between the two treatments in Passage 5 than within each treatment and was
correlated with methane oxidation rate, which suggests that changes in the
relative abundance of taxa could explain the response to selection.

Even though we observed an increase in methane oxidation rate in the Positive
treatment and a difference in composition between the two treatments, we did
not observe an increase in the relative abundance of methanotrophs. This was
surprising given that methane consumption is not a common trait among microbes
and that it is often assumed that the rate of an ecosystem function is limited
by the final enzymatic step in a metabolic pathway [@rocca2015]. In certain
ecosystems, methane production and consumption are correlated with the
abundance of methanogens and methanotrophs as estimated from marker genes
[@meyer2020; @meyer2020a]. However, it does not appear generally true that the
rate of an ecosystem function is limited by the abundance of the gene that
encodes the final step in that pathway [@rocca2015]. Our results suggest that
ecosystem-scale methane oxidation rates can be altered by non-methanotrophs,
perhaps through ecological interactions with methanotrophic species. 

Another important question is whether selection at the ecosystem scale resulted
in within-strain evolution. Using 16S rRNA gene sequencing we are unable to
resolve genomic changes within taxa that may have undergone evolution.
Therefore, it is possible that the response to selection was driven by changes
within the genomes of methanotrophs. But given the slow growth rate of
methanotrophs, often requiring weeks to years of serial batch culturing for
isolation, this seems like an unlikely explanation 
[@whittenbury1970; @hirayama2014; @kim2018]. Future work could analyze an
artificial ecosystem selection experiment using genome-resolved metagenomics to
try to detect strain-level changes in the constituent taxa. However,
reconstructing genomes from in situ soil samples has been difficult to achieve
and may require enrichment techniques, particularly for methanotrophs which are
a relatively rare component of the soil microbiome [@delmont2015]. If these
challenges could be overcome for soil metagenomics, we could begin to evaluate
whether ecosystem selection results in strain-level evolution or enriches for
certain metabolic pathways within the microbiome.

A fundamental question in microbial ecology is whether variation in the
microbiome contributes to variation in ecosystem function. We performed
artificial ecosystem selection on soil methane oxidation rate to test the
relationship between microbiome variation and soil methane oxidation rate. We
observed a response to selection on soil methane oxidation rate and found a
high heritability for methane oxidation, which suggests that variation in the
microbiome contributes to variation in soil methane oxidation rate
independent of the environment. This shows that manipulating the microbiome
could be a viable strategy for mitigating or enhancing certain
ecosystem functions.  Surprisingly, we did not observe an increase in the
relative abundance of methanotrophs as a response to selection. This
suggests that the rate of methane oxidation could be altered by
non-methanotrophs, possibly through ecological interactions with methanotrophs.
Future research should investigate other aspects of microbiome variation, such
as strain-level evolution or functional gene composition, to begin to describe
a more detailed mapping between the microbiome and ecosystem function. This
study demonstrates that soil methane oxidation rate can vary greatly with
variation in the microbiome and opens up future research opportunities to
establish a more detailed understanding of the relationship between the
microbiome and ecosystem function.

# Acknowledgments

This project was supported by the National Science Foundation Graduate Research
Fellowship Program (grant no. DGE 1255832) and the ARCS Foundation Florence and
Mike Nudelman Scholarship.

# Competing Interests

We declare we have no competing interests.

# Data Availability Statement

The 16S rRNA sequencing data generated during the current study are available
in the NCBI Sequence Read Archive (SRA) under BioProject accession number
PRJNA832314, https://www.ncbi.nlm.nih.gov/sra/PRJNA832314. The metadata
generated during the current study as well as the scripts to recreate the
analysis are available on Github,
https://github.com/amorris28/artificial_ecosystem_selection.

# Supplement


# References

::: {#refs}
:::

# Figures


```{r response, fig.width=4, fig.height=3}

p_response

```

`r figures("response", "Response to selection on soil CH~4~ oxidation rate. The y-axis is the log~10~ of the additive inverse of the first-order exponential decay constant $k$ (i.e., $-k$) with units day^-1^. The x-axis is the passage number. The orange line and points are the Positive treatment and the gray line and points are the Neutral treatment.")`


```{r herit, fig.width=5, fig.height=3}

p_herit

```

`r figures("herit", paste0("Ecosystem heritability as the regression of mid-recipient methane flux on mid-donor methane flux ($-k$ with units day^-1^). Donor is the mean oxidation rate of the three jars selected to inoculate the next passage. Recipient is the mean oxidation rate of the 12 jars in the next passage. The orange lines and points are the Positive treatment and the gray lines and points are the Neutral treatment. Slope of the regression line estimates the proportion of variation in methane oxidation rate due to variation in the microbiome. For the Positive treatment, the variance explained by the microbiome was ", myround((coefficients(herit_model)[2] + coefficients(herit_model)[4])*100, 1), "% (slope = ", myround(coefficients(herit_model)[2] + coefficients(herit_model)[4], 3), ", SE = ", myround(tidy(herit_model)$std.error[4], 2), ") and for the Neutral treatment the variance explained by the microbiome was ", myround(coefficients(herit_model)[2]*100, 1), "% (slope = ", myround(coefficients(herit_model)[2], 3), ", SE = ", myround(tidy(herit_model)$std.error[2], 2), ")."))`

```{r beta, results=FALSE, fig.height=2.5}

p_beta

```

`r figures("beta", "Principal component analysis plot of beta diversity for all jars. Dissimilarities are based on rarefied Bray-Curtis dissimilarity. Colors represent treatment and shapes represent passage number.")`

```{r da, results=FALSE}

p_log10

```

`r figures("da", "Differentially abundant family-level taxa identified by ANCOM-II, ALDEx2, and CORNCOB. Values on the x-axis are relative abundances on a log~10~ scale. Points are colored by treatment. Large points with error bars are the median relative abundance and interquartile range. Taxa in the top panel are enriched in the Positive treatment and taxa in the bottom panel are depleted in the Positive treatment. Taxa are sorted by their effect size with taxa at the top having the largest positive effect size and taxa at the bottom with the largest negative effect size.")`

