---
title: Artificial ecosystem selection reveals relationships between microbiome composition and ecosystem function
author: "Andrew H. Morris^1^* and Brendan J. M. Bohannan^1^"
output:
  word_document:
    reference_docx: reference.docx
bibliography: bibliography.bib
csl: vancouver.csl
---

^1^ Institute of Ecology and Evolution, University of Oregon, Eugene, OR, USA

\* Correspondence: Andrew H. Morris, amorris3@uoregon.edu

Competing Interests: We declare we have no competing interests.

```{r options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)

```

```{r initialize-packages}
library(tidyverse)
library(broom)
library(broman)
library(morris)
# library(phyloseq)
library(ggpubr)
library(knitr)
library(corncob)
# library(ggthemes)
library(captioner)
library(knitr)
library(ggtext)
figures <- captioner()
tables <- captioner(prefix = "Table")
```

```{r load-data}
heritability <- read_tsv('../Output/heritability.tsv')
fluxes <- read_tsv('../Output/fluxes.tsv')

load(file = '../Output/decontam.RData')


# CH4 models
load(file = '../Output/response_model.RData')
load(file = '../Output/response_plot.RData')
load(file = '../Output/herit_model.RData')
load(file = '../Output/herit_plot.RData')

# 16S models
load(file = '../Output/richness_models.RData')
load(file = '../Output/phyloseq.RData')
load(file = '../Output/beta_model.RData')
load(file = '../Output/beta_fig.RData')
load(file = '../Output/da_corncob.RData')
load(file = '../Output/da_fig.RData')
load(file = '../Output/dbrda.RData')


source('../R/functions.R')

```

# Abstract

Microbiomes mediate important ecosystem functions, yet it has proven difficult
to determine the relationship between microbiome composition and the rate of
ecosystem functions. This challenge remains because it is difficult to
manipulate microbiome composition directly, we often cannot know a priori which
members of a microbiome influence the rate of an ecosystem function, and
microbiomes can covary strongly with other drivers of ecosystem function, such
as the environment. We propose that artificial ecosystem selection can overcome
these challenges. Artificial ecosystem selection involves creating replicate
ecosystems, selecting replicates with a desired ecosystem-level trait, and
using the microbiomes from the selected ecosystems to inoculate a new set of
ecosystems. We performed artificial ecosystem selection on replicate soil
microcosms to select for increased soil methane oxidation rate. We observed a
strong response to selection with a 51% increase in methane oxidation rate per
round over five rounds of selection. This suggests that a large portion of the
variation in methane oxidation rate in these soils can be attributed to
microbiome variation. We also found that selection did not enrich for known
methane oxidizers; rather microbes that responded to selection included members
of `r length(da_fams)` families such as *Fimbriimonadaceae*, *Cytophagaceae*,
and *Diplorickettsiaceae*. This result is in contrast to the typical assumption
that the rate of an ecosystem function is limited by the final step in the
associated microbial pathway. Our study demonstrates that variation in
microbiome composition can contribute to variation in the rate of ecosystem
function and that this relationship can be revealed via artificial ecosystem
selection.

# Introduction

Microbiomes mediate a variety of important functions in ecosystems, and there
is great interest in understanding how attributes of microbiomes may influence
variation in ecosystem functions [@conrad1996; @schimel1998; @crowther2019]. Biodiversity-ecosystem
function relationships have been described for a variety of macroorganismal
communities. For example, plant species richness correlates with productivity
and marine community diversity correlates with functions such as production,
consumption, and nutrient release
[@cardinale2006; @hooper2012; @gamfeldt2015]. While there is some evidence that
variation in microbiomes can contribute to variation in ecosystem functions,
documenting this relationship has been elusive [@conrad1996; @rocca2015; @graham2016].

There are several reasons for this difficulty. One reason is that it is
challenging to manipulate microbiome composition and diversity directly.
Experimental manipulations of community composition and diversity have been
central to documenting the relationships between ecosystem functions and the
biodiversity of plants and animals [@hooper2012], but this is seldom possible
with microbiomes. Most microorganisms cannot yet be cultured and most
available methods for manipulation of microbiomes (e.g., antibiotics, dilution,
etc.) are not sufficiently specific to manipulate individual microbial
lineages. Microbial ecologists are thus often restricted to using comparative
methods, in which microbiome composition and diversity is surveyed via
environmental DNA and then correlated with the rate of an ecosystem function
{cite examples}. However, such studies are rarely successful [@graham2016]. In an
attempt to increase the likelihood of detecting such a correlation, microbial
ecologists may focus their survey efforts on a particular group of microbes
thought to be involved in the ecosystem function of interest. For example, many
studies have looked for relationships between an ecosystem function and the
abundance or diversity of a microbial gene that codes for an enzyme or
enzyme-subunit known to be involved in a particular function. But this approach
assumes that one can identify the “right” marker gene a priori which, given how
little is understood about microbial diversity, seems highly unlikely. And in
fact these studies are rarely successful [@rocca2015]. 

Another likely reason that microbiome-ecosystem function relationships have
been difficult to document is that microbiomes often covary with other drivers
of variation in ecosystem functions, such as environmental conditions or
resources. Such covariation can make it impossible to determine whether
microbiomes directly contribute to functional variation. Understanding the
degree to which microbiomes directly influence functional variation is crucial
if scientists are to develop microbiome-focused methods for altering or
maintaining ecosystem functions. 

The problem of connecting microbiome attributes to the rate of an ecosystem
function is analogous to the problem of connecting genomic variation to
phenotypic traits in organisms, and solutions to this problem may be found by
exploring this analogy [@morris2020]. For example, one approach to the problem
of genotype-phenotype mapping is to perform artificial selection on a trait of
interest and identify genes that respond to selection. Such genes are likely to
be causally linked to the trait under selection. An analogous approach for
exploring microbiome-ecosystem function relationships is gaining acceptance
among microbiologists [@swenson2000; @panke-buisse2015; @sanchez2021].
Microbiomes have been shown to respond to selection at the ecosystem-trait
level [@swenson2000], and some ecosystem-scale microbiome traits have been
shown to be transferable through microbiome inoculations [@panke-buisse2015].
Applying artificial selection to whole microbiomes could be a novel approach
for quantifying how much of the variation in the rate of an ecosystem function
could be attributed independently to microbiome variation [@goodnight2000]. In
addition, this approach could be used to identify microbiome attributes, such
as genes or taxa, associated with the rate of an ecosystem function, although,
to our knowledge this has not previously been accomplished [@arias-sanchez2019;
@morris2020]. Such an approach has the potential to expand our ability to more
accurately model microbial ecosystem-function relationships and allow us to
manage ecosystems for particular outcomes [@fuhrman2009; @schimel1998].

We use this approach to estimate how much of the variation in an ecosystem
function - the flux of CH~4~ gas between the soil and atmosphere - can be
attributed to variation in the microbiome and to identify microbial markers
associated with this ecosystem function.  Soil CH~4~ oxidation is a suitable
function for this study because there is evidence that it may vary with
microbiome attributes. For example, variation in CH~4~ emissions in arctic
permafrost is correlated with the transcriptional activity of certain
methanogens [@freitag2009; @freitag2010]. In addition, soil microbiome
phylogenetic variation is a strong predictor of CH~4~ oxidation in forests and
pastures of the Brazilian Amazon and the Congo Basin [@meyer2020; @meyer2020a]. Finally, methanogenesis and
methanotrophy are two of the most deeply conserved microbial physiologies and
are represented in a narrow range of taxa and so the taxonomic composition of
the microbiome is more likely to be associated with the rate of CH~4~ flux than
other broader or more shallowly conserved functions [@schimel1998;
@martiny2013]. This suggests that microbiome variation might drive variation in
CH~4~ oxidation in soils.

We applied artificial ecosystem selection on soil microbiomes by selecting on
the ecosystem-scale CH~4~ oxidation rate.  Using this approach, we were able to
estimate how much of the variation in CH~4~ oxidation rate was associated with
variation in the microbiome.  To identify which microorganisms could be markers
of CH~4~ oxidation rate, we compared the composition of the artificially
selected microbiomes to a control set of microbiomes with random selection.  We
then evaluated whether these markers met our underlying assumptions about
which taxa limit the rate of ecosystem function.

# Materials and Methods

## Experimental design

We performed an artificial ecosystem selection experiment [sensu @swenson2000]
by passaging replicate soil microbiomes. The trait we selected on was
CH~4~ oxidation rate. This experiment had two selection lines with twelve jars
each for a total of twenty-four jars per passage. One line was a positive
selection line where the two or three jars with the highest CH~4~ oxidation rate
were chosen to inoculate the next set of positive jars. The other line was a
neutral selection line where an equal number of jars as the positive line were
chosen at random to inoculate the next set of neutral jars. The number of jars
chosen was based on the distribution of fluxes among the positive jars, i.e., we
chose the top three jars unless only two jars had considerably greater CH~4~ oxidation
rates based on visual inspection of histograms. The experiment was carried out
over five passages until a significant divergence in functional rates was 
observed between the two selection treatments.

The initial soil microbiome was sampled from the top 10 cm of an upland mineral
soil under a deciduous forest ecosystem near the University of Oregon campus in
Eugene, OR, USA. Incubations were performed in 500 mL mason jars with rubber
septa installed in the lids. Each jar was sterilized with 70\% ethanol to which
was added 45 g of autoclaved potting mix, 5 g of living soil, and 3.5 mL of
sterile deionized water to bring them to 60\% of field capacity. The potting mix was
Lane Potting Mix from Lane Forest Products (Eugene, OR, USA). The jars were
then homogenized, capped, and injected with 4.3 mL of 99\% CH~4~ to bring
the headspace concentration to approximately 1000 ppm CH~4~. To create the
two treatment groups, twenty-four jars were created in an identical manner and
then randomly assigned to either the positive or neutral selection line. Jars
were flushed and respiked twice per week to maintain aerobic conditions and
elevated CH~4~ concentrations and were incubated at ambient temperature for
approximately three weeks. Methane oxidation rates were determined at the end
of the incubation period. For the positive treatment, the three jars with the
greatest CH~4~ oxidation rates were chosen to inoculate the next generation.
For the neutral treatment, three jars were randomly selected to inoculate the
next generation. For each treatment, the selected jars were homogenized and 5 g
of the homogenized soil was used to inoculate the next set of jars, which
represents a 10% subsampling. The next set of jars were created in an identical
manner to the first generation with fresh autoclaved potting mix and the same
moisture and CH~4~ content.

## Methane oxidation rate

Methane oxidation rates were determined after flushing and spiking jars to
approximately 1000 ppm CH~4~. Headspace samples of 1 mL were collected from
each jar immediately after spiking and then at time points 3, 6, 24, and 48
hours for a 5-point curve. Samples were immediately injected into a SRI model
8610C gas chromatograph equipped with a flame ionization detector (SRI
Instruments, Torrance, CA, USA) to determine the headspace CH~4~ concentration.
Fluxes were calculated from a first-order exponential decay function as decay
constant $k$ with units day^-1^. Oxidation rates are presented as the additive
inverse of $k$ (i.e., $-k$) so that a more positive value represents a greater
oxidation rate. The jars selected for the positive treatment in passage 2 had
the lowest CH~4~ oxidation rate of the twelve jars due to a calculation error
in the CH~4~ oxidation rate. Despite this error, we still observed a strong
response to selection demonstrating the power of this technique. In addition,
this likely reintroduced diversity lost during the selection process,
potentially increasing the variation available to be selected upon, as
is recommended for microbiome selection experiments [@sanchez2021].

## Soil DNA extraction and sequencing

A subsample of soil from the starting inoculum and from every jar in passages 2
and 5 was collected and stored at $-80^\circ$C. Soil DNA was extracted from 0.25
g soil. Negative controls were extracted from autoclaved potting mix and
DNase-free water. Extractions were performed using the DNeasy PowerSoil kit
(QIAGEN, Düsseldorf, Germany) and quantified using Qubit dsDNA HS Assay Kit (Thermo Fisher
Scientific, Inc., Waltham, MA, USA). To estimate the diversity and relative abundance of
the bacterial and archaeal taxa in our soil ecosystems, we sequenced the V4
region of the 16S rRNA gene using the 515F - 806R primer combination
[@caporaso2011]. PCR mixtures were: 10 $\mu$l NEBNext Q5 Hot Start HiFi PCR
master mix, 9.2 $\mu$l primer mixture (1.09 $\mu$M concentration), and 0.8
$\mu$l of DNA template. Reaction conditions were: $98^\circ$C for 30 s
(initialization); 35 cycles of $98^\circ$C for 10 s (denaturation), $61^\circ$C
for 20 s (annealing), and $72^\circ$C for 20 s (extension); and $72^\circ$C for
2 m (final extension). Reactions were performed in triplicate and then combined.
Amplicons were purified twice using 0.8x ratio Mag-Bind
RxnPure Plus isolation beads (Omega Bio-Tek, Norcross, GA, USA). Sequencing libraries were
prepared using a dual-indexing approach [@fadrosh2014; @kozich2013]. Amplicon
concentrations were quantified using Qubit and multiplexed at equimolar concentration. Sequencing
was performed at the University of Oregon Genomics Core Facility on the Illumina
NovaSeq 6000 with paired-end 150 bp reads (Illumina, Inc., San Diego, CA, USA).

## Bioinformatics

Bioinformatics processing was performed in 'R' [@rcoreteam2018].
Demultiplexed sequencing reads were denoised using 'DADA2' to generate a table
of amplicon sequence variants (ASVs)
[@callahan2016]. Taxonomic assignment was performed
using the Ribosomal Database Project naive Bayesian classifier [@wang2007]. The presence of
contaminants was evaluated using both the prevalence and frequency methods from
'DECONTAM' [@davis2018].
Decontam identified `r length(which(contamdf$contaminant))` potential contaminants based
on prevalence and frequency. Visual inspection of abundance-concentration
plots indicated that `r length(contaminants)` of these were likely contaminants and these
ASVs were removed. Amplicon sequence variants that were assigned chloroplast taxonomy were removed prior to analysis.

## Statistical Analysis

Statistical analyses were performed in 'R' [@rcoreteam2018]. To determine
whether there was a significant change in CH~4~ oxidation rate as a
response to selection, we tested a difference in slopes between the positive
and neutral selection lines. Residuals did not meet the assumptions of constant variance
and normal distribution. Therefore, CH~4~ oxidation rates were log~10~
transformed prior to analysis. First, we tested the difference of slopes between
the positive and neutral treatments by
fitting two nested models with and without the interaction between treatment
and passage using 'lm'. We compared these
models using an F-test with the 'anova' function. We
then fit the full model with the interaction term to estimate the slopes for each treatment, which
represented the change in CH~4~ oxidation rate per passage as a response to
selection.

To estimate the proportion of variance in CH~4~ oxidation rate due to variation in
the microbiome, we estimated narrow-sense heritability ($h^2$) as the
regression of offspring on mid-parent [@falconer1996]. The mid-parent was the
mean for all three selected jars and the offspring was the CH~4~ oxidation rate for each jar produced by those parents. First, we tested if there was an effect of treatment
on the heritability estimate. We compared nested models with and without treatment using an F-test
test with the 'anova' function. We then fit the full model to estimate the
heritability of the Positive and Neutral treatments.

Richness was estimated using the method from @hurlbert1971 with a subsample
size of `r min_seqs` using the 'rarefy' function in 'vegan' [@oksanen2019]. We
tested a difference in richness by both passage and treatment with a
Kruskal-Wallace test followed by a pairwise Wilcoxon test. Next, we estimated
beta-diversity as the Bray-Curtis dissimilarity by averaging 100 random subsets
with a subsample size of `r min_seqs` using the 'avgdist' function in 'vegan'
[@bray1957; @oksanen2019].  We tested a difference in centroid and dispersion
of beta diversity by passage and treatment using a permutational analysis of
variance (PERMANOVA) with 999 permutations using the 'adonis2' function from
'vegan' and tested a difference of group dispersions using 'betadisper' and
'anova' with 999 permutations [@mcardle2001; @oksanen2019].  Lastly, we tested
the correlation between CH~4~ oxidation rate and Bray-Curtis dissimilarity in
Passage 5 with a distance-based redundancy analysis (dbRDA) using the 'dbrda'
function in 'vegan' and estimated the p-value using a permutation F-test with
999 permutations [@mcardle2001; @oksanen2019]

To identify taxa that responded to selection on CH~4~ oxidation rate, we tested
differential abundance between the two treatments in passage 5. We first
grouped ASVs at the family level keeping ASVs that lacked a family-level taxonomic
assignment by grouping them at a higher level. We
then subset the samples in Passage 5 and removed all families with a prevalence
of less than 10\% in either treatment. We used three methods for testing differential abundance: ANCOM-II, ALDEx2, and CORNCOB. We then
identified the consensus taxa that were significant
with all three tests and plotted their relative abundances. For ANCOM-II, we used the 'ancom' function in the 'ANCOM-BC' package with a cutoff of W = 0.7 [@kaul2017; @lin2020]. For ALDEx2, we used the 'aldex' function in the 'ALDEx2' package with Welch's t-test and we used an effect size of
1 as our significance threshold [@fernandes2014]. Finally, we used CORNCOB with the 'differentialTest' function in the 'corncob' package with the Wald test and without bootstrapping [@martin2020]. Lastly, to test differentially abundant methanotrophs, we subset all ASVs within methanotrophic families and tested their differential abundance aggregated at the family and genus level using 'corncob'.

# Results

## Response to selection on methane oxidation rate

We observed a response to selection on whole-ecosystem soil CH~4~ oxidation
rate (`r figures("response", display = "cite")`; difference of slopes:
F~`r compare_response_mods[2,4]`,`r compare_response_mods[2,2]`~ = `r myround(compare_response_mods[2,5], 2)`, p = `r myround(compare_response_mods[2,6], 2)`). At the start of the experiment, the Positive
treatment had a mean CH~4~ oxidation rate that was
`r percent_less(initial = tidy(response_model)[[1, 2]], final = tidy(response_model)[[3, 2]] + tidy(response_model)[[1, 2]])`\% lower than the
Neutral treatment `r print_lm(response_model, 3, 'difference of y-intercepts')`. 
There was no change in CH~4~ oxidation rate of the Neutral
treatment over the five passages `r print_lm(response_model, 2)`. By contrast,
the Positive treatment had a `r round(per_slope)`\% increase in CH~4~
oxidation rate per passage `r print_lm(response_model, 4)`.

We estimated the proportion of variation in methane oxidation due to variation
in the microbiome as the regression of offspring on mid-parent 
(`r figures("herit", display = "cite")`).  Offspring CH~4~ oxidation rates were
correlated with parental CH~4~ oxidation rates in both the Positive treatment
`r print_lm(herit_model, 4)` and the Neutral treatment `r print_lm(herit_model,
2)`. 

## Taxonomic richness

Median ASV richness decreased from `r myround(rare_median[1, 2], 1)` (`r myround(rare_median[1, 3], 1)`) in Passage 2 to `r myround(rare_median[2, 2], 1)` (`r myround(rare_median[2, 3], 1)`) in Passage 5 (Kruskal-Wallace test: $\chi^2$ = 
`r myround(kt[1,1], 1)`, df = `r kt[1,3]`, p < 0.001; pairwise Wilcoxon test: p < 0.001). However, there was no
difference in richness between the Positive and the Neutral treatment in
Passage 2 or 5 (pairwise Wilcoxon test: p = `r myround(wt[1, 3], 1)`).  In addition, there was no
correlation between richness and CH~4~ oxidation rate across the two treatments in Passage 5 (Spearman's rho = `r myround(s5[1,1], 1)`, p = `r myround(s5[1,3], 1)`).

## Community dissimilarity

Bray-Curtis dissimilarity of the soil microbiome varied strongly by passage and
weakly by treatment with an interaction between passage and treatment (`r figures("beta", display = "cite")`). 
Passage explained `r myround(100*tidy_permanova[1, 4])`\% of the
variation in Bray-Curtis dissimilarity (F~`r tidy_permanova[1, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[1, 5], 1)`, p = `r myround(tidy_permanova[1, 6], 3)`), treatment explained 
`r myround(100*tidy_permanova[2, 4])`\%
of the variation (F~`r tidy_permanova[2, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[2, 5], 1)`, p = `r myround(tidy_permanova[2, 6], 3)`), and the interaction
between treatment and passage explained 
`r myround(100*tidy_permanova[3, 4])`\% of the variation (F~`r tidy_permanova[3, 2]`,`r tidy_permanova[4, 2]`~ = `r myround(tidy_permanova[3, 5], 1)`, p = `r myround(tidy_permanova[3, 6], 3)`). There was 
no difference in dispersion between treatments or passages (F~3,44~ = 0.91, p = 0.45). 
Finally, CH~4~ oxidation rate was correlated with Bray-Curtis dissimilarity across both treatments in Passage 5 and explained `r myround(dbrda_percent_explained[1], 1)`\% of the variation in Bray-Curtis dissimilarity (dbRDA: F~`r dbrda_aov[1, 1]`,`r dbrda_aov[2, 1]`~ = `r myround(dbrda_aov[1, 3], 2)`, p = `r myround(dbrda_aov[1, 4], 3)`)

## Taxa that responded to selection

To identify taxa that responded
to selection on soil CH~4~ oxidation rate, we tested the differential relative
abundance of families in the Positive jars relative to the Neutral jars within
Passage 5 using three methods and then plotted the taxa identified by all three methods. We identified `r length(da_fams)`
families that were enriched or depleted in the Positive treatment relative to
the Neutral treatment (`r figures("da", display = "cite")`). 

Overall, none of the families enriched in the Positive selection treatment
contain known methanotrophs. Several taxa identified had a 
higher taxonomic designation that contains methanotrophs, for example, the
Gammaproteobacteria class had a large effect size.
The Gammaproteobacteria include the type I and type X methanotrophs in the families *Methylococcaceae* and *Methylothermaceae* [@stein2012].
However, the Gammaproteobacteria is among the most diverse 
groups in the Prokaryotes, so this is not strong evidence for a selection response
by methanotrophs [@garrity2005].
In addition, the *Puniceicoccaceae* is a member of the phylum Verrucomicrobia. The Verrucomicrobia is a diverse group that contain
known methanotrophs as well as ammonia-oxidizing bacteria [@freitag2003]. Other than these two groups, none of the other taxa enriched in the Positive treatment are known to be related to methanotrophs. Two groups in the Armatimonadales were enriched in the Positive treatment including the family *Fimbriimonadaceae* and an unclassified ASV from the order Armatimonadales [@im2012].
*Cytophagaceae* was also enriched in
the Positive treatment and contains a number of mainly aerobic heterotrophs
that can digest a variety of macromolecules [@mcbride2014]. 
The remaining families include the uncultured family 0319-6G20, *Diplorickettsiaceae*, *Rhodospirillaceae*, and an unclassified Kapabacteriales.

We did not identify any methanotrophic families in the overall differential abundance analysis. 
However, we wanted to look more closely at the known methanotrophs in our dataset to be sure they did not have an effect.
To do this, we subset all of the ASVs in our dataset that were in families that contained methanotrophs.
Only two families were represented: *Methylacidiphilaceae* and *Beijerinckiaceae*. 
Aggregating reads at the family level, neither family was differentially abundant between the two treatments.
However, aggregated at the genus level, a group of unclassified genera in the *Beijerinckiaceae* were depleted in the Positive treatment and the genus
*Rhodoblastus*, a member of the *Beijerinckiaceae*, was enriched in the
Positive treatment.  While many *Beijerinckiaceae* are methanotrophs, several
taxa in this family have lost the the ability to oxidize CH~4~ and it appears
that *Rhodoblastus* species are not able to grow on CH~4~, though they can grow
on methanol [@tamas2014, @dedysh2016]. Based on this analysis,
it appears that no methanotrophs were enriched in the Positive treatment.


# Discussion

While it is well understood that the microbiome mediates a variety of ecosystem
functions, a fundamental question in microbial ecology is whether we need to
understand variation in the composition of microbiomes to understand variation
in the rate of ecosystem functions [@conrad1996; @schimel1998]. To address this question, we performed
artificial ecosystem selection on soil microbiomes by selecting for ecosystems
with high methane oxidation rates. We found that there was a significant
increase in methane oxidation rate as a response to selection. Each generation
of soil ecosystems was created from the same pool of sterile soil substrate and
a small inoculum from the previous generation. Therefore, it is unlikely that
this response to selection was due to changes in the abiotic conditions of the
ecosystem. In addition, the response was only observed in the Positive
treatment and not in the Neutral treatment, which shows that this was not a
side effect of passaging in jars.  Therefore, we conclude that variation in the
microbiome can generate variation in ecosystem functions independent of the
environment.

In quantitative genetics, heritability quantifies the degree to which variation
in a trait is due to variation in genotype [@falconer1996]. This is often estimated as the
regression of offspring phenotype on parental phenotype. Thus, a significant
correlation between the trait values of parents and offspring indicates that
the trait is at least partially genetically determined. To estimate the
variation in an ecosystem trait due to variation in the microbiome, we can
perform a similar estimation by comparing the trait values of the "offspring"
jars in one generation to the trait value of the "parental" jars chosen for
selection in the previous generation. This "ecosystem heritability" then
quantifies how much of the variation in ecosystem function is due to variation
in the microbiome [@goodnight2000; @sanchez2021; @henry2021]. 

We quantified how much of the variation in methane oxidation
rate was due to variation in the microbiome by regressing the methane oxidation
rates of the 12 offspring jars against the mean methane oxidation rate of the
three jars selected in the previous generation 
We observed a heritability of `r myround(coefficients(herit_model)[4], 2)` (add SE/CI) for the
Positive selection treatment. 
A heritability of 1 would indicate that a trait is entirely
genetically determined and thus offspring would have the exact same value of
that trait as their parents. A heritability greater than 1 indicates that
offspring exceed the value of their parents. This is unlikely in the context of
a genomic trait, but microbiomes are not limited by the constraints of genomes
because microbes are not alleles and communities are not chromosomes.
Therefore, complementarity within a microbial community could result in
offspring communities that exceed the trait values of their "parent"
ecosystems. The large heritability we observed indicates that microbiome
variation is a large contributor to variation in methane oxidation in this
experimental system.

Next, we wanted to address what aspects of the microbiome, in terms of alpha
diversity, beta diversity, or individual taxa, explain this contribution of the
microbiome to variation in methane oxidation rate. There are three ways that
the microbiome could respond to selection in this experiment: gain or loss of
species, changes in the relative abundances of species, or changes within the
genomes of the constituent species. Using 16S rRNA gene sequences, we are
unable to address the third possibility of whether taxa in this experiment
could have evolved changes in their genomes as a result of selection.
Therefore, we will focus on the first two possibilities. 

Richness at the ASV level did not vary between the two treatments and there
were relatively few taxa gained or lost in the Positive selection treatment and
none of these were prevalent across the 12 jars in passage 5. Therefore, the
gain or loss of species is unlikely to explain the increase in methane
oxidation rate. However, we found that Bray-Curtis dissimilarity was greater
between the two treatments in Passage 5 than within each treatment and was
correlated with methane oxidation rate, which suggests that changes in the
relative abundance of taxa could explain the response to selection.

Even though we observed an increase in methane oxidation rate in the Positive
treatment and a difference in composition between the two treatments, we did
not observe an increase in the relative abundance of methanotrophs. This was
surprising given that methane consumption is not a common trait among microbes
and that it is often assumed that the rate of an ecosystem function is
limited by the final enzymatic step in a metabolic pathway
[@rocca2015]. In certain ecosystems, methane production and consumption are 
correlated with the abundance of methanogens and methanotrophs as estimated from marker
genes [@meyer2020, @meyer2020a]. However, it does not appear generally true that the rate of an
ecosystem function is limited by the abundance of the gene that encodes the
final step in that pathway [@rocca2015]. Our results suggest that the rate of methane oxidation can be altered by
non-methanotrophs, perhaps through ecological interactions with methanotrophic
species. 

Another important question is whether selection at the ecosystem scale resulted
in within-strain evolution. Using 16S rRNA gene sequencing we are unable to
resolve genomic changes within taxa that may have undergone evolution. 
Therefore, it is possible that the response to selection was driven by changes 
within the genomes of methanotrophs. But given
the slow growth rate of methanotrophs, often requiring weeks to years of
serial batch culturing for isolation, this seems like an unlikely explanation 
[@whittenbury1970; @hirayama2014; @kim2018]. Future
work could analyze an artificial ecosystem selection experiment using
genome-resolved metagenomics to try to detect strain-level changes in the
constituent taxa. However, reconstructing genomes from in situ soil samples
has been difficult to achieve and may require enrichment techniques, particularly for methanotrophs which are a relatively rare component of the soil microbiome [@delmont2015]. 
If these challenges could be overcome for soil metagenomics, we could begin to evaluate whether ecosystem selection results in strain-level evolution or enriches for certain metabolic pathways within the microbiome.

As more is understood about the role of the microbiome in regulating ecosystem
functions and the specific taxa involved, more detailed experiments could
selectively add or remove important taxa or manipulate their relative abundance
to start to estimate the quantitative relationship between specific taxa and
the rate of an ecosystem function. Both our differential abundance analysis and
this reductionist approach to manipulating individual taxa assumes that taxa
act individually on ecosystem function. However, it is much more likely that
the microbiome affects the rate of ecosystem function through the interactions
of multiple taxa. We are unable to evaluate this possibility given our small 
sample size, but future analyses with greater replication should test the
role of multiple taxa simultaneously by, for example, using machine learning
methods [@topcuoglu2020].

A fundamental question in microbial ecology is whether variation in the
microbiome contributes to variation in ecosystem function. We performed
artificial ecosystem selection on soil methane oxidation rate to test the
relationship between microbiome variation and soil methane oxidation rate. We
observed a response to selection on soil methane oxidation rate and found a
high heritability for methane oxidation, which suggests that variation in the
microbiome contributes to variation in soil methane oxidation rate.
Surprisingly, we did not observe an increase in the relative of abundance of
methanotrophs as a response to selection. This suggests that the rate of methane 
oxidation could be altered by
non-methanotrophs. Future research could estimate
heritability under greater environmental variation to understand how much of
the variation in methane oxidation rate is due to variation in the microbiome
in a natural ecosystem. In addition, future selection experiments could use
genome-resolved metagenomics to observe within-strain evolution and test
changes in the functional pathways of the microbiome as a response to selection
to determine what aspects of the microbiome explain this increase in oxidation
rate. This study demonstrates that soil methane oxidation rate can vary greatly
with variation in the microbiome and opens up future research opportunities to
establish a more detailed understanding of the relationship between the
microbiome and ecosystem function.

# Acknowledgments

This project was supported by the National Science Foundation Graduate Research
Fellowship Program (grant no. DGE 1255832) and the ARCS Foundation Florence and
Mike Nudelman Scholarship.

# Competing Interests

We declare we have no competing interests.

# Data Availability Statement

The 16S rRNA sequencing data generated during the current study are available
in the NCBI Sequence Read Archive (SRA) under BioProject accession number
PRJNA832314, https://www.ncbi.nlm.nih.gov/sra/PRJNA832314. The metadata
generated during the current study as well as the scripts to recreate the
analysis are available on Github,
https://github.com/amorris28/artificial_ecosystem_selection.

# Supplement


# References

::: {#refs}
:::

# Figures


```{r response, fig.width=4, fig.height=3}

p_response

```

`r figures("response", "Response to selection on soil CH~4~ oxidation rate fit by ordinary least squares regression. The y-axis is the log~10~ of the additive inverse of the first-order exponential decay constant $k$ (i.e., $-k$) with units day^-1^. The x-axis is the passage number. The orange line and points are the Positive treatment and the gray line and points are the Neutral treatment.")`


```{r herit, fig.width=5, fig.height=3}

p_herit_each

```

`r figures("herit", "Ordinary least squares regression of offspring CH~4~ flux on mid-parent CH~4~ flux ($-k$ with units day^-1^). Mean parental is the mean of the jars selected to inoculate the next passage. Offspring is the oxidation rate of each jar in the next passage. The orange lines and points are the Positive treatment and the gray lines and points are the Neutral treatment")`

```{r beta, results=FALSE, fig.height=2.5}

p_beta

```

`r figures("beta", "Principal component analysis plot of beta diversity for all jars. Dissimilarities are based on rarefied Bray-Curtis dissimilarity. Colors represent treatment and shapes represent passage number.")`

```{r da, results=FALSE}

p_log10

```

`r figures("da", "Differentially abundant family-level taxa identified by ANCOM-II, ALDEx2, and CORNCOB. Values on the x-axis are relative abundances on a log~10~ scale. Points are colored by treatment. Large points with error bars are the median relative abundance and interquartile range. Taxa in the top panel are enriched in the Positive treatment and taxa in the bottom panel are depleted in the Positive treatment. Taxa are sorted by their effect size with taxa at the top having the largest positive effect size and taxa at the bottom with the largest negative effect size.")`

