New Outline 7/16/20

What is the nature of the map?

Why use AES rather than isolation, culturing, and AS?

We already know the genetic basis of methane oxidation on an individual level from culturing. The genetic marker for methane oxidizers is the gene that encodes the enzyme particulate methane monooxygenase subunit A (pmoA), which is shared among most cultured methanotrophs (original pmoA papers). However, knowledge of pmoA abundance has done little to elucidate patterns of methane oxidation in natural ecosystems (Meyer at al. 2020, Rocca et al. 2016, other stuff). This suggests that pmoA is insufficient for understanding the rate of ecosystem-scale methane oxidation. Group selection theory posits that selection above the individual level is plausible in situations where interactions between species are necessary as in nutrient cycling (did they say this or did I just make this up?). Nutrient cycling critically depends on relationships between producers and consumers and so it may be insufficient to understand the effect of the individual without understanding the effect of the group. For example, methane oxidation in soil is dependent not only on the presence and efficiency of a particular methane oxidizer, but is also dependent on methane producers and the release of nutrients from other processes. Therefore, methane flux at the ecosystem scale is dependent on the genetic effect of multiple taxa. Genes such as pmoA could be said to exhibit a form of community epistasis in which the effect of pmoA is dependent on the genetic background of the whole community. Therefore, imposing selection at the level of the ecosystem allows us to identify the optimal team of microorganisms to perform this process. In the short term, the response to selection should be due to sorting through many combinations of clonal strains. Over the longer term, the response to selection may be due to new mutations generating new strains.

It should be noted that though we use the language of “heritability” and “response to selection,” we do not mean to imply anything about evolution of this trait in nature. We also do not suggest that the results of our experiment demonstrate “adaptation” at the community level in the evolutionary sense. This would depend on defining fitness at the whole community or ecosystem scale. The selection process that we outline could be considered more analogous to the ecological process of community assembly. Differential extinction of ecosystems in our experiment represents determinstic fitness differences among species. Ecological drift contributes through stochastic changes in relative abundance within a generation. Dispersal could be considered non-limiting in our experiment as the migrant pool homogenization process provides the opportunity for all taxa to disperse to all jars. The propagule pool approach, by contrast, would potentially result in alternative assembly paths not limited by competitive dominance. Speciation plays a role as long as mutations take place over the course of the experiment.

Experimental Evolution vs. Artificial Selection

Taxonomic biodiversity vs. functional group biodiversity


What is the mapping between microbial community structure and ecosystem function?

Is it simple? e.g., abundance of functional group, diversity of functional group (like in plants). Or is it complex? e.g., multiple functional groups, multiple taxa.

How does it compare to macroogranisms? Have people even tested functional group diversity extensively? Is methanotrophy as deeply conserved in bacteria as photosynthesis in plants? i.e., is there a 16S sequence or other marker that maps to the methanotrophy guild? It seems like microbial BEF has focused on Prokaryote diversity or functional group abundance, which are both different from how it’s been carried out in plants.

Does BEF theory apply? Niche theory: One limiting R -> one final competitor. Multiple Rs -> multiple peristent competitors.

I expect artificial ecosystem selection to follow a similar trajectory for short- and long-term selection as in populations. In particular, short-term selection should lead to species sorting (i.e., sorting of clonal individuals, akin to alleles segregating) whereas long-term selection would be due to mutations creating novels strains.

Why use AES and not isolation, culturing, and AS?

I chose methane oxidation, because multi-level selection seems most plausible in situations where interaction sbetween species are necessary.

Introduction
Biodiversity-ecosystem function (BEF) research has the central goal of understanding whether and how biodiversity - the diversity and relative abundance of species in a community - determines the magnitude of ecosystem functions (cite Hooper, Naeem, Tilman, …). Ecosystem functions are a broad range of processes, including pollination, water filtration, carbon storage, and nutrient cycling, that are driven by both abiotic and biotic factors. The ultimate goal of this research is to predict how changes in biodiversity due to climate change or land use change will alter ecosystem function and, in turn, understand how we should protect biodiversity in order to preserve essential ecosystem functions. The fundamental question for this research is what is the nature of the “map” between elements of biodiversity and elements of ecosystem function?

One could hypothesize a relatively simple map in which a single taxon determines the rate of an ecosystem function based entirely on its presence or abundance. (e.g. of this?) Alternatively, instead of defining biodiversity based on a taxonomic unit, biodiversity could be defined by functional groups or guilds. For example, photosynthesis could be determined by the abundace of photosynthesizers in a community. (Example of abundance of photosynthesizers? measured as the Abundance of rubisco genes?). A somewhat more complex map could define biodiversity based on the diversity (i.e., richness or number of species) of photosynthesizers. Using this mapping, an abundance of literature has demonstrated that productivity increases asymptotically and variation in productivity decreases asymptotically with the richness of producers (Tilman 1997).

When microbial ecologists have attempted to define the mapping of ecosystem functions performed by microorganisms, these simple models rarely hold (Rocca et al., Graham ete al.). Most functions in most ecosystems do not exhibit a positive relationship between the abundance of functional marker genes and the rate of the associated function. Few studies test the diversity of functional groups. This could mean that the map is simple, but we are not accurately estimating the abundance of these functional groups. Alternatively, the mapping between microbial community structure and microbial processes could be more complex than the mapping between macrobial structure and ecosystem function.

To investigate the nature of the map between microbial community structure and ecosystem function, we will perform artificial ecosystem selection (sensu Swenson et al. 2000). With this approach, we will collect a single soil which we subsamble to generate multiple ecosystems. These ecosystems vary in both community structure and the rate of ecosystem function. From this variation, we select the top performing ecosystems, reshuffle their members through homogenization, and subsample this homogenate to generate more ecosystems. Through this iterative process, we test many different combinations of taxa and enrich for the top performers. This process of selection also reduces the genetic variation and diversity of the communities. By minimizing the environmental differences between jars and between passages, we can insure that the majority of functional variation from jar to jar and passage to passage is due to genetic variation in the community. As a result, genetic differences between the initial community and the selected community should be associated with the function of interest.

We hypothesize that the positive selection treatment will increase in methane oxidation rate while the neutral selection treatment will resemble a random walk over time demonstrating community heritability within our system. We also hypothesize that genetic changes over time will follow a predictable trajectory: the fastest changes in community copmosition will be due to species loss, slower changes will be due to changes in the relative abundance of species, and the slowest changes will be due to intraspecies selection (i.e., evolution). The soil communities will not only be adapting in response to selection on soil methane oxidation, but will also be adapting to survival and passaging between jars.

There are several alternatives as to the complexity of the “map” between microbes and methane. At the simplest level, methane oxidation rate could be a function of the abundance of methane oxidizers. A somewhat more complex map would be if methane flux is a function of the diversity of methane oxidizers as in other systems (Hooper et al.). If neither of these simple models hold, the mapping might be a more complex relationship between a broader level of organization, such as methane producers and consumers, a broader metabolic network, or a network of bacterial and archaeal interactions that’s hard to predict.

The gold standard of molecular biology gene mapping studies is to not only identify a correlation, but also to confirm the association through knockout and overexpression studies. To approximate this approach for microbial ecology, we cultured the important strain(s) in pure culture and applied them to the neutral soil community to see if we could recapitulate the trait level for the selected treatment with only 1 or a few taxa or if it required the whole community. We used the neutral treatment, because these communities are already bottle-adapted and offer a more similar environment for the selected strains than a natural soil would.

Our goal is to understand the identity of the community members and their interactions that result in the greatest level of ecosystem function. Artificial ecosystem selection allows us to test many “teams” or communities of microbes in terms of their combined impact on function. By performing selection, we reduce the genetic variation in our natural community to the variation that is sufficient to perform a high rate of ecosystem function. Then, we can investigate the remaining variation to generate hypotheses about the relationship between community composition and ecosystem function, which mebmers are the best performers, and how simple or complex the map is between community and funtion. Finally, assuming we can culture these members, we can test those hypotheses through enrichment of one or a group of taxa to see which members are sufficient to recreate the highest rate of ecosystem function.

Questions:

What is the “map”? And how do we get to that map? What is the nature of the map? Taxonomy? Or something other than taxonomy? Simple – taxon. Diffuse – something other than taxonomy. Complicated interactions among organisms leads to “hitchhikers.”

Can we isolate 1 or a few strains and add them to the N community? Or does it take a whole complex community? The

Lewontin posited that higher levels of selection (species, community, ecosystem) would have weaker heritability. Do we observe this?

Does soil methane oxidation rate respond to selection at the ecosystem level?

Does a community that has been selected on for high methane oxidation rate differ in diversity or composition from a community that has been passaged without selection for methane oxidation rate?

What aspects of community composition differ between the positive and neutral lines? Changes in alpha diversity, beta diversity? Presence/absence of specific microbial taxa? Relative abundance of microbial taxa?

Does selection lead to greater differences in the bacterial/archaeal community (inferred from 16S sequences) or in the methanotroph community (inferred from pmoA sequences)?

Does a community with high methane oxidation rate dominate coalescence with a natural community compared to a community with average methane oxidation rate?

Many-to-one mapping

Crossing/mixing lines or competing lines or applying to natural soils for the application of the tool.

Hypotheses:

Soil methane oxidation rate responds to selection at the ecosystem level

Questions to address in the Intro:

Is this “just enrichment” or is it community selection? i.e., is it dilution to extinction?

How do people enrich for methanotrophs?

Community-level selection? Paul Rainey, Martin Polz