---
title: "Community Analysis"
author: "Andrew Morris"
date: "9/15/2021"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = TRUE)
knitr::opts_knit$set(root.dir=normalizePath('../'))

```

```{r initialize_packages}
library(tidyverse)
library(phyloseq)
library(decontam)
library(microViz)
library(knitr)
library(corncob)
library(DivNet)
library(magrittr)
library(vegan)
library(ggpubr)
library(ggthemes)
library(broom)


source('R/functions.R')

```


```{r import-data}
load(file = "Output/phyloseq.RData")
ps <-
  ps %>% 
  ps_mutate(pas_by_treat = paste0(passage, treat))
ps_melt <-
  ps %>% 
  ps_melt()
```

# Exploratory

```{r plot-depth}

depth_data <-
  ps %>% 
  ps_melt %>% 
  group_by(Sample, passage, treat) %>% 
  summarize(depth = sum(Abundance))

summary(aov(depth ~ passage * treat, data = depth_data))
depth_data %>% 
  ggplot(aes(x = interaction(treat, passage), y = depth)) +
  theme_tufte() +
  geom_rangeframe() +
  # coord_cartesian(clip="off") +
  geom_tufteboxplot()

```

There is no systematic difference in sequencing depth/library size between treatments. It does appear that sequencing depth decreases between passage 2 and 5.


```{r plot-richness}

plot_richness(ps, x = "pas_by_treat", color="treat", measures=c("Observed", "Shannon", "Simpson"), title = "Alpha Diversity ASV") +
    scale_color_manual(name = "Treatment", 
                         labels = c('Neutral', 'Positive'),
                         values = c('gray40', 'darkorange2'))

```

Richness decreases with passage, but seems unchanged between 5n and 5p.
Initial richness appears similar between the treatments.

# Hypothesis Testing

## Richness

```{r richness}

# Test effect of passage number on richness using all samples
richness_soil <- ps %>% breakaway
plot(richness_soil, physeq=ps, color="passage")
summary(richness_soil) %>% as_tibble

meta <- ps %>%
  sample_data %>%
  as_tibble %>%
  mutate("sample_names" = ps %>% sample_names )

combined_richness <- meta %>%
  left_join(summary(richness_soil),
            by = "sample_names")

bt_passage <- betta(chats = combined_richness$estimate,
                      ses = combined_richness$error,
                      X = model.matrix(~passage, data = combined_richness))
bt_passage$table

# Test effect of treatment on richness in passage 5

ps_5 <- ps_filter(ps, passage == "5")
richness_soil <- ps_5 %>% breakaway
plot(richness_soil, physeq=ps_5, color="treat")
summary(richness_soil) %>% as_tibble

meta <- ps_5 %>%
  sample_data %>%
  as_tibble %>%
  mutate("sample_names" = ps_5 %>% sample_names )

combined_richness <- meta %>%
  left_join(summary(richness_soil),
            by = "sample_names")

bt_treat5 <- betta(chats = combined_richness$estimate,
                      ses = combined_richness$error,
                      X = model.matrix(~treat, data = combined_richness))
bt_treat5$table

# Test effect of CH4 on richness

bt_ch4 <- betta(chats = combined_richness$estimate,
                      ses = combined_richness$error,
                      X = model.matrix(~ch4, data = combined_richness))
bt_ch4$table


save(bt_passage, bt_treat5, bt_ch4, file = 'Output/richness_models.RData')

```

Test confirms the above observation: Richness decreases between Passage 2
and Passage 5. No difference in richness by treatment in passage 5 or by
methane flux.


# Beta Diversity


```{r}

# Replace unknown and missing taxa
ps_fixed <-
ps %>%
 tax_fix(
  min_length = 4,
  unknowns = c(""),
  sep = " ", anon_unique = TRUE,
  suffix_rank = "classified"
 )

# Recode treatments as binary 1/0
ps_fixed <- 
  ps_fixed %>% 
  ps_mutate(treat_bin = factor(ifelse(treat=='p', 1,0))) %>% 
  ps_mutate(passage_bin = factor(ifelse(passage==5, 1,0))) %>% 
  ps_mutate(treat_pas = interaction(treat_bin, passage_bin))

ps_fixed
```


```{r calc-beta, results=FALSE}

ord_calc_aitch <-
  ps_fixed %>% 
  tax_transform(trans = 'clr') %>% 
  ord_calc(method = "PCA")


ord_calc_bray <-
  ps_fixed %>%
  dist_calc("bray") %>%
  ord_calc(method = "NMDS")

ord_calc_jacc <-
  ps_fixed %>%
  dist_calc("jaccard", binary = TRUE) %>%
  ord_calc(method = "PCoA")

```

## Bray-Curtis and Jaccard

```{r plot-beta, results=FALSE, fig.width = 8, fig.height = 3}
bray <-
  ord_calc_bray %>% 
  ord_plot(color = "treat") +
  theme_tufte() +
  geom_rangeframe() +
  coord_cartesian(clip="off") +
  scale_color_manual(name = "Treatment", 
                       labels = c('Neutral', 'Positive'),
                       values = c('gray40', 'darkorange2')) +
  labs(title = "Bray-Curtis")

jacc <- 
  ord_calc_jacc %>% 
  ord_plot(color = "treat") +
  theme_tufte() +
  geom_rangeframe() +
  coord_fixed(3.9/48.3) +
  # coord_cartesian(clip="off") +
  scale_color_manual(name = "Treatment", 
                       labels = c('Neutral', 'Positive'),
                       values = c('gray40', 'darkorange2')) +
  labs(title = "Jaccard")

                       
ggarrange(bray, jacc)

```

## Aitchison

```{r}
plot_beta(ps_fixed)
```


## Beta diversity test passage by treatment

```{r beta-test}

ps_clr <- microbiome::transform(ps, "clr")

fit_bin_jacc <- adonis2(otu_table(ps) ~ passage * treat, data = as(sample_data(ps), 'data.frame'), method = "jaccard", binary = TRUE)

fit_aitch <- adonis2(otu_table(ps_clr) ~ passage * treat, data = as(sample_data(ps_clr), 'data.frame'), method = "euclidean")

fit_adonis <- rbind(tidy(fit_aitch), tidy(fit_bin_jacc))
fit_adonis$term <- c('Passage', 'Treatment', 'Pass:Treat', 'Residual', 'Total',
      'Passage', 'Treatment', 'Pass:Treat', 'Residual', 'Total')
save(fit_adonis, file = 'Output/beta_model.RData')

############

clr_dist_matrix <- phyloseq::distance(ps_clr, method = "euclidean") 

dispr <- vegan::betadisper(clr_dist_matrix, phyloseq::sample_data(ps_clr)$passage)
disper_test <- permutest(dispr)

dispr <- vegan::betadisper(clr_dist_matrix, phyloseq::sample_data(ps_clr)$treat)
disper_test <- permutest(dispr)

dist_aitch5 <-
  ps %>% 
  ps_filter(passage == 5) %>% 
  dist_calc(dist = "aitchison")
dist_permanova(dist_aitch5, "treat")
groups <- factor(c(rep(1,12), rep(2,12)), labels = c("N","P"))

dist_aitch5 %>% 
  dist_get %>% 
  vegan::betadisper(groups) %>% 
  permutest()

dist_aitch2 <-
  ps %>% 
  ps_filter(passage == 2) %>% 
  dist_calc(dist = "aitchison")
dist_permanova(dist_aitch5, "treat")
groups <- factor(c(rep(1,12), rep(2,10)), labels = c("N","P"))

dist_aitch2 %>% 
  dist_get %>% 
  vegan::betadisper(groups) %>% 
  permutest()

```


There is a significant main effect and interaction of passage and treatment on
taxonomic dissimilarity of the soil microbiome. In addition, there is no difference
in dispersion between treatments, but there is between passage2 and 5.

# Differential Abundance


First, I looked at which taxa are unique to treatments N and P. Several families
present in P and absent in N. Most of them had low prevalance (present in 3 or
fewer samples) and low abundance (median < 5 reads across samples). There were two families unique
to the Positive treatment with relatively high prevalence. This included an ASV
that was a member of the Bacteroidia Class with no lower taxonomic designation. 
This ASV had a prevalence of 10/12 and a median abundance of ~2 reads. The other
prevalent family was a member of the Silvanigrellaceae, a newly described family placed in
its own order. This family was present in
all 12 samples and was represented by three OTUs: one with no lower designation, one from the genus
Silvanigrella and one from the genus Candidatus Turbobacter. Silvanigrella was
isolated from a temperate fresh water lake (https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.001965).

Of the families unique to the Neutral treatment, only one had a prevalence greater than 2/12. 
This family, Armatimonadaceae, was present in half of the Neutral samples (6/12). This
family was represented by two ASVs in the genus Armatimonas. The type strain for
Armatimonas was isolated from the rhizoplane of Phragmites australis (https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijs.0.025643-0)


```{r}
# families in to Passage 5, Treatment P
g5p<- 
  ps_fixed %>% 
  ps_filter(treat == "p", passage == 5) %>% 
  tax_agg('Family') %>% 
  ps_get() %>% 
  ps_melt()

# families in to Passage 5, Treatment N
g5n<- 
  ps_fixed %>% 
  ps_filter(treat == "n", passage == 5) %>% 
  tax_agg('Family') %>% 
  ps_get() %>% 
  ps_melt()

# Families unique to p in passage 5
g5p_unique <-
  g5p %>% 
  filter(Family %in% unique(g5p$Family)[!(unique(g5p$Family) %in% unique(g5n$Family))]) %>% 
  mutate(Prevalence = as.numeric(Abundance > 0))

# Abundance
Family_order<-
  g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(median = median(Abundance), max = max(Abundance)) %>% 
  arrange(median, max)
g5p_unique$OTU <- factor(g5p_unique$OTU, levels = Family_order$OTU)

# Plot families unique to P5 in order of median abundance
g5p_unique %>% 
  ggplot(aes(y = Abundance, x = OTU)) +
  geom_tufteboxplot() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(0, 5, 10, 15, 20, 24)) +
  labs(title = "Unique to P5 by abundance")


# Prevalence
Family_order<-
  g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(sum = sum(Prevalence)) %>% 
  arrange(sum)
g5p_unique$OTU <- factor(g5p_unique$OTU, levels = Family_order$OTU)

# Plot families unique to P5 in order of prevalence
g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(Prevalence = sum(Prevalence)) %>% 
  ggplot(aes(y = Prevalence, x = OTU)) +
  geom_point() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(1, 2, 3, 5, 10, 12)) +
  labs(title = "Unique to P5 by prevalence")

# Families unique to n in passage 5
g5n_unique <-
  g5n %>% 
  filter(Family %in% unique(g5n$Family)[!(unique(g5n$Family) %in% unique(g5p$Family))]) %>% 
  mutate(Prevalence = as.numeric(Abundance > 0))

# Abundance
Family_order<-
  g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(median = median(Abundance), max = max(Abundance)) %>% 
  arrange(median, max)
g5n_unique$OTU <- factor(g5n_unique$OTU, levels = Family_order$OTU)

# Plot families unique to N5 in order of abundance
g5n_unique %>% 
  ggplot(aes(y = Abundance, x = OTU)) +
  geom_tufteboxplot() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(0, 10, 20, 30, 37)) +
  labs(title = "Unique to N5 by abundance")


# Prevalence
Family_order<-
  g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(sum = sum(Prevalence)) %>% 
  arrange(sum)
g5n_unique$OTU <- factor(g5n_unique$OTU, levels = Family_order$OTU)

# Plot families unique to N5 in order of prevalence

g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(Prevalence = sum(Prevalence)) %>% 
  ggplot(aes(y = Prevalence, x = OTU)) +
  geom_point() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(1, 2, 6)) +
  labs(title = "Unique to N5 by prevalence")

```



## Differential Abundance Treatment in passage 5


```{r corncob_treatment, results=FALSE, fig.hight=40}
ps_fixed_family <- speedyseq::tax_glom(ps_fixed, taxrank="Family")

ps_fam_pas5 <- microViz::ps_filter(ps_fixed_family, passage == 5)

ps_fam_pas5_prev <- microViz::tax_filter(ps_fam_pas5, min_prevalence = 0.10)

sample_data(ps_fam_pas5_prev) =
  sample_data(ps_fam_pas5_prev)[,c('sample_id', 'treat')] %>% 
  sample_data()
ps = ps_fam_pas5_prev

# Differential abundance test
da_analysis <- differentialTest(formula = ~ treat,
                                 phi.formula = ~ 1,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ 1,
                                 test = "Wald", boot = FALSE,
                                 data = ps_fam_pas5_prev,
                                 fdr_cutoff = 0.05, B = 50)

#plot(da_analysis, level = c("Family"))
da_analysis$significant_models
dt_data <- plot(da_analysis, level = c("Family"), data_only = TRUE)

sig_taxa_treat <- tax_select(ps_fam_pas5_prev, da_analysis$significant_taxa)
ntaxa(sig_taxa_treat)
save(da_analysis, sig_taxa_treat, dt_data, file = 'Output/da_corncob.RData')


```

```{r corncob-plot, fig.width=6, fig.height=6}
dt_plot(dt_data)

```

```{r stop}
knitr::opts_chunk$set(eval = FALSE)

```


```{r corncob_ch4}

ps_fam_pas5_trep <- ps_filter(ps_fam_pas5, treat == "p")


ps_fam_prev_trep_prev <- tax_filter(ps_fam_pas5_trep, min_prevalence = 0.10)


da_analysis <- differentialTest(formula = ~ ch4,
                                 phi.formula = ~ ch4,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ ch4,
                                 test = "Wald", boot = FALSE,
                                 data = ps_fam_prev_trep_prev,
                                 fdr_cutoff = 0.05)
otu_to_taxonomy(OTU = da_analysis$significant_taxa, data = ps_fam_prev_trep_prev)

plot(da_analysis, level = c("Family"))

# Check unfitted taxa
# otu_to_taxonomy(which(is.na(da_analysis$p)) %>% names, data = ps_gen_prev)
# otu_table(ps_gen_prev)[,which(is.na(da_analysis$p))]

# Differential variability test

dv_analysis <- differentialTest(formula = ~ passage + treat + ch4,
                                 phi.formula = ~ passage + treat + ch4,
                                 formula_null = ~ passage + treat + ch4,
                                 phi.formula_null = ~ passage + treat,
                                 data = ps_gen_prev,
                                 test = "Wald", boot = FALSE,
                                 fdr_cutoff = 0.05)
# otu_to_taxonomy(OTU = dv_analysis$significant_taxa, data = ps_gen_prev)
plot(dv_analysis, level = c("Family", "Genus"))

# Check unfitted taxa

# otu_to_taxonomy(which(is.na(dv_analysis$p)) %>% names, data = ps_gen_prev)
# otu_table(ps_gen_prev)[,which(is.na(dv_analysis$p))]

```

