---
title: "Community Analysis"
author: "Andrew Morris"
date: "6/28/2022"
output: 
  html_document:
    toc: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = TRUE)
knitr::opts_knit$set(root.dir=normalizePath('../'))
set.seed(1337)

```

```{r initialize_packages}

library(phyloseq)
library(knitr)
library(magrittr)
library(vegan)
library(ggpubr)
library(ggthemes)
library(broom)
library(ggtext)
library(ALDEx2)
library(corncob)
library(ANCOMBC)
library(tidyverse)

source('R/functions.R')

```


```{r import-data}

aes_data <- read_tsv('Output/community_data.tsv')

```


```{r fix-treatment}

aes_data <-
  aes_data %>% 
  mutate(passage_treat = treat) %>% 
  separate(passage_treat, c('passage', "treat"), sep = 2, remove = FALSE) %>% 
  select(sample, passage_treat, passage, treat, selected, ch4, asv, count, Kingdom:Species, seq)

```



```{r fix-taxonomy}

# There are "Unknown Family" taxa with Genus and Class, but Order is Incertae Sedis.
# I'm just going to remove everything below Class and call it "Unclassified"

fill_taxonomy <- function(tab) {
    if(is.na(tab[[1]])) tab[[1]] = "Unclassified"
    for(i in 2:length(tab)) {
      if(is.na(tab[[i]])){
        if(grepl("unclassified", tab[[i-1]])) tab[[i]] = tab[[i-1]]
        else tab[[i]] = paste(tab[[i-1]], "unclassified", sep = "_")
      }
    }
    tab
}

aes_data <-
  aes_data %>% 
  nest(data = -c(asv, Kingdom:Species)) %>% 
  nest(taxa = Kingdom:Species) %>% 
  mutate(taxa = map(taxa, ~fill_taxonomy(.x))) %>% 
  unnest(taxa) %>% 
  unnest(data)

```

# Exploratory


```{r n-seqs}

aes_data %>% 
  group_by(sample) %>% 
  summarize(total = sum(count)) %>% 
  ggplot(aes(x = total)) +
  geom_histogram() +
  theme_classic() +
  labs(title = "Distribution sequencing depths",
       x = "Library size",
       y = "Frequency")

seq_range <-
  aes_data %>% 
  group_by(sample) %>% 
  summarize(total = sum(count)) %>% 
  summarize(seq_range = range(total)) %$% 
  seq_range

a_analysis <- 
  aes_data %>%
  count(sample, count) %>%
  nest(data = -sample) %>%
  mutate(n_seqs = map_dbl(data, ~sum(.x$count * .x$n)),
         sobs = map_dbl(data, ~sum(.x$n)),
         rare = map_dbl(data, ~rarefy(rep(.x$count, .x$n), sample=seq_range[1]))) %>%
  select(-data)


a_analysis %>%
  select(sample, n_seqs, sobs, rare) %>%
  pivot_longer(-c(sample, n_seqs)) %>%
  ggplot(aes(x=n_seqs, y=value, color=name)) +
  geom_point() +
  geom_smooth(se=FALSE) +
  coord_cartesian(ylim=c(0, 6000)) +
  theme_classic() +
  labs(x = "Sequencing Depth", y = "Alpha Diversity")

a_analysis %>%
  select(sample, n_seqs, sobs, rare) %>%
  mutate(treat = substr(sample, 1, 3)) %>% 
  select(-sample) %>% 
  group_by(treat) %>% 
  pivot_longer(-c(treat)) %>%
  ggplot(aes(x=treat, y=value)) +
  facet_wrap(~ name, nrow = 3, scales = "free") +
  geom_boxplot() +
  geom_jitter() +
  theme_classic() +
  labs(title = "AES", x = "Treatment", y = "Value")

```

This is the number of reads (n_seqs), the rarefied richness (rare), and 
the observed richness (sobs for "species observed") for the four treatments:
generations 2 and 5 and selection treatments neutral and positive. As you can
see, there were more sequences from G2 than G5 and also greater richness
in G2 than G5. However, that richness difference is robust to rarefying
to the level of the shallowest sample. There doesn't seem to be a systemic
difference in sequencing depth or richness between the neutral and positive 
treatments in either generation.


## Richness

### Plot

```{r alpha-diversity-plot}

# aes_data %>% 
#   group_by(sample) %>% 
#   summarize(n_seqs = sum(count)) %>% 
#   summarize(min = min(n_seqs))

aes_df <-
  aes_data %>% 
  select(sample, asv, count) %>%
  pivot_wider(names_from = "asv", values_from = "count", values_fill = 0) %>% 
  as.data.frame()

rownames(aes_df) <- aes_df$sample
aes_df <- aes_df[, -1]

rare <- 
  rarefy(aes_df, seq_range[1]) %>% 
  as_tibble(rownames = "sample") %>% 
  rename(rare = value)

aes_data %>% 
  inner_join(rare, by = 'sample') %>% 
  select(sample, passage_treat, rare) %>% 
  distinct %>% 
  ggplot(aes(x = passage_treat, y = rare)) +
  geom_boxplot() +
  geom_jitter() +
  labs(x = "Passage x Treatment", y = "Rarefied Richness") +
  theme_bw() +
  theme(
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

```

### Hypothesis Test

```{r alpha-diversity-test}

rare_data <-
  aes_data %>% 
  select(sample, passage_treat, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(rare, by = "sample")

kt <- kruskal.test(rare ~ passage_treat, data = rare_data)
kt <- tidy(kt)

wt <- pairwise.wilcox.test(rare_data$rare, 
                           rare_data$passage_treat, 
                           p.adjust.method = "BH")
wt <- tidy(wt)

rare_median <-
  rare_data %>% 
  group_by(passage) %>% 
  summarize(median = median(rare),
            IQR = quart(rare)[2] - quart(rare)[1])

  
```

There is an effect of passage x treatment on richness (Kruskal-Wallace test: chi-squared = 
`r kt[1,1]`, df = `r kt[1,3]`, p = `r kt[1, 2]`). The pairwise test shows that
G2 is different from G5, but N and P are not different in either generation
(Wilcoxon/Mann-Whitney Test).

```{r}

kable(wt)

```



```{r alpha-diversity-ch4-test}

s <- cor.test(~ ch4 + rare, data = rare_data, method = "spearman", alternative = "two.sided") %>% 
  tidy

s5 <-
  rare_data %>% 
  filter(passage == "G5") %>% 
  cor.test(~ ch4 + rare, data = ., method = "spearman", alternative = "two.sided")
save(kt, wt, rare_median, s, file = "Output/richness_models.RData")

```

A Spearman's rank correlation test shows no correlation between
methane oxidation rate and richness (Spearman's rho = `r s[1,1]`, p = `r s[1,3]`).
 
# Beta Diversity

```{r bray-curtis-nmds, messages=FALSE}

dist <- vegdist(aes_df, method="bray")

if(file.exists("Output/rare_dist_bray.rds")) {
  rare_dist_bray <- 
    readRDS("Output/rare_dist_bray.rds")
} else {
  rare_dist_bray <- avgdist(aes_df, dmethod="bray", sample=seq_range[1])
  rare_dist_bray %>% 
    saveRDS("Output/rare_dist_bray.rds")
}

nmds <- metaMDS(rare_dist_bray)

nmds_scores <-
  scores(nmds, display = "sites") %>% 
  as_tibble(rownames = "sample")
  
nmds_data <-
  aes_data %>% 
  select(sample, passage, treat) %>% 
  distinct %>% 
  inner_join(nmds_scores, by = "sample")

centroids <-
  nmds_data %>% 
  group_by(passage, treat) %>% 
  summarize(NMDS1 = mean(NMDS1),
            NMDS2 = mean(NMDS2),
            .groups = "drop")
p_beta <-
  nmds_data %>%   
  ggplot(aes(x = NMDS1, y = NMDS2, shape = passage, color = treat)) +
  geom_point() +
  stat_ellipse(show.legend = FALSE) +
  coord_fixed() +
  scale_shape_discrete(name = "Passage") +
  scale_color_manual(name = "Treatment",
                       labels = c('Neutral', 'Positive'),
                       values = c('gray40', 'darkorange2')) #+
  # theme_bw() +
  # theme(
  #   plot.background = element_blank(),
  #   panel.grid.major = element_blank(),
  #   panel.grid.minor = element_blank())

save(p_beta, file = "Output/beta_fig.RData")

ggsave('Manuscript/fig3.jpg', p_beta, height = 2.5, width = 4) 

```

## Beta diversity test passage by treatment

```{r permanova-passage-treatment, warnings=FALSE}

dist_df <-
  rare_dist_bray %>% 
  as.matrix %>% 
  as_tibble(rownames = "sample")

meta_dist <-
  aes_data %>% 
  select(sample, passage_treat, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(dist_df, by = "sample")

permanova <- adonis2(rare_dist_bray ~ passage * treat, data = meta_dist, permutations = 999)

betadisper(rare_dist_bray, meta_dist$passage_treat) %>% 
  permutest()

tidy_permanova <- 
  tidy(permanova)

save(tidy_permanova, file = "Output/beta_model.RData")

```

There is a significant difference of centroid between passage, treatment, and
their interaction. Testing a difference of dispersion using betadisper
shows no difference in variance between the four groups.

# Distance-based RDA Methane

```{r full-dbrda}

meta_dist %>% 
  select(passage_treat, ch4) %>% 
  mutate(log_ch4 = log_methane(ch4)) %>% 
  ggplot(aes(x = passage_treat, y = log_ch4)) +
  geom_boxplot() +
  theme_classic() +
  labs(x = "Passage x Treatment", y = "Log10 Methane")

dbrda_ch4 <- dbrda(rare_dist_bray ~ rank(ch4), 
                      data = meta_dist, 
                      na.action = "na.omit",
                      add = TRUE) 

dbrda_ch4 %>% 
  permutest(permutations = 9999)

percent_explained <- 100 * eigenvals(dbrda_ch4) / sum(eigenvals(dbrda_ch4))

dbrda_scores <-
  scores(dbrda_ch4, display = "sites") %>% 
  as_tibble(rownames = "sample")
  
dbrda_data <-
  aes_data %>% 
  select(sample, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(dbrda_scores, by = "sample")

dbrda_data %>%   
  ggplot(aes(x = dbRDA1, y = MDS1, shape = passage, color = treat)) +
  geom_point() +
  stat_ellipse(show.legend = FALSE) +
  coord_fixed() +
  scale_shape_discrete(name = "Passage") +
  scale_color_manual(name = "Treatment",
                       labels = c('Neutral', 'Positive'),
                       values = c('gray40', 'darkorange2')) +
  theme_bw() +
  theme(
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

```


```{r pre-dbrda-data-subsets}

# aes_data %>% 
#   filter(treat == "P") %>% 
#   group_by(asv) %>% 
#   mutate(total = sum(count)) %>% 
#   filter(total != 0) %>% 
#   ungroup() %>% 
#   select(-total) %>% 
#   group_by(sample) %>% 
#   summarize(n_seqs = sum(count)) %>% 
#   summarize(min = min(n_seqs))

treatP_df <-
  aes_data %>%  
  filter(treat == "P") %>% 
  group_by(asv) %>% 
  mutate(total = sum(count)) %>% 
  filter(total != 0) %>% 
  ungroup() %>% 
  select(-total) %>% 
  select(sample, asv, count) %>%
  pivot_wider(names_from = "asv", values_from = "count", values_fill = 0) %>% 
  as.data.frame()

rownames(treatP_df) <- treatP_df$sample
treatP_df <- treatP_df[, -1]

# aes_data %>% 
#   filter(passage == "G5") %>% 
#   group_by(asv) %>% 
#   mutate(total = sum(count)) %>% 
#   filter(total != 0) %>% 
#   ungroup() %>% 
#   select(-total) %>% 
#   group_by(sample) %>% 
#   summarize(n_seqs = sum(count)) %>% 
#   summarize(min = min(n_seqs))

passage5_df <-
  aes_data %>%  
  filter(passage == "G5") %>% 
  group_by(asv) %>% 
  mutate(total = sum(count)) %>% 
  filter(total != 0) %>% 
  ungroup() %>% 
  select(-total) %>% 
  select(sample, asv, count) %>%
  pivot_wider(names_from = "asv", values_from = "count", values_fill = 0) %>% 
  as.data.frame()

rownames(passage5_df) <- passage5_df$sample
passage5_df <- passage5_df[, -1]

```

```{r treat-dbrda}

if(file.exists("Output/treat_rare_dist_bray.rds")) {
  treat_rare_dist_bray <- 
    readRDS("Output/treat_rare_dist_bray.rds")
} else {
treat_rare_dist_bray <- avgdist(treatP_df, dmethod="bray", sample=seq_range[1])
  treat_rare_dist_bray %>% 
    saveRDS("Output/treat_rare_dist_bray.rds")
}



treat_dist_df <-
  treat_rare_dist_bray %>% 
  as.matrix %>% 
  as_tibble(rownames = "sample")

treat_meta_dist <-
  aes_data %>% 
  select(sample, passage_treat, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(treat_dist_df, by = "sample")


treat_dbrda_ch4 <- dbrda(treat_rare_dist_bray ~ rank(ch4), 
                      data = treat_meta_dist, 
                      na.action = "na.omit",
                      add = TRUE) 

treat_dbrda_ch4 %>% 
  permutest(permutations = 9999)

percent_explained <- 100 * eigenvals(treat_dbrda_ch4) / sum(eigenvals(treat_dbrda_ch4))

treat_dbrda_scores <-
  scores(treat_dbrda_ch4, display = "sites") %>% 
  as_tibble(rownames = "sample")
  
treat_dbrda_data <-
  aes_data %>% 
  select(sample, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(treat_dbrda_scores, by = "sample")

treat_dbrda_data %>%   
  ggplot(aes(x = dbRDA1, y = MDS1, shape = passage, color = treat)) +
  geom_point() +
  stat_ellipse(show.legend = FALSE) +
  coord_fixed() +
  scale_shape_discrete(name = "Passage") +
  scale_color_manual(name = "Treatment",
                       labels = c('Positive'),
                       values = c('darkorange2')) +
  theme_bw() +
  theme(
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())

```

```{r passage-dbrda}
if(file.exists("Output/passage_rare_dist_bray.rds")) {
  passage_rare_dist_bray <- 
    readRDS("Output/passage_rare_dist_bray.rds")
} else {
passage_rare_dist_bray <- avgdist(passage5_df, dmethod="bray", sample=seq_range[1])
  passage_rare_dist_bray %>% 
    saveRDS("Output/passage_rare_dist_bray.rds")
}

passage_dist_df <-
  passage_rare_dist_bray %>% 
  as.matrix %>% 
  as_tibble(rownames = "sample")

passage_meta_dist <-
  aes_data %>% 
  select(sample, passage_treat, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(passage_dist_df, by = "sample")


passage_dbrda_ch4 <- dbrda(passage_rare_dist_bray ~ rank(ch4), 
                      data = passage_meta_dist, 
                      na.action = "na.omit",
                      add = TRUE) 

passage_dbrda_ch4 %>% 
  permutest(permutations = 9999)

passage_percent_explained <- 100 * eigenvals(passage_dbrda_ch4) / sum(eigenvals(passage_dbrda_ch4))

passage_dbrda_scores <-
  scores(passage_dbrda_ch4, display = "sites") %>% 
  as_tibble(rownames = "sample")
  
passage_dbrda_data <-
  aes_data %>% 
  select(sample, passage, treat, ch4) %>% 
  distinct %>% 
  inner_join(passage_dbrda_scores, by = "sample")

passage_dbrda_data %>%   
  ggplot(aes(x = dbRDA1, y = MDS1, shape = passage, color = treat)) +
  geom_point() +
  stat_ellipse(show.legend = FALSE) +
  coord_fixed() +
  scale_shape_discrete(name = "Passage") +
  scale_color_manual(name = "Treatment",
                       labels = c('Neutral', 'Positive'),
                       values = c('gray40', 'darkorange2')) +
  theme_bw() +
  theme(
    plot.background = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank())


```

The overall dbRDA is significant and rank of methane explains 5% of the variation.
Subsetting the Positive treatment in both G2 and G5, it is not significant: no
correlation between dissimilarity and rank of methane between G2 and G5. This
makes sense, because most of that variation would be due to selection in the jar.
However, subsetting passage G5 and comparing N and P, 10% of dissimilarity is
explained by rank of CH4. 

# Differential Abundance

## Unique to each treatment

```{r}
g5n <- 
  aes_data %>% 
  filter(passage_treat == "G5N") %>% 
  select(Family) %>% 
  distinct() %$%
  Family

g5p <- 
  aes_data %>% 
  filter(passage_treat == "G5P") %>% 
  select(Family) %>% 
  distinct() %$%
  Family


p_not_n <- g5p[!g5p %in% g5n]

n_not_p <- g5n[!g5n %in% g5p]

n_not_p_prevalence <-
  aes_data %>% 
  filter(passage_treat == "G5N") %>% 
  filter(Family %in% n_not_p) %>% 
  select(sample, Family, count) %>% 
  mutate(prevalence = as.numeric(count > 0)) %>% 
  select(-count) %>% 
  group_by(Family) %>% 
  summarize(prevalence = sum(prevalence))

n_not_p_prevalence %>% 
  ggplot(aes(x = prevalence)) +
  geom_histogram() +
  scale_x_continuous(breaks = 1:12) +
  theme_classic() +
  labs(title = "Unique to Positive")

# n_not_p_prevalence %>%
#   filter(prevalence == 1) %>% 
#   nrow

# n_not_p_prevalence %>%
#   filter(prevalence == 2) %>% 
#   nrow
# n_not_p_prevalence %>%
#   filter(prevalence > 2) %$%
#   Family

p_not_n_prevalence <-
  aes_data %>% 
  filter(passage_treat == "G5P") %>% 
  filter(Family %in% p_not_n) %>% 
  select(sample, Family, count) %>% 
  mutate(prevalence = as.numeric(count > 0)) %>% 
  select(-count) %>% 
  group_by(Family) %>% 
  summarize(prevalence = sum(prevalence))

p_not_n_prevalence %>% 
  ggplot(aes(x = prevalence)) +
  geom_histogram() +
  scale_x_continuous(breaks = 1:12) +
  theme_classic() +
  labs(title = "Unique to Neutral")
# 
# p_not_n_prevalence %>%
#   filter(prevalence == 1) %>% 
#   nrow
# 
# p_not_n_prevalence %>%
#   filter(prevalence == 2) %>% 
#   nrow
p_not_n_over_6 <- p_not_n_prevalence %>%
  filter(prevalence > 6)

```

First, I looked at which Families are unique to each treatment in passage 5.
There were `r length(n_not_p)` families unique to N and `r length(p_not_n)`
unique to P. However, of the  `r length(n_not_p)` unique to P, 
`r n_not_p_prevalence %>% filter(prevalence == 1) %>% nrow` had a prevalence
of 1 and `r n_not_p_prevalence %>% filter(prevalence == 2) %>% nrow` had
a prevalence of 2 leaving only a single family, which had a prevalence of
`r n_not_p_prevalence %>% filter(prevalence > 2) %$% prevalence`, which was `r n_not_p_prevalence %>% filter(prevalence > 2) %$% Family`.
Of the `r length(p_not_n)` Families unique to P, 2 had prevalences greater than
6, which were `r p_not_n_over_6$Family[1]` with a prevalence of
`r p_not_n_over_6$prevalence[1]` and `r p_not_n_over_6$Family[2]` with
a prevalence of `r p_not_n_over_6$prevalence[2]`.


## Setup Differential Abundance Data

```{r setup-differential-abundance-data}

aes_G5 <-
  aes_data %>% 
  filter(passage == "G5")


aes_family_G5 <-
  aes_G5 %>% 
  select(asv, sample, treat, count, Family) %>%
  group_by(sample, Family) %>% 
  mutate(total = sum(count)) %>% 
  ungroup() %>% 
  select(sample, treat, Family, total) %>% 
  distinct()

prevalent_families <- 
  aes_family_G5 %>% 
  group_by(Family, treat) %>% 
  summarize(prev = sum(total > 0), .groups = "drop") %>% 
  pivot_wider(names_from = "treat", values_from = "prev", values_fill = 0) %>% 
  filter(N/12 > 0.1 & P/12 > 0.1) %$%
  Family

otu_table <- 
  aes_family_G5 %>% 
  filter(Family %in% prevalent_families) %>% 
  select(sample, Family, total) %>% 
  pivot_wider(names_from = sample, values_from = total, values_fill = 0) %>% 
  as.data.frame()

rownames(otu_table) <- otu_table[,1]
otu_table <- otu_table[,-1]

sample_data <- aes_family_G5 %>% 
  select(sample, treat) %>% 
  distinct() %>% 
  as.data.frame()

rownames(sample_data) <- sample_data[,1]

```

## Corncob



```{r}

da_analysis <- differentialTest(formula = ~ treat,
                                phi.formula = ~ 1,
                                formula_null = ~ 1,
                                phi.formula_null = ~ 1,
                                test = "Wald", 
                                boot = FALSE,
                                data = otu_table,
                                sample_data = sample_data,
                                fdr_cutoff = 0.05, B = 100,
                                taxa_are_rows = TRUE)

dt_data <- plot(da_analysis, 
                # level = c("Family"), 
                data_only = TRUE)

dt_plot <- function(dt_data) {
  ggplot(dt_data, aes(x = x, xmin = xmin, xmax = xmax, y = reorder(taxa, x))) +
    geom_pointrange() +
    geom_vline(xintercept = 0) +
    labs(x = "Estimate", y = NULL) +
    coord_cartesian(clip="off") +
    # theme(panel.border = element_blank()) +
    theme_classic() +
      theme(axis.text.y = element_markdown())
    
  
}

dt_data %>%
  mutate(taxa = str_replace(taxa, "(.*)", "*\\1*"),
         taxa = str_replace(taxa, "\\*(.*)_unclassified\\*",
                                "Unclassified *\\1*")) %>% 
  dt_plot()

```

## ANCOM

```{r}

# Setup phyloseq object
aes_ps <- phyloseq(otu_table(otu_table, taxa_are_rows=TRUE), sample_data(sample_data))

if(file.exists("Output/ancom_out.rds")) {
  out <- 
    readRDS("Output/ancom_out.rds")
} else {
  # Run Ancom
  out = ancom(phyloseq = aes_ps,  
              p_adj_method = "BH",
              prv_cut = 0.1, 
              lib_cut = 0, 
              main_var = "treat",
              rand_formula = NULL, 
              lme_control = NULL,
              struc_zero = TRUE, 
              neg_lb = FALSE, 
              alpha = 0.05, 
              n_cl = 1)  
  out %>% 
    saveRDS("Output/ancom_out.rds")
}


res = out$res
q_val = out$q_data
beta_val = out$beta_data
# Only consider the effect sizes with the corresponding q-value less than alpha
beta_val = beta_val * (q_val < 0.05) 
# Choose the maximum of beta's as the effect size
beta_pos = apply(abs(beta_val), 2, which.max) 
beta_max = vapply(seq_along(beta_pos), function(i) beta_val[beta_pos[i], i],
                  FUN.VALUE = double(1))
# Number of taxa except structural zeros
n_taxa = ifelse(is.null(out$zero_ind), 
                nrow(feature_table), 
                sum(apply(out$zero_ind, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = 0.7 * (n_taxa - 1)

df_fig_w = res %>%
  dplyr::mutate(beta = beta_max,
                direct = case_when(
                  detected_0.7 == TRUE & beta > 0 ~ "Positive",
                  detected_0.7 == TRUE & beta <= 0 ~ "Negative",
                  TRUE ~ "Not Significant"
                  )) %>%
  dplyr::arrange(W)
df_fig_w$taxon_id = factor(df_fig_w$taxon_id, levels = df_fig_w$taxon_id)
df_fig_w$W = replace(df_fig_w$W, is.infinite(df_fig_w$W), n_taxa - 1)
df_fig_w$direct = factor(df_fig_w$direct, 
                     levels = c("Negative", "Positive", "Not Significant"))

p_w = df_fig_w %>%
  ggplot(aes(x = taxon_id, y = W, color = direct)) +
  geom_point(size = 2, alpha = 0.6) +
  labs(x = "Taxon", y = "W") +
  scale_color_discrete(name = NULL) + 
  geom_hline(yintercept = cut_off, linetype = "dotted", 
             color = "blue", size = 1.5) +
  geom_text(aes(x = 2, y = cut_off + 0.5, label = "W[0.7]"), 
            size = 5, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE) +
  theme_bw() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        panel.grid.major = element_blank())
p_w

```

## Aldex2

```{r}

conds <- colnames(otu_table) %>% 
  substr(3, 3)

if(file.exists("Output/aldex_x_all.rds")) {
  x.all <- 
    readRDS("Output/aldex_x_all.rds")
} else {
  x.all <- aldex(otu_table, 
                 conds, mc.samples=1000, test="t", effect=TRUE,
                 include.sample.summary=FALSE, denom="all", verbose=FALSE, paired.test=FALSE)
  x.all %>% 
    saveRDS("Output/aldex_x_all.rds")
}



par(mfrow=c(1,2))
aldex.plot(x.all, type="MA", test="welch", xlab="Log-ratio abundance",
    ylab="Difference")
aldex.plot(x.all, type="MW", test="welch", xlab="Dispersion",
    ylab="Difference")

par(mfrow=c(1,2))
plot(x.all$effect, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(x.all$effect, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red"))

plot(x.all$diff.btw, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(x.all$diff.btw, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

x.all_effect_1 <-
  x.all %>% 
  filter(abs(effect) > 1) %>% 
  as_tibble(rownames = "Family")

# x.all_effect_1 %>% 
#   # inner_join(taxa_table, by = "asv")
#   ggplot(aes(y = reorder(Family, effect), x = effect)) +
#   geom_point()

# x.all_effect_1 %>% 
#   select(Family, effect)
  

```

## Comparison

### All Significant

```{r}

# CORNCOB
corncob_taxa <- dt_data$taxa

# ANCOM
ancom_taxa <- as.character(df_fig_w[df_fig_w$detected_0.7, ]$taxon_id)

# ALDEX2
aldex_taxa <- x.all_effect_1$Family



at_least_one_taxa <-
	aes_family_G5 %>%
	group_by(sample) %>%
	mutate(rel_abund = (total / sum(total)) * 100) %>%
	filter(Family %in% ancom_taxa | Family %in% aldex_taxa | Family %in% corncob_taxa) %>%
	mutate(Family = str_replace(Family, "(.*)", "*\\1*"),
	       Family = str_replace(Family, "\\*(.*)_unclassified\\*",
				    "Unclassified *\\1*"))

at_least_one_iqm <- 
	at_least_one_taxa %>%
	group_by(treat, Family) %>%
	summarize(median = median(rel_abund), 
		  first_quartile = quart(rel_abund)[1],
		  third_quartile = quart(rel_abund)[2], .groups = "drop")

sort_order <-
	at_least_one_taxa %>%
	group_by(Family) %>%
	summarize(median = median(rel_abund)) %>%
	arrange(median)

at_least_one_taxa$Family <- factor(at_least_one_taxa$Family, levels = sort_order$Family)

at_least_one_taxa %>%
	ggplot(aes(x = rel_abund, y = Family, color = treat)) +
	geom_point(position=position_jitterdodge()) +
	geom_pointrange(aes(x = median, xmax = third_quartile, xmin = first_quartile, fill = treat), 
			data = at_least_one_iqm, 
			position = position_dodge(width = 0.75), 
			shape = 21, 
			color = "black") +
	theme_bw() +
	theme(axis.text.y = element_markdown()) +
	labs(x = "Relative Abundance (%)", y = NULL) +
	scale_color_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2')) +
	scale_fill_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2'))

```

### Consensus Taxa

```{r}
aldex_effect <-
  x.all_effect_1 %>% 
  select(Family, effect)

all_taxa <-
	aes_family_G5 %>%
	group_by(sample) %>%
	mutate(rel_abund = (total / sum(total)) * 100) %>%
  left_join(aldex_effect, by = "Family") %>% 
	filter(Family %in% ancom_taxa & Family %in% aldex_taxa & Family %in% corncob_taxa) %>% 
  ungroup()

da_fams <-
  all_taxa %>% 
  select(Family) %>% 
  distinct() %$%
  Family

all_taxa <-
  all_taxa %>%
	mutate(Family = str_replace(Family, "(.*)", "*\\1*"),
	       Family = str_replace(Family, "\\*(.*)_unclassified\\*",
				    "Unclassified *\\1*"))
all_taxa$Family[all_taxa$Family == "Unclassified *0319-6G20*"] <- "Uncultured family 0319-6G20"
all_taxa$Family[all_taxa$Family == "Unclassified *Chloroplast*"] <- "Chloroplast"
all_taxa$Family[all_taxa$Family == "Unclassified *Armatimonadales*"] <- "Unclassified Armatimonadales"
all_taxa$Family[all_taxa$Family == "Unclassified *Gammaproteobacteria*"] <- "Unclassified Gammaproteobacteria"
all_taxa$Family[all_taxa$Family == "Unclassified *Kapabacteriales*"] <- "Unclassified Kapabacteriales"

all_iqm <- 
	all_taxa %>%
	group_by(treat, Family) %>%
	summarize(median = median(rel_abund), 
		  first_quartile = quart(rel_abund)[1],
		  third_quartile = quart(rel_abund)[2], .groups = "drop")

sort_order <-
	all_taxa %>%
  select(Family, effect) %>% 
  distinct() %>% 
  arrange(effect)
	# group_by(Family) %>%
	# summarize(median = median(rel_abund)) %>%
	# arrange(median)

all_taxa$Family <- factor(all_taxa$Family, levels = sort_order$Family)
all_iqm$Family <- factor(all_iqm$Family, levels = sort_order$Family)


p <-
	all_taxa %>%
	ggplot(aes(x = rel_abund, y = Family, color = treat)) +
	geom_point(position=position_jitterdodge()) +
	geom_pointrange(aes(x = median, xmax = third_quartile, xmin = first_quartile, fill = treat), 
			data = all_iqm, 
			position = position_dodge(width = 0.75), 
			shape = 21, 
			color = "black") +
	theme_bw() +
	theme(axis.text.y = element_markdown()) +
	labs(x = "Relative Abundance (%)", y = NULL) +
	scale_color_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2')) +
	scale_fill_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2'))



p_log10 <- 
	p +
	scale_x_log10(breaks = c(0.001, 0.01, 0.1, 1),
		      labels = c(0.001, 0.01, 0.1, 1))

p
p_log10

save(p_log10, da_fams, file = "Output/da_fig.RData")

ggsave('Manuscript/fig4.jpg', p_log10, width = 7, height = 7) 


## All
# ggsave("shared_taxa.png", p)
ggsave("shared_taxa_log.png", p_log10, width = 7, height = 7)

# aes_family_rclr_plot <-
#   aes_family_rclr %>% 
# 	filter(Family %in% ancom_taxa & Family %in% aldex_taxa & Family %in% corncob_taxa) %>% 
#   pivot_longer(G2:G5, names_to = "passage", values_to = "rclr") %>% 
#   select(Family, treat, rep, passage, rclr) %>% 
#   distinct() %>% 	
#   mutate(Family = str_replace(Family, "(.*)", "*\\1*"),
# 	       Family = str_replace(Family, "\\*(.*)_unclassified\\*",
# 				    "Unclassified *\\1*"))
# 
# aes_family_rclr_plot$Family[aes_family_rclr_plot$Family == "Unclassified *0319-6G20*"] <- "Uncultured family 0319-6G20"
# aes_family_rclr_plot$Family[aes_family_rclr_plot$Family == "Unclassified *Chloroplast*"] <- "Chloroplast"
# aes_family_rclr_plot$Family <- factor(aes_family_rclr_plot$Family, levels = sort_order$Family)
# ggplot(aes_family_rclr_plot, aes(x = rclr, y = Family, color = treat)) +
#   facet_grid( ~ passage) +
#   geom_point(position = position_jitterdodge())

p_effect <-
	all_taxa %>%
	ggplot(aes(x = effect, y = Family)) +
	geom_point() +

	theme_bw() +
	theme(axis.text.y = element_markdown()) +
	labs(x = "Aldex2 Effect Size", y = NULL) +
	scale_color_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2')) +
	scale_fill_manual(name = "Treatment",
			   labels = c('Neutral', 'Positive'),
			   values = c('gray40', 'darkorange2'))
p_effect


```


## Methanogens and Methanotrophs

Methanotrophy is (mostly) conserved at the family level. The Type I and Type X methanotrophs are in the Gammaproteobacteria in the families _Methylococcaceae_ and _Methylothermaceae_. There are also methanotrophs in the Verrumicrobiota in the family _Methylacidiphilaceae_. The Type II methanotrophs are in the _Alphaproteobacteria_ and are found in the families _Methylocystaceae_ and _Beijerinckiaceae_. However, the _Beijerinckiaceae_ also contain non-methanotrophs including N2-fixers and photoautotrophs. 


```{r methanotroph-lists}

methanotrophs <- c('Methylococcaceae', 
                   'Methylothermaceae', 
                   'Methylocystaceae', 
                   'Beijerinckiaceae', 
                   'Methylacidiphilaceae')
methanotrophs <- c(methanotrophs, paste(methanotrophs, "unclassified", sep = "_"))

troph_orders <- c('Methylococcales')

methanogen_classes <- c("Methanococci", "Methanomicrobia", "Methanobacteria")

methanogen_classes <- c(methanogen_classes, paste(methanogen_classes, "unclassified", sep = "_"))

```

```{r}

aes_data %>% 
  select(Family, Genus, count, sample) %>% 
  group_by(sample, Genus, Family) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  group_by(sample) %>% 
  mutate(rel_abund = total / sum(total)) %>% 
  mutate(rclr = log(total/gm(total))) %>% 
  filter(Family %in% methanotrophs) %>% 
  select(-Family) %>% 
  mutate(treat = substr(sample, 3, 3), passage = substr(sample, 1, 2)) %>% 
  pivot_longer(rel_abund:rclr, names_to = "method", values_to = "value") %>% 
  ggplot(aes(x = value, y = Genus, color = treat)) +
  geom_jitter() +
  facet_grid(passage ~ method, scales = "free") +
  theme_bw()

aes_data %>% 
  select(Family, count, sample) %>% 
  group_by(sample, Family) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  group_by(sample) %>% 
  mutate(rel_abund = total / sum(total)) %>% 
  mutate(rclr = log(total/gm(total))) %>% 
  filter(Family %in% methanotrophs) %>% 
  mutate(treat = substr(sample, 3, 3), passage = substr(sample, 1, 2)) %>% 
  pivot_longer(rel_abund:rclr, names_to = "method", values_to = "value") %>% 
  ggplot(aes(x = value, y = Family, color = treat)) +
  geom_jitter() +
  facet_grid(passage ~ method, scales = "free") +
  theme_bw()


troph_asvs <-
  aes_data %>% 
  select(Kingdom:Species, asv) %>% 
  distinct() %>% 
  filter(Family %in% methanotrophs) %$%
  asv

```


Of the five methanotroph families, two are represented in this dataset:
_Methylacidiphilaceae_ and _Beijerinckiaceae_. From these two families,
there are eleven genera represented. These are spread across
`r length(troph_asvs)` ASVs

```{r}

aes_data %>% 
  select(Class, count, sample) %>% 
  group_by(sample, Class) %>% 
  mutate(total = sum(count)) %>% 
  select(-count) %>% 
  distinct() %>% 
  filter(Class %in% methanogen_classes)

```


A grand total of 10 reads map to methanogen classes across the entire dataset,
including 2 reads in Methanobacteria (sample G2N02) and 8 reads of
Methanomicrobia (sample G2N07).

### Methanotroph differential abundance

```{r}

troph_data <-
  aes_data %>% 
  select(sample, asv, treat, passage, count, Genus, Family) %>% 
  filter(passage == "G5") %>% 
  select(-passage) %>% 
  filter(asv %in% troph_asvs)
  
troph_data_family <-
  troph_data %>% 
  select(sample, Family, count) %>% 
  group_by(sample, Family) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  pivot_wider(names_from = sample, values_from = total, values_fill = 0) %>% 
  as.data.frame()

rownames(troph_data_family) <- troph_data_family[, 1]
troph_data_family <- troph_data_family[, -1]


troph_data_genus <-
  troph_data %>% 
  select(sample, Genus, count) %>% 
  group_by(sample, Genus) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  pivot_wider(names_from = sample, values_from = total, values_fill = 0) %>% 
  as.data.frame()

rownames(troph_data_genus) <- troph_data_genus[, 1]
troph_data_genus <- troph_data_genus[, -1]

sample_data <- 
  troph_data %>% 
  select(sample, treat) %>% 
  distinct() %>% 
  as.data.frame()
conds <- sample_data[, 2]

```


```{r corncob-methanotrophs}
if(file.exists("Output/cc_family.rds")) {
  cc_family <- 
    readRDS("Output/cc_family.rds")
} else {
cc_family <- differentialTest(formula = ~ treat,
                                phi.formula = ~ 1,
                                formula_null = ~ 1,
                                phi.formula_null = ~ 1,
                                test = "Wald", 
                                boot = TRUE,
                                data = troph_data_family,
                                sample_data = sample_data,
                                fdr_cutoff = 0.05, B = 1000)  
cc_family %>% 
    saveRDS("Output/cc_family.rds")
}

plot(cc_family)

if(file.exists("Output/cc_genus.rds")) {
  cc_genus <- 
    readRDS("Output/cc_genus.rds")
} else {
cc_genus <- differentialTest(formula = ~ treat,
                                phi.formula = ~ 1,
                                formula_null = ~ 1,
                                phi.formula_null = ~ 1,
                                test = "Wald", 
                                boot = TRUE,
                                data = troph_data_genus,
                                sample_data = sample_data,
                                fdr_cutoff = 0.05, B = 1000) 
cc_genus %>% 
    saveRDS("Output/cc_genus.rds")
}


plot(cc_genus)

```


```{r stop}

knitr::opts_chunk$set(eval = FALSE)

```




```{r}


troph_genus_data <-
	aes_data %>%
	filter(Family %in% methanotrophs) %>%
	select(sample, count, Genus, passage, treat) %>%
	group_by(sample, Genus) %>%
	mutate(total = sum(count)) %>%
	select(-count) %>%
	ungroup() %>%
	group_by(sample) %>%
	mutate(rel_abund = total / sum(total)) %>%
	ungroup() %>%
	distinct() 
troph_genus_iqm <- 
	troph_genus_data %>%
	group_by(passage, treat, Genus) %>%
	summarize(median = median(rel_abund), 
		  first_quartile = quart(rel_abund)[1],
		  third_quartile = quart(rel_abund)[2], .groups = "drop")
troph_genus_data %>%
	ggplot(aes(x = rel_abund, y = Genus, color = treat)) +
	geom_point(position=position_jitterdodge()) +
	facet_grid( ~ passage) +
	geom_pointrange(aes(x = median, xmax = third_quartile, xmin = first_quartile, fill = treat), data = troph_genus_iqm, position = position_dodge(width = 0.75), shape = 21, color = "black") +
	theme_bw()
ggsave("troph_genus.png")

aes_data %>%
	select(sample, count, Genus, passage, treat) %>%
	group_by(sample, Genus) %>%
	mutate(total = sum(count)) %>%
	select(-count) %>%
	ungroup() %>%
	group_by(sample) %>%
	mutate(rel_abund = total / sum(total)) %>%
	ungroup() %>%
	filter(Genus %in% c("Methylocella", "Microvirga", "Methylorosula")) %>%
	distinct() %>%
	ggplot(aes(x = rel_abund, y = Genus)) +
	geom_jitter() +
	facet_grid(treat ~ passage)


troph_family_data <-
	aes_data %>%
	select(sample, count, Family, passage, treat) %>%
	group_by(sample, Family) %>%
	mutate(total = sum(count)) %>%
	select(-count) %>%
	ungroup() %>%
	group_by(sample) %>%
	mutate(rel_abund = total / sum(total)) %>%
	ungroup() %>%
	filter(Family %in% methanotrophs) %>%
	distinct()
troph_family_iqm <- 
	troph_family_data %>%
	group_by(passage, treat, Family) %>%
	summarize(median = median(rel_abund), 
		  first_quartile = quart(rel_abund)[1],
		  third_quartile = quart(rel_abund)[2], .groups = "drop")
troph_family_data %>%
	ggplot(aes(x = rel_abund, y = Family, color = treat)) +
	geom_point(position=position_jitterdodge()) +
	facet_grid( ~ passage) +
	geom_pointrange(aes(x = median, xmax = third_quartile, xmin = first_quartile, fill = treat), data = troph_family_iqm, position = position_dodge(width = 0.75), shape = 21, color = "black") +
	theme_bw()
ggsave("troph_family.png")


troph_species_data <-
	aes_data %>%
	filter(Family %in% methanotrophs) %>%
	select(sample, count, Species, passage, treat) %>%
	group_by(sample, Species) %>%
	mutate(total = sum(count)) %>%
	select(-count) %>%
	ungroup() %>%
	group_by(sample) %>%
	mutate(rel_abund = total / sum(total)) %>%
	ungroup() %>%
	distinct()
troph_species_iqm <- 
	troph_species_data %>%
	group_by(passage, treat, Species) %>%
	summarize(median = median(rel_abund), 
		  first_quartile = quart(rel_abund)[1],
		  third_quartile = quart(rel_abund)[2], .groups = "drop")
troph_species_data %>%
	ggplot(aes(x = rel_abund, y = Species, color = treat)) +
	geom_point(position=position_jitterdodge()) +
	facet_grid( ~ passage) +
	geom_pointrange(aes(x = median, xmax = third_quartile, xmin = first_quartile, fill = treat), data = troph_species_iqm, position = position_dodge(width = 0.75), shape = 21, color = "black") +
	theme_bw()
ggsave("troph_species.png")

```

```{r aldex-methanotrophs-family}

x.all <- aldex(troph_data_family, 
               conds, mc.samples=1000, test="t", effect=TRUE,
     include.sample.summary=FALSE, denom="all", verbose=FALSE, paired.test=FALSE)

par(mfrow=c(1,2))
aldex.plot(x.all, type="MA", test="welch", xlab="Log-ratio abundance",
    ylab="Difference")
aldex.plot(x.all, type="MW", test="welch", xlab="Dispersion",
    ylab="Difference")

par(mfrow=c(1,2))
plot(x.all$effect, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(x.all$effect, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red"))

plot(x.all$diff.btw, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(x.all$diff.btw, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

x.all

```

```{r aldex-methanotrophs-genus}

x.all <- aldex(troph_data_genus, 
               conds, mc.samples=1000, test="t", effect=TRUE,
     include.sample.summary=FALSE, denom="all", verbose=FALSE, paired.test=FALSE)

par(mfrow=c(1,2))
aldex.plot(x.all, type="MA", test="welch", xlab="Log-ratio abundance",
    ylab="Difference")
aldex.plot(x.all, type="MW", test="welch", xlab="Dispersion",
    ylab="Difference")

par(mfrow=c(1,2))
plot(x.all$effect, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(x.all$effect, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red"))

plot(x.all$diff.btw, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(x.all$diff.btw, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

x.all

```


```{r}
## ANCOM
```
```{r}

# Setup phyloseq objects

ps_family <- phyloseq(otu_table(troph_data_family,
                                taxa_are_rows = TRUE),
                      sample_data(sample_data))
ps_genus <- phyloseq(otu_table(troph_data_genus,
                                taxa_are_rows = TRUE),
                      sample_data(sample_data))

# Run Ancom

out_family = ancom(phyloseq = ps_family,  
                  p_adj_method = "BH",
                  prv_cut = 0.1, 
                  lib_cut = 0, 
                  main_var = "treat",
                  rand_formula = NULL, 
                  lme_control = NULL,
                  struc_zero = TRUE, 
                  neg_lb = FALSE, 
                  alpha = 0.05, 
                  n_cl = 1)

out = ancom(phyloseq = ps_genus,  
                  p_adj_method = "BH",
                  prv_cut = 0.1, 
                  lib_cut = 0, 
                  main_var = "treat",
                  rand_formula = NULL, 
                  lme_control = NULL,
                  struc_zero = TRUE, 
                  neg_lb = FALSE, 
                  alpha = 0.05, 
                  n_cl = 1)

```

```{r}

q_val = out$q_data
beta_val = out$beta_data
# Only consider the effect sizes with the corresponding q-value less than alpha
beta_val = beta_val * (q_val < 0.05) 
# Choose the maximum of beta's as the effect size
beta_pos = apply(abs(beta_val), 2, which.max) 
beta_max = vapply(seq_along(beta_pos), function(i) beta_val[beta_pos[i], i],
                  FUN.VALUE = double(1))

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(out$zero_ind), 
                nrow(feature_table), 
                sum(apply(out$zero_ind, 1, sum) == 0))

# Cutoff values for declaring differentially abundant taxa
cut_off = 0.7 * (n_taxa - 1)

df_fig_w = out$res %>%
  dplyr::mutate(beta = beta_max,
                direct = case_when(
                  detected_0.7 == TRUE & beta > 0 ~ "Positive",
                  detected_0.7 == TRUE & beta <= 0 ~ "Negative",
                  TRUE ~ "Not Significant"
                  )) %>%
  dplyr::arrange(W)
df_fig_w$taxon_id = factor(df_fig_w$taxon_id, levels = df_fig_w$taxon_id)
df_fig_w$W = replace(df_fig_w$W, is.infinite(df_fig_w$W), n_taxa - 1)
df_fig_w$direct = factor(df_fig_w$direct, 
                     levels = c("Negative", "Positive", "Not Significant"))

p_w = df_fig_w %>%
  ggplot(aes(x = taxon_id, y = W, color = direct)) +
  geom_point(size = 2, alpha = 0.6) +
  labs(x = "Taxon", y = "W") +
  scale_color_discrete(name = NULL) + 
  geom_hline(yintercept = cut_off, linetype = "dotted", 
             color = "blue", size = 1.5) +
  geom_text(aes(x = 2, y = cut_off + 0.5, label = "W[0.7]"), 
            size = 5, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE) +
  theme_bw() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        panel.grid.major = element_blank())
p_w

```

```{r}

otu_table <-  aes_data %>% 
  filter(asv %in% troph_asvs) %>% 
  filter(passage == "G5") %>% 
  select(sample, count, Genus) %>% 
  group_by(Genus, sample) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  mutate(treat = substr(sample, 3, 3)) %>% 
  group_by(Genus, treat) %>% 
  mutate(prev = sum(total > 0)) %>% 
  ungroup() %>% 
  filter(prev > 2) %>% 
  select(-treat, -prev) %>% 
  pivot_wider(names_from = sample, values_from = total, values_fill = 0) %>% 
  as.data.frame()

rownames(otu_table) <- otu_table[, 1]
otu_table <- otu_table[, -1]

conds <- 
  otu_table %>% 
  colnames() %>% 
  substr(3, 3)


x.all <- aldex(otu_table, 
               conds, mc.samples=1000, test="t", effect=TRUE,
     include.sample.summary=FALSE, denom="all", verbose=FALSE, paired.test=FALSE)

par(mfrow=c(1,2))
aldex.plot(x.all, type="MA", test="welch", xlab="Log-ratio abundance",
    ylab="Difference")
aldex.plot(x.all, type="MW", test="welch", xlab="Dispersion",
    ylab="Difference")

par(mfrow=c(1,2))
plot(x.all$effect, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(x.all$effect, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red"))

plot(x.all$diff.btw, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(x.all$diff.btw, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

x.all_effect_1 <-
  x.all %>% 
  filter(abs(effect) > 0.5) %>% 
  as_tibble(rownames = "asv")


x.all_effect_1 %>% 
  ggplot(aes(y = reorder(asv, effect), x = effect)) +
  geom_point()

```

```{r aldex_genus_everything}

otu_table <- 
  aes_data %>% 
  filter(passage == "G5") %>% 
  select(Genus, count, sample) %>% 
  group_by(Genus, sample) %>% 
  summarize(total = sum(count), .groups = "drop") %>% 
  mutate(treat = substr(sample, 3, 3)) %>% 
  group_by(Genus, treat) %>% 
  mutate(prev = sum(total > 0)) %>% 
  ungroup() %>% 
  filter(prev > 2) %>% 
  select(-treat, -prev) %>% 
  pivot_wider(names_from = "sample", values_from = "total", values_fill = 0) %>% 
  as.data.frame()

rownames(otu_table) <- otu_table[, 1]

otu_table <- otu_table[, -1]



x.all <- aldex(otu_table, 
               conds, mc.samples=1000, test="t", effect=TRUE,
     include.sample.summary=FALSE, denom="all", verbose=FALSE, paired.test=FALSE)

par(mfrow=c(1,2))
aldex.plot(x.all, type="MA", test="welch", xlab="Log-ratio abundance",
    ylab="Difference")
aldex.plot(x.all, type="MW", test="welch", xlab="Dispersion",
    ylab="Difference")

par(mfrow=c(1,2))
plot(x.all$effect, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(x.all$effect, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red"))

plot(x.all$diff.btw, x.all$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(x.all$diff.btw, x.all$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

x.all_effect_1 <-
  x.all %>% 
  filter(abs(effect) > 1) %>% 
  as_tibble(rownames = "Genus")


x.all_effect_1 %>% 
  ggplot(aes(y = reorder(Genus, effect), x = effect)) +
  geom_point()

```


```{r ancova-clr}

gm <- function(x){
  
  exp(mean(log(x[x>0])))
  
}

rclr <- function(x) {
  log(x/gm(x))
}

aes_family <-
  aes_data %>%
  select(Family, count, sample, passage, treat) %>%
  group_by(Family, sample) %>%
  mutate(total = sum(count)) %>%
  select(-count) %>%
  ungroup() %>%
  distinct() %>%
  mutate(rep = as.numeric(substr(sample, 4, 5))) %>%
  select(-sample)

prevalent_families <- 
  aes_family %>% 
  mutate(passagetreat = paste0(passage, treat)) %>%
  group_by(Family, passagetreat) %>% 
  summarize(prev = sum(total > 0), .groups = "drop") %>% 
  pivot_wider(names_from = "passagetreat", values_from = "prev", values_fill = 0) %>% 
  filter(G2N/12 > 0.1 & G2P/12 > 0.1 & G5N/12 > 0.1 & G5P/12 > 0.1) %$%
  Family

aes_family_rclr <-
  aes_family %>% 
  filter(Family %in% prevalent_families) %>% 
  group_by(passage, treat, rep) %>%
  mutate(rclr = log(total/gm(total))) %>%
  select(-total) %>%
  ungroup() %>%
  pivot_wider(names_from = passage, values_from = rclr)

ancova <- function(df) {
  lm(G5 ~ G2 + treat, data = df, na.action = )
}

model_results <-
aes_family_rclr %>%
  nest(data = -Family) %>%
  mutate(model = map(data, ancova)) %>%
  mutate(tidy = map(model, tidy)) %>% 
  unnest(tidy) %>% 
  filter(term == "treatP")

model_results$p.adjust <- p.adjust(model_results$p.value, method = "BH")

model_results %>%
  filter(p.adjust < 0.05) %>% 
  arrange(estimate) %>% 
  print(n = 22)
  
model_results %>% 
  filter(p.adjust < 0.05) %>% 
  arrange(estimate) %>% 
  unnest(data) %>% 
  ggplot(aes(x = estimate, xmax = estimate + std.error, xmin = estimate - std.error, y = reorder(Family, estimate))) +
  geom_pointrange() +
  labs(x = "Estimate \n(log-ratio increase in P)", y = NULL) +
  theme_bw()
ggsave("ancova-plot.png", width = 5, height = 5)

model_results %>% 
  filter(p.adjust < 0.05) %>% 
  unnest(data) %>% 
  pivot_longer(G2:G5, names_to = "passage", values_to = "rclr") %>% 
  select(Family, treat, rep, passage, rclr) %>% 
  distinct() %>% 
  ggplot(aes(x = rclr, y = Family, color = treat)) +
  facet_grid( ~ passage) +
  geom_point(position = position_jitterdodge())


```

```{r}

aes_family_data <-
  aes_data %>% 
  select(asv, sample, passage, treat, count, Family) %>%
  group_by(sample, Family) %>% 
  mutate(total = sum(count)) %>% 
  ungroup() %>% 
  select(sample, treat, passage, Family, total) %>% 
  distinct()
  
prevalent_families <- 
  aes_family_data %>% 
  group_by(Family, treat, passage) %>% 
  summarize(prev = sum(total > 0), .groups = "drop") %>% 
  pivot_wider(names_from = "treat", values_from = "prev", values_fill = 0) %>% 
  filter(N/12 > 0.1 & P/12 > 0.1) %$%
  Family

aes_family_data <-
  aes_family_data %>% 
  filter(Family %in% prevalent_families) %>% 
  mutate(jar = substr(sample, 3, 5)) %>%
  select(-sample) %>%
  pivot_wider(names_from = "passage", values_from = "total", values_fill = 0)

ancova_model <- function(df) {
	lm(G5 ~ G2 + treat, data = df)
}

aes_family_nested <-
	aes_family_data %>%
	group_by(Family) %>%
	nest() %>%
	mutate(model = map(data, ancova_model)) %>%
	mutate(tidy = map(model, tidy)) %>%
	ungroup()

aes_ancova <- aes_family_nested %>%
	unnest(tidy) %>%
	filter(term == "treatP")

aes_ancova$p.adjust <- p.adjust(aes_ancova$p.value, method = "BH")

aes_ancova %>%
	arrange(estimate) %>%
	filter(p.adjust < 0.05) %>%
	print(n = 22)

```

```{r}

aes_family_data <-
  aes_data %>% 
  select(asv, sample, passage, treat, count, Family) %>%
  group_by(sample, Family) %>% 
  mutate(total = sum(count)) %>% 
  ungroup() %>% 
  select(sample, treat, passage, Family, total) %>% 
  distinct()

prevalent_families <- 
  aes_family_data %>% 
  group_by(Family, treat, passage) %>% 
  summarize(prev = sum(total > 0), .groups = "drop") %>% 
  pivot_wider(names_from = "treat", values_from = "prev", values_fill = 0) %>% 
  filter(N/12 > 0.1 & P/12 > 0.1) %$%
  Family

otu_table <- 
  aes_family_data %>% 
  filter(Family %in% prevalent_families) %>% 
  select(sample, Family, total) %>% 
  pivot_wider(names_from = sample, values_from = total, values_fill = 0) %>% 
  as.data.frame()

rownames(otu_table) <- otu_table[,1]
otu_table <- otu_table[,-1]

sample_data <- aes_family_data %>% 
  select(sample, treat, passage) %>% 
  distinct() %>% 
  as.data.frame()

rownames(sample_data) <- sample_data[,1]
sample_data <- sample_data[, -1]
sample_data <-
	sample_data %>%
	mutate(passage12 = if_else(passage == "G2", 1, 2))

# Setup phyloseq object
aes_ps <- phyloseq(otu_table(otu_table, taxa_are_rows=TRUE), sample_data(sample_data))

set.seed(123)
out2 = ancom(phyloseq = aes_ps,  p_adj_method = "holm",
             prv_cut = 0.10, lib_cut = 1000, main_var = "treat", 
             adj_formula = "passage12", 
             rand_formula = "~ passage12 | 1",
             lme_control = list(maxIter = 100, msMaxIter = 100, opt = "optim"), 
             struc_zero = TRUE, neg_lb = FALSE, alpha = 0.05, n_cl = 2)

res2 = out2$res

out2 = ancom(phyloseq = family_data2,  p_adj_method = "holm",
             prv_cut = 0.10, lib_cut = 1000, main_var = "group", 
             adj_formula = "nationality + timepoint + bmi_group", 
             rand_formula = "~ timepoint | subject",
             lme_control = list(maxIter = 100, msMaxIter = 100, opt = "optim"), 
             struc_zero = TRUE, neg_lb = TRUE, alpha = 0.05, n_cl = 2)

res2 = out2$res

```

```{r}

x.all_effect_1$Family

dt_data$taxa

```

```{r}
combined_da <-
  dt_data %>% 
  as_tibble %>% 
  full_join(sig_families, by = c("taxa" = "Family")) %>% 
  full_join(x.all_effect_1, by = c("taxa" = "Family")) %>% 
  full_join(df_fig_w[df_fig_w$detected_0.7,], by = c("taxa" = "taxon_id"))

shared_da <-
  dt_data %>% 
  as_tibble %>% 
  inner_join(sig_families, by = c("taxa" = "Family")) %>% 
  inner_join(x.all_effect_1, by = c("taxa" = "Family")) %>% 
  inner_join(df_fig_w[df_fig_w$detected_0.7,], by = c("taxa" = "taxon_id"))


shared_da %>% 
  ggplot(aes(y = reorder(taxa, effect), x = effect)) +
  geom_point()

aes_data %>% 
  filter(passage == "G5") %>% 
  select(sample, Family, count) %>% 
  group_by(sample, Family) %>% 
  summarize(total = sum(count)) %>% 
  filter(Family %in% shared_da$taxa) %>% 
  mutate(treat = substr(sample, 3, 3)) %>% 
  ggplot(aes(x = total, y = Family, color = treat)) +
  geom_jitter() +
  scale_x_log10()

rubini_seqs <-
  aes_data %>% 
  filter(Family == "Rubinisphaeraceae") %>% 
  select(asv, seq, Kingdom:Species) %>% 
  distinct()

write(rubini_seqs$seq, file = 'Output/rubini_seqs.txt')

```

```{r}

##############

First, I looked at which taxa are unique to treatments N and P. Several families
present in P and absent in N. Most of them had low prevalence (present in 3 or
fewer samples) and low abundance (median < 5 reads across samples). There were two families unique
to the Positive treatment with relatively high prevalence. This included an ASV
that was a member of the Bacteroidia Class with no lower taxonomic designation. 
This ASV had a prevalence of 10/12 and a median abundance of ~2 reads. The other
prevalent family was a member of the Silvanigrellaceae, a newly described family placed in
its own order. This family was present in
all 12 samples and was represented by three OTUs: one with no lower designation, one from the genus
Silvanigrella and one from the genus Candidatus Turbobacter. Silvanigrella was
isolated from a temperate fresh water lake (https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.001965).

Of the families unique to the Neutral treatment, only one had a prevalence greater than 2/12. 
This family, Armatimonadaceae, was present in half of the Neutral samples (6/12). This
family was represented by two ASVs in the genus Armatimonas. The type strain for
Armatimonas was isolated from the rhizoplane of Phragmites australis (https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijs.0.025643-0)
```

```{r}
# families in to Passage 5, Treatment P
g5p<- 
  ps_fixed %>% 
  ps_filter(treat == "p", passage == 5) %>% 
  tax_agg('Family') %>% 
  ps_get() %>% 
  ps_melt()

# families in to Passage 5, Treatment N
g5n<- 
  ps_fixed %>% 
  ps_filter(treat == "n", passage == 5) %>% 
  tax_agg('Family') %>% 
  ps_get() %>% 
  ps_melt()

# Families unique to p in passage 5
g5p_unique <-
  g5p %>% 
  filter(Family %in% unique(g5p$Family)[!(unique(g5p$Family) %in% unique(g5n$Family))]) %>% 
  mutate(Prevalence = as.numeric(Abundance > 0))

# Abundance
Family_order<-
  g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(median = median(Abundance), max = max(Abundance)) %>% 
  arrange(median, max)
g5p_unique$OTU <- factor(g5p_unique$OTU, levels = Family_order$OTU)

# Plot families unique to P5 in order of median abundance
g5p_unique %>% 
  ggplot(aes(y = Abundance, x = OTU)) +
  geom_tufteboxplot() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(0, 5, 10, 15, 20, 24)) +
  labs(title = "Unique to P5 by abundance")


# Prevalence
Family_order<-
  g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(sum = sum(Prevalence)) %>% 
  arrange(sum)
g5p_unique$OTU <- factor(g5p_unique$OTU, levels = Family_order$OTU)

# Plot families unique to P5 in order of prevalence
g5p_unique %>% 
  group_by(OTU) %>% 
  summarize(Prevalence = sum(Prevalence)) %>% 
  ggplot(aes(y = Prevalence, x = OTU)) +
  geom_point() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(1, 2, 3, 5, 10, 12)) +
  labs(title = "Unique to P5 by prevalence")

# Families unique to n in passage 5
g5n_unique <-
  g5n %>% 
  filter(Family %in% unique(g5n$Family)[!(unique(g5n$Family) %in% unique(g5p$Family))]) %>% 
  mutate(Prevalence = as.numeric(Abundance > 0))

# Abundance
Family_order<-
  g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(median = median(Abundance), max = max(Abundance)) %>% 
  arrange(median, max)
g5n_unique$OTU <- factor(g5n_unique$OTU, levels = Family_order$OTU)

# Plot families unique to N5 in order of abundance
g5n_unique %>% 
  ggplot(aes(y = Abundance, x = OTU)) +
  geom_tufteboxplot() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(0, 10, 20, 30, 37)) +
  labs(title = "Unique to N5 by abundance")


# Prevalence
Family_order<-
  g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(sum = sum(Prevalence)) %>% 
  arrange(sum)
g5n_unique$OTU <- factor(g5n_unique$OTU, levels = Family_order$OTU)

# Plot families unique to N5 in order of prevalence

g5n_unique %>% 
  group_by(OTU) %>% 
  summarize(Prevalence = sum(Prevalence)) %>% 
  ggplot(aes(y = Prevalence, x = OTU)) +
  geom_point() +
  theme_tufte() +
  geom_rangeframe() +
  theme(axis.text.x = element_text(angle = 90)) +
  scale_y_continuous(breaks = c(1, 2, 6)) +
  labs(title = "Unique to N5 by prevalence")

```

```{r}
# Differential Abundance Treatment in passage 5
```

```{r corncob_treatment, results=FALSE, fig.hight=40}

ps_fixed_family <- speedyseq::tax_glom(ps_fixed, taxrank="Family")

ps_fam_pas5 <- microViz::ps_filter(ps_fixed_family, passage == 5)

ps_fam_pas5_prev <- microViz::tax_filter(ps_fam_pas5, min_prevalence = 0.10)

sample_data(ps_fam_pas5_prev) =
  sample_data(ps_fam_pas5_prev)[,c('sample_id', 'treat')] %>% 
  sample_data()
ps = ps_fam_pas5_prev

# Differential abundance test
da_analysis <- differentialTest(formula = ~ treat,
                                 phi.formula = ~ 1,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ 1,
                                 test = "Wald", boot = FALSE,
                                 data = ps_fam_pas5_prev,
                                 fdr_cutoff = 0.05, B = 50)

#plot(da_analysis, level = c("Family"))
da_analysis$significant_models
dt_data <- plot(da_analysis, level = c("Family"), data_only = TRUE)

sig_taxa_treat <- tax_select(ps_fam_pas5_prev, da_analysis$significant_taxa)
ntaxa(sig_taxa_treat)
save(da_analysis, sig_taxa_treat, dt_data, file = 'Output/da_corncob.RData')

```

```{r corncob-plot, fig.width=6, fig.height=6}
dt_plot(dt_data)

```



```{r corncob_ch4}

ps_fam_pas5_trep <- ps_filter(ps_fam_pas5, treat == "p")


ps_fam_prev_trep_prev <- tax_filter(ps_fam_pas5_trep, min_prevalence = 0.10)


da_analysis <- differentialTest(formula = ~ ch4,
                                 phi.formula = ~ ch4,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ ch4,
                                 test = "Wald", boot = FALSE,
                                 data = ps_fam_prev_trep_prev,
                                 fdr_cutoff = 0.05)
otu_to_taxonomy(OTU = da_analysis$significant_taxa, data = ps_fam_prev_trep_prev)

plot(da_analysis, level = c("Family"))

# Check unfitted taxa
# otu_to_taxonomy(which(is.na(da_analysis$p)) %>% names, data = ps_gen_prev)
# otu_table(ps_gen_prev)[,which(is.na(da_analysis$p))]

# Differential variability test

dv_analysis <- differentialTest(formula = ~ passage + treat + ch4,
                                 phi.formula = ~ passage + treat + ch4,
                                 formula_null = ~ passage + treat + ch4,
                                 phi.formula_null = ~ passage + treat,
                                 data = ps_gen_prev,
                                 test = "Wald", boot = FALSE,
                                 fdr_cutoff = 0.05)
# otu_to_taxonomy(OTU = dv_analysis$significant_taxa, data = ps_gen_prev)
plot(dv_analysis, level = c("Family", "Genus"))

# Check unfitted taxa

# otu_to_taxonomy(which(is.na(dv_analysis$p)) %>% names, data = ps_gen_prev)
# otu_table(ps_gen_prev)[,which(is.na(dv_analysis$p))]

```

